diff --git a/node_modules/react-native-worklets/Common/cpp/worklets/NativeModules/JSIWorkletsModuleProxy.cpp b/node_modules/react-native-worklets/Common/cpp/worklets/NativeModules/JSIWorkletsModuleProxy.cpp
index a7cb25ddf3..c6bd17a90d 100644
--- a/node_modules/react-native-worklets/Common/cpp/worklets/NativeModules/JSIWorkletsModuleProxy.cpp
+++ b/node_modules/react-native-worklets/Common/cpp/worklets/NativeModules/JSIWorkletsModuleProxy.cpp
@@ -321,8 +321,8 @@ jsi::Value JSIWorkletsModuleProxy::get(jsi::Runtime &rt, const jsi::PropNameID &
 
   if (name == "createSerializableFunction") {
     return jsi::Function::createFromHostFunction(
-        rt, propName, 1, [](jsi::Runtime &rt, const jsi::Value &thisValue, const jsi::Value *args, size_t count) {
-          return makeSerializableFunction(rt, args[0].asObject(rt).asFunction(rt));
+        rt, propName, 2, [](jsi::Runtime &rt, const jsi::Value &thisValue, const jsi::Value *args, size_t count) {
+          return makeSerializableFunction(rt, args[0].asObject(rt).asFunction(rt), args[1].asNumber());
         });
   }
 
diff --git a/node_modules/react-native-worklets/Common/cpp/worklets/SharedItems/Serializable.cpp b/node_modules/react-native-worklets/Common/cpp/worklets/SharedItems/Serializable.cpp
index 67ff75a9a5..6f8a29055d 100644
--- a/node_modules/react-native-worklets/Common/cpp/worklets/SharedItems/Serializable.cpp
+++ b/node_modules/react-native-worklets/Common/cpp/worklets/SharedItems/Serializable.cpp
@@ -31,7 +31,7 @@ jsi::Value makeSerializableClone(
     } else if (!object.getProperty(rt, "__init").isUndefined()) {
       return makeSerializableInitializer(rt, object);
     } else if (object.isFunction(rt)) {
-      return makeSerializableFunction(rt, object.asFunction(rt));
+      return makeSerializableFunction(rt, object.asFunction(rt), -1);
     } else if (object.isArray(rt)) {
       if (shouldRetainRemote.isBool() && shouldRetainRemote.getBool()) {
         serializable = std::make_shared<RetainingSerializable<SerializableArray>>(rt, object.asArray(rt));
@@ -121,12 +121,12 @@ jsi::Value makeSerializableInitializer(jsi::Runtime &rt, const jsi::Object &init
   return SerializableJSRef::newNativeStateObject(rt, serializable);
 }
 
-jsi::Value makeSerializableFunction(jsi::Runtime &rt, jsi::Function function) {
+jsi::Value makeSerializableFunction(jsi::Runtime &rt, jsi::Function function, double id) {
   std::shared_ptr<Serializable> serializable;
   if (function.isHostFunction(rt)) {
     serializable = std::make_shared<SerializableHostFunction>(rt, std::move(function));
   } else {
-    serializable = std::make_shared<SerializableRemoteFunction>(rt, std::move(function));
+    serializable = std::make_shared<SerializableRemoteFunction>(rt, id);
   }
   return SerializableJSRef::newNativeStateObject(rt, serializable);
 }
@@ -369,7 +369,11 @@ jsi::Value SerializableImport::toJSValue(jsi::Runtime &rt) {
 
 jsi::Value SerializableRemoteFunction::toJSValue(jsi::Runtime &rt) {
   if (&rt == runtime_) {
-    return jsi::Value(rt, *function_);
+    auto global = rt.global();
+    auto remoteFunctionCache = global.getPropertyAsObject(rt, "__remoteFunctionCache");
+    auto cached =
+        remoteFunctionCache.getPropertyAsFunction(rt, "get").callWithThis(rt, remoteFunctionCache, jsi::Value(id_));
+    return cached;
   } else {
 #ifndef NDEBUG
     return getValueUnpacker(rt).call(
diff --git a/node_modules/react-native-worklets/Common/cpp/worklets/SharedItems/Serializable.h b/node_modules/react-native-worklets/Common/cpp/worklets/SharedItems/Serializable.h
index 771b6be96c..5e60aa7e11 100644
--- a/node_modules/react-native-worklets/Common/cpp/worklets/SharedItems/Serializable.h
+++ b/node_modules/react-native-worklets/Common/cpp/worklets/SharedItems/Serializable.h
@@ -162,7 +162,7 @@ jsi::Value makeSerializableSet(jsi::Runtime &rt, const jsi::Array &values);
 
 jsi::Value makeSerializableInitializer(jsi::Runtime &rt, const jsi::Object &initializerObject);
 
-jsi::Value makeSerializableFunction(jsi::Runtime &rt, jsi::Function function);
+jsi::Value makeSerializableFunction(jsi::Runtime &rt, jsi::Function function, double id);
 
 jsi::Value makeSerializableWorklet(jsi::Runtime &rt, const jsi::Object &object, const bool &shouldRetainRemote);
 
@@ -295,21 +295,19 @@ class SerializableRemoteFunction : public Serializable,
 #ifndef NDEBUG
   const std::string name_;
 #endif
-  std::unique_ptr<jsi::Value> function_;
+  int id_;
 
  public:
-  SerializableRemoteFunction(jsi::Runtime &rt, jsi::Function &&function)
+  SerializableRemoteFunction(jsi::Runtime &rt, jsi::Value id)
       : Serializable(ValueType::RemoteFunctionType),
         runtime_(&rt),
 #ifndef NDEBUG
-        name_(function.getProperty(rt, "name").asString(rt).utf8(rt)),
+        name_("placeholder"),
 #endif
-        function_(std::make_unique<jsi::Value>(rt, std::move(function))) {
+        id_(id.asNumber()) {
   }
 
-  ~SerializableRemoteFunction() override {
-    cleanupIfRuntimeExists(runtime_, function_);
-  }
+  ~SerializableRemoteFunction() override {}
 
   jsi::Value toJSValue(jsi::Runtime &rt) override;
 };
diff --git a/node_modules/react-native-worklets/Common/cpp/worklets/WorkletRuntime/WorkletRuntimeDecorator.cpp b/node_modules/react-native-worklets/Common/cpp/worklets/WorkletRuntime/WorkletRuntimeDecorator.cpp
index d64c04647e..6d5b604d3a 100644
--- a/node_modules/react-native-worklets/Common/cpp/worklets/WorkletRuntime/WorkletRuntimeDecorator.cpp
+++ b/node_modules/react-native-worklets/Common/cpp/worklets/WorkletRuntime/WorkletRuntimeDecorator.cpp
@@ -141,7 +141,7 @@ void WorkletRuntimeDecorator::decorate(
   });
 
   jsi_utils::installJsiFunction(rt, "_createSerializableFunction", [](jsi::Runtime &rt, const jsi::Value &value) {
-    return makeSerializableFunction(rt, value.asObject(rt).asFunction(rt));
+    return makeSerializableFunction(rt, value.asObject(rt).asFunction(rt), -1);
   });
 
   jsi_utils::installJsiFunction(rt, "_createSerializableSynchronizable", [](jsi::Runtime &rt, const jsi::Value &value) {
diff --git a/node_modules/react-native-worklets/src/WorkletsModule/NativeWorklets.native.ts b/node_modules/react-native-worklets/src/WorkletsModule/NativeWorklets.native.ts
index 9a12fada57..ada9bf467e 100644
--- a/node_modules/react-native-worklets/src/WorkletsModule/NativeWorklets.native.ts
+++ b/node_modules/react-native-worklets/src/WorkletsModule/NativeWorklets.native.ts
@@ -140,9 +140,10 @@ See https://docs.swmansion.com/react-native-worklets/docs/guides/troubleshooting
   }
 
   createSerializableFunction<TArgs extends unknown[], TReturn>(
-    func: (...args: TArgs) => TReturn
-  ) {
-    return this.#workletsModuleProxy.createSerializableFunction(func);
+    func: (...args: TArgs) => TReturn,
+    id: number
+  ): SerializableRef<TReturn> {
+    return this.#workletsModuleProxy.createSerializableFunction(func, id);
   }
 
   createSerializableWorklet(worklet: object, shouldPersistRemote: boolean) {
diff --git a/node_modules/react-native-worklets/src/WorkletsModule/workletsModuleProxy.ts b/node_modules/react-native-worklets/src/WorkletsModule/workletsModuleProxy.ts
index b7d9fb2f11..c9bfd6dca1 100644
--- a/node_modules/react-native-worklets/src/WorkletsModule/workletsModuleProxy.ts
+++ b/node_modules/react-native-worklets/src/WorkletsModule/workletsModuleProxy.ts
@@ -61,7 +61,8 @@ export interface WorkletsModuleProxy {
   createSerializableInitializer(obj: object): SerializableRef<object>;
 
   createSerializableFunction<TArgs extends unknown[], TReturn>(
-    func: (...args: TArgs) => TReturn
+    func: (...args: TArgs) => TReturn,
+    id: number
   ): SerializableRef<TReturn>;
 
   createSerializableWorklet(
diff --git a/node_modules/react-native-worklets/src/memory/serializable.native.ts b/node_modules/react-native-worklets/src/memory/serializable.native.ts
index 1b5f579011..fb82e3127e 100644
--- a/node_modules/react-native-worklets/src/memory/serializable.native.ts
+++ b/node_modules/react-native-worklets/src/memory/serializable.native.ts
@@ -428,10 +428,18 @@ function cloneArray<T extends unknown[]>(
   return clone;
 }
 
+// eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
+const remoteFunctionCache = new Map<number, Function>();
+// @ts-expect-error it's ok
+globalThis.__remoteFunctionCache = remoteFunctionCache;
+let index = 0;
+
 function cloneRemoteFunction<TArgs extends unknown[], TReturn>(
   value: (...args: TArgs) => TReturn
 ): SerializableRef<TReturn> {
-  const clone = WorkletsModule.createSerializableFunction(value);
+  const currentIndex = index++;
+  remoteFunctionCache.set(currentIndex, value);
+  const clone = WorkletsModule.createSerializableFunction(value, currentIndex);
   serializableMappingCache.set(value, clone);
   serializableMappingCache.set(clone);
 
