diff --git a/node_modules/@types/react-native-web/index.d.ts b/node_modules/@types/react-native-web/index.d.ts
index dd91bb1..d97ca3c 100644
--- a/node_modules/@types/react-native-web/index.d.ts
+++ b/node_modules/@types/react-native-web/index.d.ts
@@ -1389,14 +1389,36 @@ declare module "react-native" {
             | AriaRole;
         // @deprecated
         accessibilityRole?:
+            | "none"
             | "button"
-            | "header"
-            | "heading"
-            | "label"
+            | "togglebutton"
             | "link"
-            | "listitem"
-            | "none"
-            | "text";
+            | "search"
+            | "image"
+            | "keyboardkey"
+            | "text"
+            | "adjustable"
+            | "imagebutton"
+            | "header"
+            | "summary"
+            | "alert"
+            | "checkbox"
+            | "combobox"
+            | "menu"
+            | "menubar"
+            | "menuitem"
+            | "progressbar"
+            | "radio"
+            | "radiogroup"
+            | "scrollbar"
+            | "spinbutton"
+            | "switch"
+            | "tab"
+            | "tabbar"
+            | "tablist"
+            | "timer"
+            | "list"
+            | "toolbar";
     }
 
     /**
@@ -1481,6 +1503,7 @@ declare module "react-native" {
         paddingRight?: CSSProperties["paddingRight"] | DimensionValue | undefined;
         paddingBottom?: CSSProperties["paddingBottom"] | DimensionValue | undefined;
         paddingLeft?: CSSProperties["paddingLeft"] | DimensionValue | undefined;
+        userSelect?: CSSProperties["userSelect"] | 'contain' | undefined;
     }
 
     interface TextStyle extends WebStyle {
