diff --git a/node_modules/@callstack/repack/dist/rules/nodeModulesLoadingRules.js b/node_modules/@callstack/repack/dist/rules/nodeModulesLoadingRules.js
index 54de83c..7f5e518 100644
--- a/node_modules/@callstack/repack/dist/rules/nodeModulesLoadingRules.js
+++ b/node_modules/@callstack/repack/dist/rules/nodeModulesLoadingRules.js
@@ -11,7 +11,8 @@ const makeSwcLoaderConfig = (syntax, jsx) => ({
     env: {
       targets: {
         'react-native': '0.74'
-      }
+      },
+      exclude: ["transform-property-literals"],
     },
     jsc: {
       assumptions: {
@@ -30,7 +31,7 @@ const makeSwcLoaderConfig = (syntax, jsx) => ({
         react: {
           runtime: 'automatic'
         }
-      }
+      },
     },
     module: {
       type: 'commonjs',
