diff --git a/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/LiveMarkdownModule.java b/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/LiveMarkdownModule.java
index ed1428b..80641ce 100644
--- a/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/LiveMarkdownModule.java
+++ b/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/LiveMarkdownModule.java
@@ -1,7 +1,6 @@
 package com.expensify.livemarkdown;
 
 import com.facebook.react.bridge.ReactApplicationContext;
-import com.facebook.react.bridge.UIManager;
 import com.facebook.react.fabric.FabricUIManager;
 import com.facebook.react.uimanager.UIManagerHelper;
 import com.facebook.react.uimanager.common.UIManagerType;
diff --git a/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java b/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java
index 7711d8b..0000caa 100644
--- a/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java
+++ b/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java
@@ -1,6 +1,5 @@
 package com.expensify.livemarkdown;
 
-import static com.facebook.infer.annotation.ThreadConfined.UI;
 
 import android.content.res.AssetManager;
 import android.text.SpannableStringBuilder;
@@ -8,10 +7,7 @@ import android.text.Spanned;
 
 import androidx.annotation.NonNull;
 
-import com.facebook.infer.annotation.Assertions;
-import com.facebook.infer.annotation.ThreadConfined;
-import com.facebook.react.bridge.UiThreadUtil;
-import com.facebook.react.views.text.CustomLineHeightSpan;
+import com.facebook.react.views.text.internal.span.CustomLineHeightSpan;
 import com.facebook.soloader.SoLoader;
 
 import org.json.JSONArray;
diff --git a/node_modules/@expensify/react-native-live-markdown/android/src/main/new_arch/CMakeLists.txt b/node_modules/@expensify/react-native-live-markdown/android/src/main/new_arch/CMakeLists.txt
index f5dfedf..1609c60 100644
--- a/node_modules/@expensify/react-native-live-markdown/android/src/main/new_arch/CMakeLists.txt
+++ b/node_modules/@expensify/react-native-live-markdown/android/src/main/new_arch/CMakeLists.txt
@@ -42,6 +42,8 @@ target_link_libraries(
   ReactAndroid::rrc_textinput
   ReactAndroid::react_render_textlayoutmanager
   ReactAndroid::react_render_imagemanager
+  ReactAndroid::reactnativejni
+  ReactAndroid::mapbufferjni
   fabricjni
   fbjni
   folly_runtime
