diff --git a/node_modules/@expensify/react-native-live-markdown/RNLiveMarkdown.podspec b/node_modules/@expensify/react-native-live-markdown/RNLiveMarkdown.podspec
index b1620ad..b3ea39c 100644
--- a/node_modules/@expensify/react-native-live-markdown/RNLiveMarkdown.podspec
+++ b/node_modules/@expensify/react-native-live-markdown/RNLiveMarkdown.podspec
@@ -23,10 +23,10 @@ Pod::Spec.new do |s|
   install_modules_dependencies(s)
 
   if ENV['USE_FRAMEWORKS'] && ENV['RCT_NEW_ARCH_ENABLED']
-    add_dependency(s, "React-Fabric", :additional_framework_paths => [
+    add_dependency(s, "React-FabricComponents", :additional_framework_paths => [
       "react/renderer/textlayoutmanager/platform/ios",
-      "react/renderer/components/textinput/iostextinput",
-    ])
+      "react/renderer/components/textinput/platform/ios",
+    ]);
   end
 
   s.subspec "common" do |ss|
