diff --git a/node_modules/@expensify/react-native-live-markdown/lib/module/MarkdownTextInput.web.js b/node_modules/@expensify/react-native-live-markdown/lib/module/MarkdownTextInput.web.js
index f218295..8bee3ff 100644
--- a/node_modules/@expensify/react-native-live-markdown/lib/module/MarkdownTextInput.web.js
+++ b/node_modules/@expensify/react-native-live-markdown/lib/module/MarkdownTextInput.web.js
@@ -396,6 +396,9 @@ const MarkdownTextInput = /*#__PURE__*/React.forwardRef(({
   }, [onClick, updateSelection]);
   const handlePaste = useCallback(e => {
     pasteRef.current = true;
+    if(e.isDefaultPrevented()){
+      return;
+    }
     e.preventDefault();
     const clipboardData = e.clipboardData;
     const text = clipboardData.getData('text/plain');
