diff --git a/node_modules/@expensify/react-native-live-markdown/src/web/cursorUtils.ts b/node_modules/@expensify/react-native-live-markdown/src/web/cursorUtils.ts
index 1cda659..ba5c3c3 100644
--- a/node_modules/@expensify/react-native-live-markdown/src/web/cursorUtils.ts
+++ b/node_modules/@expensify/react-native-live-markdown/src/web/cursorUtils.ts
@@ -66,7 +66,7 @@ function setCursorPosition(target: HTMLElement, start: number, end: number | nul
         // 3. Caret at the end of whole input, when pressing enter
         // 4. All other placements
         if (prevChar === '\n' && prevTextLength !== undefined && prevTextLength < textCharacters.length) {
-          if (nextChar !== '\n') {
+          if (nextChar && nextChar !== '\n' && i !== n - 1) {
             range.setStart(textNodes[i + 1] as Node, 0);
           } else if (i !== textNodes.length - 1) {
             range.setStart(textNodes[i] as Node, 1);
