diff --git a/node_modules/@expo/cli/build/src/export/embed/index.js b/node_modules/@expo/cli/build/src/export/embed/index.js
index eed477f..00e1787 100644
--- a/node_modules/@expo/cli/build/src/export/embed/index.js
+++ b/node_modules/@expo/cli/build/src/export/embed/index.js
@@ -80,6 +80,7 @@ const expoExportEmbed = async (argv)=>{
         "--assets-dest": String,
         "--asset-catalog-dest": String,
         "--unstable-transform-profile": String,
+        '--config-cmd': String,
         "--config": String,
         // This is here for compatibility with the `npx react-native bundle` command.
         // devs should use `DEBUG=expo:*` instead.
