diff --git a/node_modules/@expensify/react-native-live-markdown/cpp/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.cpp b/node_modules/@expensify/react-native-live-markdown/cpp/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.cpp
index 104363d..9240e9e 100644
--- a/node_modules/@expensify/react-native-live-markdown/cpp/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.cpp
+++ b/node_modules/@expensify/react-native-live-markdown/cpp/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.cpp
@@ -11,7 +11,7 @@ namespace react {
 extern const char MarkdownTextInputDecoratorViewComponentName[] =
     "MarkdownTextInputDecoratorView";
 
-const ShadowNodeFragment::Value
+const OwningShadowNodeFragment
 MarkdownTextInputDecoratorShadowNode::updateFragmentState(
     ShadowNodeFragment const &fragment,
     ShadowNodeFamily::Shared const &family) {
@@ -24,12 +24,12 @@ MarkdownTextInputDecoratorShadowNode::updateFragmentState(
   // propagated on every clone we need it to clear the reference in the registry
   // when the view is removed from window it cannot be done in the destructor,
   // as multiple shadow nodes for the same family may be created
-  return ShadowNodeFragment::Value({
+  return OwningShadowNodeFragment{
       .props = fragment.props,
       .children = fragment.children,
       .state =
           std::make_shared<const ConcreteState>(newStateData, *fragment.state),
-  });
+  };
 }
 
 } // namespace react
diff --git a/node_modules/@expensify/react-native-live-markdown/cpp/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.h b/node_modules/@expensify/react-native-live-markdown/cpp/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.h
index 294e0d3..597752c 100644
--- a/node_modules/@expensify/react-native-live-markdown/cpp/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.h
+++ b/node_modules/@expensify/react-native-live-markdown/cpp/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.h
@@ -11,6 +11,20 @@
 namespace facebook {
 namespace react {
 
+struct OwningShadowNodeFragment {
+  Props::Shared props;
+  ShadowNode::SharedListOfShared children;
+  State::Shared state;
+  
+  operator ShadowNodeFragment() const {
+    return ShadowNodeFragment {
+      .props = props,
+      .children = children,
+      .state = state
+    };
+  }
+};
+
 JSI_EXPORT extern const char MarkdownTextInputDecoratorViewComponentName[];
 
 class JSI_EXPORT MarkdownTextInputDecoratorShadowNode final
@@ -22,8 +36,7 @@ public:
   MarkdownTextInputDecoratorShadowNode(ShadowNodeFragment const &fragment,
                                        ShadowNodeFamily::Shared const &family,
                                        ShadowNodeTraits traits)
-      : ConcreteViewShadowNode(static_cast<ShadowNodeFragment>(
-                                   updateFragmentState(fragment, family)),
+      : ConcreteViewShadowNode(updateFragmentState(fragment, family),
                                family, traits) {}
 
   MarkdownTextInputDecoratorShadowNode(ShadowNode const &sourceShadowNode,
@@ -37,7 +50,7 @@ public:
   }
 
 private:
-  static const ShadowNodeFragment::Value
+  static const OwningShadowNodeFragment
   updateFragmentState(ShadowNodeFragment const &fragment,
                       ShadowNodeFamily::Shared const &family);
 };
