diff --git a/node_modules/@expensify/react-native-live-markdown/android/src/main/new_arch/MarkdownCommitHook.cpp b/node_modules/@expensify/react-native-live-markdown/android/src/main/new_arch/MarkdownCommitHook.cpp
index 51029b4..4153df7 100644
--- a/node_modules/@expensify/react-native-live-markdown/android/src/main/new_arch/MarkdownCommitHook.cpp
+++ b/node_modules/@expensify/react-native-live-markdown/android/src/main/new_arch/MarkdownCommitHook.cpp
@@ -6,8 +6,6 @@
 #include "MarkdownCommitHook.h"
 #include "react/renderer/components/RNLiveMarkdownSpec/MarkdownShadowFamilyRegistry.h"
 
-#include <android/Log.h>
-
 using namespace facebook;
 using namespace react;
 
