diff --git a/node_modules/@expensify/react-native-live-markdown/RNLiveMarkdown.podspec b/node_modules/@expensify/react-native-live-markdown/RNLiveMarkdown.podspec
index 21de59d..7165118 100644
--- a/node_modules/@expensify/react-native-live-markdown/RNLiveMarkdown.podspec
+++ b/node_modules/@expensify/react-native-live-markdown/RNLiveMarkdown.podspec
@@ -24,7 +24,7 @@ Pod::Spec.new do |s|
 
   s.source_files = "apple/**/*.{h,m,mm}", "cpp/**/*.{h,cpp}"
 
-  s.dependency "RNReanimated/worklets"
+  s.dependency "RNWorklets"
 
   s.xcconfig = {
     "OTHER_CFLAGS" => "$(inherited) -DREACT_NATIVE_MINOR_VERSION=#{react_native_minor_version}",
diff --git a/node_modules/@expensify/react-native-live-markdown/lib/typescript/src/MarkdownTextInput.d.ts b/node_modules/@expensify/react-native-live-markdown/lib/typescript/src/MarkdownTextInput.d.ts
index e6ad088..90f2487 100644
--- a/node_modules/@expensify/react-native-live-markdown/lib/typescript/src/MarkdownTextInput.d.ts
+++ b/node_modules/@expensify/react-native-live-markdown/lib/typescript/src/MarkdownTextInput.d.ts
@@ -2,7 +2,7 @@ import { TextInput } from 'react-native';
 import React from 'react';
 import type { TextInputProps } from 'react-native';
 import type { WorkletRuntime } from 'react-native-reanimated';
-import type { ShareableRef, WorkletFunction } from 'react-native-reanimated/lib/typescript/commonTypes';
+import type { ShareableRef, WorkletFunction } from 'react-native-worklets';
 import type { PartialMarkdownStyle } from './styleUtils';
 import type { InlineImagesInputProps, MarkdownRange } from './commonTypes';
 declare global {
diff --git a/node_modules/@expensify/react-native-live-markdown/src/MarkdownTextInput.tsx b/node_modules/@expensify/react-native-live-markdown/src/MarkdownTextInput.tsx
index 82ee895..55c0e90 100644
--- a/node_modules/@expensify/react-native-live-markdown/src/MarkdownTextInput.tsx
+++ b/node_modules/@expensify/react-native-live-markdown/src/MarkdownTextInput.tsx
@@ -3,7 +3,7 @@ import React from 'react';
 import type {TextInputProps} from 'react-native';
 import {createWorkletRuntime, makeShareableCloneRecursive} from 'react-native-reanimated';
 import type {WorkletRuntime} from 'react-native-reanimated';
-import type {ShareableRef, WorkletFunction} from 'react-native-reanimated/lib/typescript/commonTypes';
+import type {ShareableRef, WorkletFunction} from 'react-native-worklets';
 
 import MarkdownTextInputDecoratorViewNativeComponent from './MarkdownTextInputDecoratorViewNativeComponent';
 import type {MarkdownStyle} from './MarkdownTextInputDecoratorViewNativeComponent';
diff --git a/node_modules/@expensify/react-native-live-markdown/src/parseExpensiMark.ts b/node_modules/@expensify/react-native-live-markdown/src/parseExpensiMark.ts
index 6e14e25..38d92bb 100644
--- a/node_modules/@expensify/react-native-live-markdown/src/parseExpensiMark.ts
+++ b/node_modules/@expensify/react-native-live-markdown/src/parseExpensiMark.ts
@@ -4,7 +4,7 @@ import {Platform} from 'react-native';
 import {ExpensiMark} from 'expensify-common';
 import {unescapeText} from 'expensify-common/dist/utils';
 import {decode} from 'html-entities';
-import type {WorkletFunction} from 'react-native-reanimated/lib/typescript/commonTypes';
+import type {WorkletFunction} from 'react-native-worklets';
 import type {MarkdownType, MarkdownRange} from './commonTypes';
 import {groupRanges, sortRanges, splitRangesOnEmojis} from './rangeUtils';
 
