diff --git a/node_modules/@expensify/react-native-wallet/android/src/main/java/com/expensify/wallet/Utils.kt b/node_modules/@expensify/react-native-wallet/android/src/main/java/com/expensify/wallet/Utils.kt
index 6bb09ef..120b8e3 100644
--- a/node_modules/@expensify/react-native-wallet/android/src/main/java/com/expensify/wallet/Utils.kt
+++ b/node_modules/@expensify/react-native-wallet/android/src/main/java/com/expensify/wallet/Utils.kt
@@ -56,21 +56,21 @@ object Utils {
               )
             }

-            override fun reject(code: String, userInfo: WritableMap) {
+            override fun reject(code: String?, userInfo: WritableMap) {
               val errorMessage = "Error: $code\nUserInfo: $userInfo"
               continuation.resumeWithException(
                 Exception(errorMessage)
               )
             }

-            override fun reject(code: String, message: String?) {
+            override fun reject(code: String?, message: String?) {
               val errorMessage = "Error: $code\nMessage: ${message ?: "No message provided"}"
               continuation.resumeWithException(
                 Exception(errorMessage)
               )
             }

-            override fun reject(code: String, message: String?, userInfo: WritableMap) {
+            override fun reject(code: String?, message: String?, userInfo: WritableMap) {
               val errorMessage =
                 "Error: $code\nMessage: ${message ?: "No message provided"}\nUserInfo: $userInfo"
               continuation.resumeWithException(
@@ -78,21 +78,21 @@ object Utils {
               )
             }

-            override fun reject(code: String, message: String?, throwable: Throwable?) {
+            override fun reject(code: String?, message: String?, throwable: Throwable?) {
               val errorMessage = "Error: $code\nMessage: ${message ?: "No message provided"}"
               continuation.resumeWithException(
                 throwable ?: Exception(errorMessage)
               )
             }

-            override fun reject(code: String, throwable: Throwable?) {
+            override fun reject(code: String?, throwable: Throwable?) {
               val errorMessage = "Error: $code"
               continuation.resumeWithException(
                 throwable ?: Exception(errorMessage)
               )
             }

-            override fun reject(code: String, throwable: Throwable?, userInfo: WritableMap) {
+            override fun reject(code: String?, throwable: Throwable?, userInfo: WritableMap) {
               val errorMessage = "Error: $code\nUserInfo: $userInfo"
               continuation.resumeWithException(
                 throwable ?: Exception(errorMessage)
