diff --git a/node_modules/@expensify/react-native-wallet/android/src/main/java/com/expensify/wallet/WalletModule.kt b/node_modules/@expensify/react-native-wallet/android/src/main/java/com/expensify/wallet/WalletModule.kt
index 76035c8..61b7af9 100644
--- a/node_modules/@expensify/react-native-wallet/android/src/main/java/com/expensify/wallet/WalletModule.kt
+++ b/node_modules/@expensify/react-native-wallet/android/src/main/java/com/expensify/wallet/WalletModule.kt
@@ -64,7 +64,7 @@ class WalletModule internal constructor(context: ReactApplicationContext) :
 
   private val cardListener = object : ActivityEventListener {
     override fun onActivityResult(
-      activity: Activity?, requestCode: Int, resultCode: Int, intent: Intent?
+      activity: Activity, requestCode: Int, resultCode: Int, intent: Intent?
     ) {
       if (requestCode == REQUEST_CREATE_WALLET) {
         pendingCreateWalletPromise?.resolve(resultCode == RESULT_OK)
@@ -91,7 +91,7 @@ class WalletModule internal constructor(context: ReactApplicationContext) :
       }
     }
 
-    override fun onNewIntent(p0: Intent?) {}
+    override fun onNewIntent(p0: Intent) {}
   }
 
   @ReactMethod
