diff --git a/node_modules/@invertase/react-native-apple-authentication/android/build.gradle b/node_modules/@invertase/react-native-apple-authentication/android/build.gradle
index d034fb0..7e038fd 100644
--- a/node_modules/@invertase/react-native-apple-authentication/android/build.gradle
+++ b/node_modules/@invertase/react-native-apple-authentication/android/build.gradle
@@ -61,7 +61,7 @@ repositories {
   } else {
     defaultDir = new File(
             projectDir,
-            '/../../../node_modules/react-native/android'
+            '/../../../react-native/node_modules/react-native/android'
     )
   }
 
