diff --git a/node_modules/@perf-profiler/android/dist/src/commands/platforms/UnixProfiler.js b/node_modules/@perf-profiler/android/dist/src/commands/platforms/UnixProfiler.js
index 59aeed9..ee1d8a6 100644
--- a/node_modules/@perf-profiler/android/dist/src/commands/platforms/UnixProfiler.js
+++ b/node_modules/@perf-profiler/android/dist/src/commands/platforms/UnixProfiler.js
@@ -28,7 +28,7 @@ exports.CppProfilerName = `BAMPerfProfiler`;
 // into the Flipper plugin directory
 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
 // @ts-expect-error
-const binaryFolder = global.Flipper
+const binaryFolder = (global.Flipper || process.env.AWS)
     ? `${__dirname}/bin`
     : `${__dirname}/../../..${__dirname.includes("dist") ? "/.." : ""}/cpp-profiler/bin`;
 class UnixProfiler {
diff --git a/node_modules/@perf-profiler/android/src/commands/platforms/UnixProfiler.ts b/node_modules/@perf-profiler/android/src/commands/platforms/UnixProfiler.ts
index ccacf09..1eea659 100644
--- a/node_modules/@perf-profiler/android/src/commands/platforms/UnixProfiler.ts
+++ b/node_modules/@perf-profiler/android/src/commands/platforms/UnixProfiler.ts
@@ -26,7 +26,7 @@ export const CppProfilerName = `BAMPerfProfiler`;
 // into the Flipper plugin directory
 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
 // @ts-expect-error
-const binaryFolder = global.Flipper
+const binaryFolder = (global.Flipper || process.env.AWS)
   ? `${__dirname}/bin`
   : `${__dirname}/../../..${__dirname.includes("dist") ? "/.." : ""}/cpp-profiler/bin`;
 
