diff --git a/node_modules/@react-native-community/cli-config-android/build/config/findBuildGradle.js b/node_modules/@react-native-community/cli-config-android/build/config/findBuildGradle.js
index 5f2ebb0..a19b0e6 100644
--- a/node_modules/@react-native-community/cli-config-android/build/config/findBuildGradle.js
+++ b/node_modules/@react-native-community/cli-config-android/build/config/findBuildGradle.js
@@ -19,9 +19,9 @@ function _path() {
   return data;
 }
 function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
-function findBuildGradle(sourceDir, isLibrary) {
-  const buildGradlePath = _path().default.join(sourceDir, isLibrary ? 'build.gradle' : 'app/build.gradle');
-  const buildGradleKtsPath = _path().default.join(sourceDir, isLibrary ? 'build.gradle.kts' : 'app/build.gradle.kts');
+function findBuildGradle(sourceDir, appName) {
+  const buildGradlePath = _path().default.join(sourceDir, path.join(appName, 'build.gradle'));
+  const buildGradleKtsPath = _path().default.join(sourceDir, path.join(appName, 'build.gradle.kts'));
   if (_fs().default.existsSync(buildGradlePath)) {
     return buildGradlePath;
   } else if (_fs().default.existsSync(buildGradleKtsPath)) {
diff --git a/node_modules/@react-native-community/cli-config-android/build/config/index.js b/node_modules/@react-native-community/cli-config-android/build/config/index.js
index 8a86f08..a97c516 100644
--- a/node_modules/@react-native-community/cli-config-android/build/config/index.js
+++ b/node_modules/@react-native-community/cli-config-android/build/config/index.js
@@ -55,7 +55,7 @@ function projectConfig(root, userConfig = {}) {
   const sourceDir = _path().default.join(root, src);
   const appName = getAppName(sourceDir, userConfig.appName);
   const manifestPath = userConfig.manifestPath ? _path().default.join(sourceDir, userConfig.manifestPath) : (0, _findManifest.default)(_path().default.join(sourceDir, appName));
-  const buildGradlePath = (0, _findBuildGradle.findBuildGradle)(sourceDir, false);
+  const buildGradlePath = (0, _findBuildGradle.findBuildGradle)(sourceDir, appName);
   if (!manifestPath && !buildGradlePath) {
     return null;
   }
@@ -108,7 +108,7 @@ function dependencyConfig(root, userConfig = {}) {
   }
   const sourceDir = _path().default.join(root, src);
   const manifestPath = userConfig.manifestPath ? _path().default.join(sourceDir, userConfig.manifestPath) : (0, _findManifest.default)(sourceDir);
-  const buildGradlePath = (0, _findBuildGradle.findBuildGradle)(sourceDir, true);
+  const buildGradlePath = (0, _findBuildGradle.findBuildGradle)(sourceDir, '');
   const isPureCxxDependency = userConfig.cxxModuleCMakeListsModuleName != null && userConfig.cxxModuleCMakeListsPath != null && userConfig.cxxModuleHeaderName != null && !manifestPath && !buildGradlePath;
   if (!manifestPath && !buildGradlePath && !isPureCxxDependency) {
     return null;
