diff --git a/node_modules/@react-native-community/cli-config/build/findDependencies.js b/node_modules/@react-native-community/cli-config/build/findDependencies.js
index d59a1ba..3c0983c 100644
--- a/node_modules/@react-native-community/cli-config/build/findDependencies.js
+++ b/node_modules/@react-native-community/cli-config/build/findDependencies.js
@@ -24,8 +24,9 @@ function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { de
  */
 function findDependencies(root) {
   let pjson;
+  let pkg = process.env.IS_HYBRID_APP ? '../package.json' : 'package.json'
   try {
-    pjson = JSON.parse(_fs().default.readFileSync(_path().default.join(root, 'package.json'), 'utf8'));
+    pjson = JSON.parse(_fs().default.readFileSync(_path().default.join(root, pkg), 'utf8'));
   } catch (e) {
     return [];
   }
