diff --git a/node_modules/@react-native-community/cli-platform-ios/native_modules.rb b/node_modules/@react-native-community/cli-platform-ios/native_modules.rb
index 1b6eece..173e45b 100644
--- a/node_modules/@react-native-community/cli-platform-ios/native_modules.rb
+++ b/node_modules/@react-native-community/cli-platform-ios/native_modules.rb
@@ -12,7 +12,7 @@
 require 'pathname'
 require 'cocoapods'
 
-def use_native_modules!(config = nil)
+def updateConfig(config = nil)
   if (config.is_a? String)
     Pod::UI.warn("Passing custom root to use_native_modules! is deprecated.",
       [
@@ -36,8 +36,21 @@ def use_native_modules!(config = nil)
 
     config = JSON.parse(json.join("\n"))
   end
+end
+
+def use_native_modules!(config = nil)
+  if (ENV['REACT_NATIVE_DIR'])
+    Dir.chdir(ENV['REACT_NATIVE_DIR']) do
+      config = updateConfig(config)
+    end
+  else
+    config = updateConfig(config)
+  end
 
   project_root = Pathname.new(config["project"]["ios"]["sourceDir"])
+  if(ENV["PROJECT_ROOT_DIR"])
+    project_root = File.join(Dir.pwd, ENV["PROJECT_ROOT_DIR"])
+  end
 
   packages = config["dependencies"]
   found_pods = []
