diff --git a/node_modules/@react-native/gradle-plugin/settings-plugin/src/main/kotlin/com/facebook/react/ReactSettingsExtension.kt b/node_modules/@react-native/gradle-plugin/settings-plugin/src/main/kotlin/com/facebook/react/ReactSettingsExtension.kt
index 8f2f447..cf35d99 100644
--- a/node_modules/@react-native/gradle-plugin/settings-plugin/src/main/kotlin/com/facebook/react/ReactSettingsExtension.kt
+++ b/node_modules/@react-native/gradle-plugin/settings-plugin/src/main/kotlin/com/facebook/react/ReactSettingsExtension.kt
@@ -31,6 +31,17 @@ abstract class ReactSettingsExtension @Inject constructor(val settings: Settings
         it.toString()
       }
 
+  private var defaultWorkingDirectory: File = settings.layout.rootDirectory.dir("../").asFile
+
+  private var defaultLockFiles: FileCollection =
+    settings.layout.rootDirectory
+        .dir("../")
+        .files("yarn.lock", "package-lock.json", "package.json", "react-native.config.js")
+
+  public fun autolinkLibrariesFromCommandForPath(path: String) {
+    autolinkLibrariesFromCommand(defaultConfigCommand, settings.layout.rootDirectory.dir(path).asFile, defaultLockFiles)
+  }
+
   /**
    * Utility function to autolink libraries using an external command as source of truth.
    *
@@ -46,11 +57,8 @@ abstract class ReactSettingsExtension @Inject constructor(val settings: Settings
   @JvmOverloads
   public fun autolinkLibrariesFromCommand(
       command: List<String> = defaultConfigCommand,
-      workingDirectory: File? = settings.layout.rootDirectory.dir("../").asFile,
-      lockFiles: FileCollection =
-          settings.layout.rootDirectory
-              .dir("../")
-              .files("yarn.lock", "package-lock.json", "package.json", "react-native.config.js")
+      workingDirectory: File? = defaultWorkingDirectory,
+      lockFiles: FileCollection = defaultLockFiles
   ) {
     outputFile.parentFile.mkdirs()
     val lockFilesChanged = checkAndUpdateLockfiles(lockFiles, outputFolder)
