diff --git a/node_modules/@react-native-firebase/perf/android/build.gradle b/node_modules/@react-native-firebase/perf/android/build.gradle
index b4a9c7b..5835e3d 100644
--- a/node_modules/@react-native-firebase/perf/android/build.gradle
+++ b/node_modules/@react-native-firebase/perf/android/build.gradle
@@ -19,6 +19,7 @@ buildscript {
 
 plugins {
   id "io.invertase.gradle.build" version "1.5"
+  id 'com.android.library'
 }
 
 def appProject
@@ -129,4 +130,3 @@ dependencies {
 ReactNative.shared.applyPackageVersion()
 ReactNative.shared.applyDefaultExcludes()
 ReactNative.module.applyAndroidVersions()
-ReactNative.module.applyReactNativeDependency("api")
