diff --git a/node_modules/@react-native-async-storage/async-storage/RNCAsyncStorage.podspec b/node_modules/@react-native-async-storage/async-storage/RNCAsyncStorage.podspec
index 25c9420..dc685e8 100644
--- a/node_modules/@react-native-async-storage/async-storage/RNCAsyncStorage.podspec
+++ b/node_modules/@react-native-async-storage/async-storage/RNCAsyncStorage.podspec
@@ -23,7 +23,7 @@ Pod::Spec.new do |s|
       'HEADER_SEARCH_PATHS' => '"$(PODS_ROOT)/boost" "$(PODS_ROOT)/boost-for-react-native" "$(PODS_ROOT)/RCT-Folly"',
       'CLANG_CXX_LANGUAGE_STANDARD' => 'c++17',
     }
-    s.platforms       = { ios: '13.4', tvos: '11.0', :osx => "10.15" }
+    s.platforms       = { ios: '12.4', tvos: '11.0', :osx => "10.15" }
     s.compiler_flags  = folly_compiler_flags + ' -DRCT_NEW_ARCH_ENABLED'
 
     install_modules_dependencies(s)
