diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/internal/PreparePrefabHeadersTask.kt b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/internal/PreparePrefabHeadersTask.kt
index f30392d..ed87f85 100644
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/internal/PreparePrefabHeadersTask.kt
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/internal/PreparePrefabHeadersTask.kt
@@ -45,15 +45,9 @@ abstract class PreparePrefabHeadersTask : DefaultTask() {
         fs.copy {
           it.from(headerPath)
           it.include("**/*.h")
+          it.include("**/*.hpp")
           it.exclude("**/*.cpp")
           it.exclude("**/*.txt")
-          // We don't want to copy all the boost headers as they are 250Mb+
-          it.include("boost/config.hpp")
-          it.include("boost/config/**/*.hpp")
-          it.include("boost/core/*.hpp")
-          it.include("boost/detail/workaround.hpp")
-          it.include("boost/operators.hpp")
-          it.include("boost/preprocessor/**/*.hpp")
           // Headers needed for exposing rrc_text and rrc_textinput
           it.include("boost/container_hash/**/*.hpp")
           it.include("boost/detail/**/*.hpp")
