diff --git a/node_modules/@react-native-async-storage/async-storage/android/build.gradle b/node_modules/@react-native-async-storage/async-storage/android/build.gradle
index ff1baa4..32d8020 100644
--- a/node_modules/@react-native-async-storage/async-storage/android/build.gradle
+++ b/node_modules/@react-native-async-storage/async-storage/android/build.gradle
@@ -112,7 +112,7 @@ repositories {
 dependencies {
 
     if (useNextStorage) {
-        def room_version = getVersionOrDefault('AsyncStorage_next_roomVersion', '2.4.2')
+        def room_version = getVersionOrDefault('AsyncStorage_next_roomVersion', '2.4.3')
         def coroutines_version = "1.6.0"
         def coroutinesTest_version = "1.6.0"
         // if we don't provide explicit dependency on reflection, kotlin plugin