diff --git a/node_modules/@react-native-community/cli-config-android/build/config/findAndroidDir.js b/node_modules/@react-native-community/cli-config-android/build/config/findAndroidDir.js
index 075b10a..20b4695 100644
--- a/node_modules/@react-native-community/cli-config-android/build/config/findAndroidDir.js
+++ b/node_modules/@react-native-community/cli-config-android/build/config/findAndroidDir.js
@@ -28,7 +28,7 @@ function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { de
  */
 
 function findAndroidDir(root) {
-  if (_fs().default.existsSync(_path().default.join(root, 'android'))) {
+  if (_fs().default.existsSync(_path().default.join(root, 'Android'))) {
     return 'android';
   }
   return null;
