diff --git a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js
index 95e6871..5483c93 100644
--- a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js
+++ b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js
@@ -353,6 +353,9 @@ export default class CardStack extends React.Component {
           extrapolate: 'clamp'
         }) : STATE_TRANSITIONING_OR_BELOW_TOP;
       }
+      
+      const isHomeScreenAndNotOnTop = route.name === 'Home' && isScreenActive !== STATE_ON_TOP;
+
       const {
         headerShown = true,
         headerTransparent,
@@ -386,7 +389,7 @@ export default class CardStack extends React.Component {
         key: route.key,
         style: StyleSheet.absoluteFill,
         enabled: detachInactiveScreens,
-        active: isScreenActive,
+        active: isHomeScreenAndNotOnTop ? STATE_TRANSITIONING_OR_BELOW_TOP : isScreenActive,
         freezeOnBlur: freezeOnBlur,
         pointerEvents: "box-none"
       }, /*#__PURE__*/React.createElement(CardContainer, {
@@ -420,7 +423,7 @@ export default class CardStack extends React.Component {
         onTransitionStart: onTransitionStart,
         onTransitionEnd: onTransitionEnd,
         isNextScreenTransparent: isNextScreenTransparent,
-        detachCurrentScreen: detachCurrentScreen
+        detachCurrentScreen: isHomeScreenAndNotOnTop ? false : detachCurrentScreen
       }));
     })), isFloatHeaderAbsolute ? floatingHeader : null);
   }
