diff --git a/node_modules/@react-navigation/native/lib/module/findFocusedRouteKey.js b/node_modules/@react-navigation/native/lib/module/findFocusedRouteKey.js
new file mode 100644
index 0000000..b996822
--- /dev/null
+++ b/node_modules/@react-navigation/native/lib/module/findFocusedRouteKey.js
@@ -0,0 +1,8 @@
+"use strict";
+
+import { findFocusedRoute } from '@react-navigation/core';
+export function findFocusedRouteKey(state) {
+    const focusedRoute = findFocusedRoute(state);
+    return focusedRoute?.key;
+}
+//# sourceMappingURL=findFocusedRouteKey.js.map
