diff --git a/node_modules/@react-navigation/core/lib/module/useDescriptors.js b/node_modules/@react-navigation/core/lib/module/useDescriptors.js
index 76fdab1..75f315c 100644
--- a/node_modules/@react-navigation/core/lib/module/useDescriptors.js
+++ b/node_modules/@react-navigation/core/lib/module/useDescriptors.js
@@ -112,6 +112,19 @@ export default function useDescriptors(_ref, convertCustomScreenOptions) {
       }
       return o;
     });
+    const SidePane = customOptions.sidePane;
+    let element = /*#__PURE__*/React.createElement(React.Fragment, {
+      children: [/*#__PURE__*/React.createElement(SceneView, {
+        navigation: navigation,
+        route: route,
+        screen: screen,
+        routeState: state.routes[i].state,
+        getState: getState,
+        setState: setState,
+        options: customOptions,
+        clearOptions: clearOptions
+      }), SidePane && /*#__PURE__*/React.createElement(SidePane, {})]
+    });
     acc[route.key] = {
       route,
       // @ts-expect-error: it's missing action helpers, fix later
@@ -123,17 +136,10 @@ export default function useDescriptors(_ref, convertCustomScreenOptions) {
         }, /*#__PURE__*/React.createElement(NavigationContext.Provider, {
           value: navigation
         }, /*#__PURE__*/React.createElement(NavigationRouteContext.Provider, {
-          value: route
-        }, /*#__PURE__*/React.createElement(SceneView, {
-          navigation: navigation,
-          route: route,
-          screen: screen,
-          routeState: state.routes[i].state,
-          getState: getState,
-          setState: setState,
-          options: mergedOptions,
-          clearOptions: clearOptions
-        }))));
+          value: route,
+          children: element
+        }, 
+      )));
       },
       options: mergedOptions
     };
diff --git a/node_modules/@react-navigation/core/src/useDescriptors.tsx b/node_modules/@react-navigation/core/src/useDescriptors.tsx
index 2e4ee0f..11ece43 100644
--- a/node_modules/@react-navigation/core/src/useDescriptors.tsx
+++ b/node_modules/@react-navigation/core/src/useDescriptors.tsx
@@ -238,6 +238,23 @@ export default function useDescriptors<
         return o;
       });
 
+    const SidePane = (customOptions as any).sidePane;
+    let element = (
+        <>
+            <SceneView
+                navigation={navigation}
+                route={route}
+                screen={screen}
+                routeState={state.routes[i].state}
+                getState={getState}
+                setState={setState}
+                options={customOptions}
+                clearOptions={clearOptions}
+            />
+            {SidePane && <SidePane />}
+        </>
+    );
+
     acc[route.key] = {
       route,
       // @ts-expect-error: it's missing action helpers, fix later
@@ -247,16 +264,7 @@ export default function useDescriptors<
           <NavigationBuilderContext.Provider key={route.key} value={context}>
             <NavigationContext.Provider value={navigation}>
               <NavigationRouteContext.Provider value={route}>
-                <SceneView
-                  navigation={navigation}
-                  route={route}
-                  screen={screen}
-                  routeState={state.routes[i].state}
-                  getState={getState}
-                  setState={setState}
-                  options={mergedOptions}
-                  clearOptions={clearOptions}
-                />
+                {element}
               </NavigationRouteContext.Provider>
             </NavigationContext.Provider>
           </NavigationBuilderContext.Provider>
