diff --git a/node_modules/@react-native-firebase/crashlytics/android/build.gradle b/node_modules/@react-native-firebase/crashlytics/android/build.gradle
index 6b6de57..9b89ae7 100644
--- a/node_modules/@react-native-firebase/crashlytics/android/build.gradle
+++ b/node_modules/@react-native-firebase/crashlytics/android/build.gradle
@@ -18,6 +18,7 @@ buildscript {
 
 plugins {
   id "io.invertase.gradle.build" version "1.5"
+  id 'com.android.library'
 }
 
 def appProject
@@ -92,4 +93,3 @@ dependencies {
 ReactNative.shared.applyPackageVersion()
 ReactNative.shared.applyDefaultExcludes()
 ReactNative.module.applyAndroidVersions()
-ReactNative.module.applyReactNativeDependency("api")
