diff --git a/node_modules/@react-native-community/geolocation/android/build.gradle b/node_modules/@react-native-community/geolocation/android/build.gradle
index a489454..78ca1c0 100644
--- a/node_modules/@react-native-community/geolocation/android/build.gradle
+++ b/node_modules/@react-native-community/geolocation/android/build.gradle
@@ -73,7 +73,7 @@ repositories {
   } else {
     defaultDir = new File(
       projectDir,
-      '/../../../node_modules/react-native/android'
+      '/../../../../node_modules/react-native/android'
     )
   }
 
