diff --git a/node_modules/@react-native/virtualized-lists/Lists/VirtualizedList.js b/node_modules/@react-native/virtualized-lists/Lists/VirtualizedList.js
index 70b2e20..2b15e73 100644
--- a/node_modules/@react-native/virtualized-lists/Lists/VirtualizedList.js
+++ b/node_modules/@react-native/virtualized-lists/Lists/VirtualizedList.js
@@ -1708,6 +1708,10 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
       visibleLength,
       zoomScale,
     };
+    this._listMetrics.notifyListContentLayout({
+      layout: {width: e.nativeEvent.contentSize.width, height: e.nativeEvent.contentSize.height},
+      orientation: this._orientation(),
+    });
     if (this.state.pendingScrollUpdateCount > 0) {
       this.setState(state => ({
         pendingScrollUpdateCount: state.pendingScrollUpdateCount - 1,
