diff --git a/node_modules/@react-native-community/cli-tools/build/findProjectRoot.js b/node_modules/@react-native-community/cli-tools/build/findProjectRoot.js
index 3c7a0c8..15dd82e 100644
--- a/node_modules/@react-native-community/cli-tools/build/findProjectRoot.js
+++ b/node_modules/@react-native-community/cli-tools/build/findProjectRoot.js
@@ -24,7 +24,7 @@ function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { de
  * Finds project root by looking for a closest `package.json`.
  */
 function findProjectRoot(cwd = process.cwd()) {
-  const packageLocation = _findUp().default.sync('package.json', {
+  const packageLocation = _findUp().default.sync('react-native/package.json', {
     cwd
   });
   /**
