diff --git a/node_modules/@react-navigation/drawer/src/views/modern/Overlay.tsx b/node_modules/@react-navigation/drawer/src/views/modern/Overlay.tsx
index 6ce0d88..9b40e76 100644
--- a/node_modules/@react-navigation/drawer/src/views/modern/Overlay.tsx
+++ b/node_modules/@react-navigation/drawer/src/views/modern/Overlay.tsx
@@ -21,7 +21,7 @@ const Overlay = React.forwardRef(function Overlay(
       opacity: progress.value,
       // We don't want the user to be able to press through the overlay when drawer is open
       // We can send the overlay behind the screen to avoid it
-      zIndex: progress.value > PROGRESS_EPSILON ? 0 : -1,
+      // zIndex: progress.value > PROGRESS_EPSILON ? 0 : -1,
     };
   });
 