diff --git a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/images/RNMBXImagesManager.kt b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/images/RNMBXImagesManager.kt
index a9e2b68..1e08fdf 100644
--- a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/images/RNMBXImagesManager.kt
+++ b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/images/RNMBXImagesManager.kt
@@ -233,7 +233,7 @@ class RNMBXImagesManager(private val mContext: ReactApplicationContext) :
         parent.mImageViews.remove(view)
     }
 
-    override fun removeAllViews(parent: RNMBXImages) {
+    override fun removeAllViews(parent: RNMBXImages?) {
         if (parent == null) {
             Logger.e("RNMBXImages", "removeAllViews parent is null")
             return
diff --git a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/utils/ViewTagResolver.kt b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/utils/ViewTagResolver.kt
index f45cc25..5b0bb86 100644
--- a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/utils/ViewTagResolver.kt
+++ b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/utils/ViewTagResolver.kt
@@ -30,7 +30,7 @@ open class ViewTagResolver(val context: ReactApplicationContext) {
         if (list != null) {
             context.runOnUiQueueThread {
                 try {
-                    val view = manager.resolveView(viewTag)
+                    val view = manager.resolveView(viewTag)!!
 
                     list.forEach { it.fn(view) }
                 } catch (err: IllegalViewOperationException) {
