diff --git a/node_modules/@rnmapbox/maps/ios/RNMBX/RNMBXMarkerViewComponentView.mm b/node_modules/@rnmapbox/maps/ios/RNMBX/RNMBXMarkerViewComponentView.mm
index b7f2ddc..4853e3c 100644
--- a/node_modules/@rnmapbox/maps/ios/RNMBX/RNMBXMarkerViewComponentView.mm
+++ b/node_modules/@rnmapbox/maps/ios/RNMBX/RNMBXMarkerViewComponentView.mm
@@ -109,6 +109,7 @@ - (void)updateLayoutMetrics:(const LayoutMetrics &)layoutMetrics
         layoutMetrics.borderWidth,
         layoutMetrics.displayType,
         layoutMetrics.layoutDirection,
+        false,
         layoutMetrics.pointScaleFactor,
         layoutMetrics.overflowInset
     };
