diff --git a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/camera/RNMBXCamera.kt b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/camera/RNMBXCamera.kt
index bf149f9..2d3441b 100644
--- a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/camera/RNMBXCamera.kt
+++ b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/camera/RNMBXCamera.kt
@@ -190,7 +190,7 @@ class RNMBXCamera(private val mContext: Context, private val mManager: RNMBXCame
 
     private fun setInitialCamera() {
         mDefaultStop?.let {
-            val mapView = mMapView!!
+            val mapView = mMapView ?: return
             val map = mapView.getMapboxMap()
 
             it.setDuration(0)
diff --git a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/camera/RNMBXCameraModule.kt b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/camera/RNMBXCameraModule.kt
index 0c9f710..512092f 100644
--- a/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/camera/RNMBXCameraModule.kt
+++ b/node_modules/@rnmapbox/maps/android/src/main/java/com/rnmapbox/rnmbx/components/camera/RNMBXCameraModule.kt
@@ -49,4 +49,8 @@ class RNMBXCameraModule(context: ReactApplicationContext, val viewTagResolver: V
             promise.resolve(null)
         }
     }
+
+    public fun updateCameraStop(viewRef: Double?, stop: ReadableMap, promise: Promise) {
+        updateCameraStop(viewRef?.toInt(), stop, promise)
+    }
 }
\ No newline at end of file