diff --git a/node_modules/@ua/react-native-airship/android/src/main/java/com/urbanairship/reactnative/Utils.kt b/node_modules/@ua/react-native-airship/android/src/main/java/com/urbanairship/reactnative/Utils.kt
index d4ea6da..c57b7c2 100644
--- a/node_modules/@ua/react-native-airship/android/src/main/java/com/urbanairship/reactnative/Utils.kt
+++ b/node_modules/@ua/react-native-airship/android/src/main/java/com/urbanairship/reactnative/Utils.kt
@@ -54,7 +54,7 @@ object Utils {
     fun convertDynamic(obj: Dynamic?): JsonValue {
         return if (obj == null) {
             JsonValue.NULL
-        } else when (obj.type) {
+        } else when (obj.getType()) {
             ReadableType.Null -> JsonValue.NULL
             ReadableType.Boolean -> JsonValue.wrapOpt(obj.asBoolean())
             ReadableType.String -> JsonValue.wrapOpt(obj.asString())
