diff --git a/node_modules/@ua/react-native-airship/lib/typescript/types.d.ts b/node_modules/@ua/react-native-airship/lib/typescript/types.d.ts
index 9e21186..1e7f466 100644
--- a/node_modules/@ua/react-native-airship/lib/typescript/types.d.ts
+++ b/node_modules/@ua/react-native-airship/lib/typescript/types.d.ts
@@ -1,6 +1,6 @@
 export type JsonValue = string | number | boolean | null | JsonObject | JsonArray;
 export type JsonObject = {
-    [key: string]: JsonValue;
+    [key: string]: JsonValue | undefined;
 };
 export type JsonArray = JsonValue[];
 export interface ChannelCreatedEvent {
