diff --git a/node_modules/eslint-plugin-react-compiler/dist/index.js b/node_modules/eslint-plugin-react-compiler/dist/index.js
index a0f47a7..f649250 100644
--- a/node_modules/eslint-plugin-react-compiler/dist/index.js
+++ b/node_modules/eslint-plugin-react-compiler/dist/index.js
@@ -69108,7 +69108,7 @@ const rule = {
       return false;
     }
     let babelAST;
-    if (context.filename.endsWith(".tsx") || context.filename.endsWith(".ts")) {
+    if (filename.endsWith(".tsx") || filename.endsWith(".ts")) {
       try {
         const { parse: babelParse } = require("@babel/parser");
         babelAST = babelParse(sourceCode, {
