diff --git a/node_modules/expo/android/src/main/java/expo/modules/ExpoReactHostFactory.kt b/node_modules/expo/android/src/main/java/expo/modules/ExpoReactHostFactory.kt
index ed65d12..09bd068 100644
--- a/node_modules/expo/android/src/main/java/expo/modules/ExpoReactHostFactory.kt
+++ b/node_modules/expo/android/src/main/java/expo/modules/ExpoReactHostFactory.kt
@@ -92,7 +92,6 @@ object ExpoReactHostFactory {
     if (reactHost == null) {
       val useDeveloperSupport = reactNativeHost.useDeveloperSupport
       val reactHostDelegate = ExpoReactHostDelegate(WeakReference(context), reactNativeHost)
-      val reactJsExceptionHandler = ReactJsExceptionHandler { _ -> }
       val componentFactory = ComponentFactory()
       DefaultComponentsRegistry.register(componentFactory)
 
@@ -106,7 +105,6 @@ object ExpoReactHostFactory {
           reactHostDelegate,
           componentFactory,
           true,
-          reactJsExceptionHandler,
           useDeveloperSupport
         )
           .apply {
