\ No newline at end of file
diff --git a/node_modules/expo-image/android/build/tmp/kapt3/stubs/debug/expo/modules/image/thumbhash/ThumbhashModule.kapt_metadata b/node_modules/expo-image/android/build/tmp/kapt3/stubs/debug/expo/modules/image/thumbhash/ThumbhashModule.kapt_metadata
new file mode 100644
index 0000000..f469e9e
Binary files /dev/null and b/node_modules/expo-image/android/build/tmp/kapt3/stubs/debug/expo/modules/image/thumbhash/ThumbhashModule.kapt_metadata differ
diff --git a/node_modules/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt b/node_modules/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt
index 64619c6..3d94142 100644
--- a/node_modules/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt
+++ b/node_modules/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt
@@ -211,14 +211,14 @@ class ExpoImageView(
     super.onDraw(canvas)
     // Draw borders on top of the background and image
     if (borderDrawableLazyHolder.isInitialized()) {
-      val layoutDirection = if (I18nUtil.getInstance().isRTL(context)) {
+      val layoutDirection = if (I18nUtil.instance.isRTL(context)) {
         LAYOUT_DIRECTION_RTL
       } else {
         LAYOUT_DIRECTION_LTR
       }
 
       borderDrawable.apply {
-        resolvedLayoutDirection = layoutDirection
+        setLayoutDirectionOverride(layoutDirection)
         setBounds(0, 0, width, height)
         draw(canvas)
       }
diff --git a/node_modules/expo-image/android/src/main/java/expo/modules/image/ResourceIdHelper.kt b/node_modules/expo-image/android/src/main/java/expo/modules/image/ResourceIdHelper.kt
index aabe829..aac94db 100644
--- a/node_modules/expo-image/android/src/main/java/expo/modules/image/ResourceIdHelper.kt
+++ b/node_modules/expo-image/android/src/main/java/expo/modules/image/ResourceIdHelper.kt
@@ -32,7 +32,7 @@ object ResourceIdHelper {
   }
 
   fun getResourceUri(context: Context, name: String): Uri? {
-    val drawableUri = ResourceDrawableIdHelper.getInstance().getResourceDrawableUri(context, name)
+    val drawableUri = ResourceDrawableIdHelper.instance.getResourceDrawableUri(context, name)
     if (drawableUri != Uri.EMPTY) {
       return drawableUri
     }
diff --git a/node_modules/expo-image/android/src/main/java/expo/modules/image/drawing/OutlineProvider.kt b/node_modules/expo-image/android/src/main/java/expo/modules/image/drawing/OutlineProvider.kt
index 72b7289..5178d03 100644
--- a/node_modules/expo-image/android/src/main/java/expo/modules/image/drawing/OutlineProvider.kt
+++ b/node_modules/expo-image/android/src/main/java/expo/modules/image/drawing/OutlineProvider.kt
@@ -52,7 +52,7 @@ class OutlineProvider(private val mContext: Context) : ViewOutlineProvider() {
     }
 
     val isRTL = mLayoutDirection == View.LAYOUT_DIRECTION_RTL
-    val isRTLSwap = I18nUtil.getInstance().doLeftAndRightSwapInRTL(mContext)
+    val isRTLSwap = I18nUtil.instance.doLeftAndRightSwapInRTL(mContext)
     updateCornerRadius(
       CornerRadius.TOP_LEFT,
       BorderRadiusConfig.TOP_LEFT,
