diff --git a/node_modules/expo-image-manipulator/src/index.ts b/node_modules/expo-image-manipulator/src/index.ts
index 65e63fb..9613c39 100644
--- a/node_modules/expo-image-manipulator/src/index.ts
+++ b/node_modules/expo-image-manipulator/src/index.ts
@@ -1,9 +1,9 @@
 export { ImageManipulator, manipulateAsync, useImageManipulator } from './ImageManipulator';
 
-export { SaveOptions, SaveFormat, ImageResult } from './ImageManipulator.types';
+export type { SaveOptions, SaveFormat, ImageResult } from './ImageManipulator.types';
 
 // Export types that are deprecated as of SDK 52
-export {
+export type {
   ActionResize,
   ActionRotate,
   FlipType,
