diff --git a/node_modules/expo-modules-core/ios/AppDelegates/EXReactNativeFactoryDelegate.mm b/node_modules/expo-modules-core/ios/AppDelegates/EXReactNativeFactoryDelegate.mm
index fab8929..5cce027 100644
--- a/node_modules/expo-modules-core/ios/AppDelegates/EXReactNativeFactoryDelegate.mm
+++ b/node_modules/expo-modules-core/ios/AppDelegates/EXReactNativeFactoryDelegate.mm
@@ -6,7 +6,7 @@
 #import <React/CoreModulesPlugins.h>
 #endif
 
-#import <React-RCTAppDelegate/RCTAppSetupUtils.h>
+#import <React_RCTAppDelegate/RCTAppSetupUtils.h>
 #import <react/nativemodule/defaults/DefaultTurboModules.h>
 
 @implementation EXReactNativeFactoryDelegate
