diff --git a/node_modules/expo/src/winter/FormData.ts b/node_modules/expo/src/winter/FormData.ts
index a1cb797..7c4a14c 100644
--- a/node_modules/expo/src/winter/FormData.ts
+++ b/node_modules/expo/src/winter/FormData.ts
@@ -119,7 +119,7 @@ export function installFormDataPatch(formData: typeof FormData) {
   };
 
   // Required for RSC: https://github.com/facebook/react/blob/985747f81033833dca22f30b0c04704dd4bd3714/packages/react-server/src/ReactFlightServer.js#L2117
-  formData.prototype.entries = function* entries(
+  formData.prototype.entries ??= function* entries(
     this: ReactNativeFormDataInternal
   ): IterableIterator<[string, FormDataEntryValue]> {
     for (const part of this._parts) {
