diff --git a/node_modules/expo-modules-core/build/ts-declarations/global.d.ts b/node_modules/expo-modules-core/build/ts-declarations/global.d.ts
index 6589274..9e3e46c 100644
--- a/node_modules/expo-modules-core/build/ts-declarations/global.d.ts
+++ b/node_modules/expo-modules-core/build/ts-declarations/global.d.ts
@@ -59,7 +59,7 @@ export interface ExpoProcess {
         EXPO_BASE_URL?: string;
         /** Build-time representation of the `Platform.OS` value that the current JavaScript was bundled for. Does not support platform shaking wrapped require statements. */
         EXPO_OS?: string;
-        [key: string]: any;
+        [key: string]: string;
     };
     [key: string]: any;
 }
