diff --git a/node_modules/expo-av/android/build.gradle b/node_modules/expo-av/android/build.gradle
index 11e7574..6dae6a0 100644
--- a/node_modules/expo-av/android/build.gradle
+++ b/node_modules/expo-av/android/build.gradle
@@ -3,12 +3,13 @@ apply plugin: 'com.android.library'
 group = 'host.exp.exponent'
 version = '14.0.7'
 
+def REACT_NATIVE_PATH = this.hasProperty('reactNativeProject') ? this.reactNativeProject + '/node_modules/react-native/package.json' : 'react-native/package.json'
 def REACT_NATIVE_BUILD_FROM_SOURCE = findProject(":ReactAndroid") != null
 def REACT_NATIVE_DIR = REACT_NATIVE_BUILD_FROM_SOURCE
   ? findProject(":ReactAndroid").getProjectDir().parent
   : file(providers.exec {
       workingDir(rootDir)
-      commandLine("node", "--print", "require.resolve('react-native/package.json')")
+      commandLine("node", "--print", "require.resolve('${REACT_NATIVE_PATH}')")
     }.standardOutput.asText.get().trim()).parent
 
 def reactNativeArchitectures() {
