diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/ModuleRegistryReadyNotifier.java b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/ModuleRegistryReadyNotifier.java
index c873ec0..98628e6 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/ModuleRegistryReadyNotifier.java
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/ModuleRegistryReadyNotifier.java
@@ -20,7 +20,7 @@ public class ModuleRegistryReadyNotifier extends BaseJavaModule {
 
   @Override
   public String getName() {
-    return null;
+    return "ModuleRegistryReadyNotifier";
   }
 
   @Override
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt
index 7f1b5e6..bc127ff 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt
@@ -170,7 +170,7 @@ class AppContext(
             this,
             jsRuntimePointer,
             jniDeallocator,
-            reactContext.runtimeExecutor!!
+            reactContext.catalystInstance.runtimeExecutor!!
           )
         } else {
           jsiInterop.installJSI(
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/defaultmodules/CoreModule.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/defaultmodules/CoreModule.kt
index 71725db..62da68a 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/defaultmodules/CoreModule.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/defaultmodules/CoreModule.kt
@@ -4,7 +4,7 @@ import com.facebook.react.ReactActivity
 import com.facebook.react.ReactDelegate
 import com.facebook.react.bridge.UiThreadUtil
 import com.facebook.react.config.ReactFeatureFlags
-import com.facebook.react.devsupport.DisabledDevSupportManager
+import com.facebook.react.devsupport.ReleaseDevSupportManager
 import expo.modules.kotlin.events.normalizeEventName
 import expo.modules.kotlin.modules.Module
 import expo.modules.kotlin.modules.ModuleDefinition
@@ -70,7 +70,7 @@ class CoreModule : Module() {
         ?: return@AsyncFunction
       if (!ReactFeatureFlags.enableBridgelessArchitecture) {
         val reactInstanceManager = reactDelegate.reactInstanceManager
-        if (reactInstanceManager.devSupportManager is DisabledDevSupportManager) {
+        if (reactInstanceManager.devSupportManager is ReleaseDevSupportManager) {
           UiThreadUtil.runOnUiThread {
             reactInstanceManager.recreateReactContextInBackground()
           }
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/FilteredReadableMap.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/FilteredReadableMap.kt
index 0ff2a51..a6ff1f8 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/FilteredReadableMap.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/FilteredReadableMap.kt
@@ -41,8 +41,8 @@ class FilteredReadableMap(
   private val backingMap: ReadableMap,
   private val filteredKeys: List<String>
 ) : ReadableMap by backingMap {
-  override fun getEntryIterator(): Iterator<Map.Entry<String, Any>> =
-    FilteredIterator(backingMap.entryIterator) {
+  override val entryIterator: Iterator<Map.Entry<String, Any>>
+    get() = FilteredIterator(backingMap.entryIterator) {
       !filteredKeys.contains(it.key)
     }
 
