diff --git a/node_modules/expo-modules-autolinking/scripts/android/autolinking_implementation.gradle b/node_modules/expo-modules-autolinking/scripts/android/autolinking_implementation.gradle
index f085818..fcb9594 100644
--- a/node_modules/expo-modules-autolinking/scripts/android/autolinking_implementation.gradle
+++ b/node_modules/expo-modules-autolinking/scripts/android/autolinking_implementation.gradle
@@ -152,12 +152,13 @@ class ExpoAutolinkingManager {
   }

   static private String[] convertOptionsToCommandArgs(String command, Map options) {
+    def expoPath = options.searchPaths ?  "../react-native/node_modules/expo" : "expo"
     String[] args = [
       'node',
       '--no-warnings',
       '--eval',
       // Resolve the `expo` > `expo-modules-autolinking` chain from the project root
-      'require(require.resolve(\'expo-modules-autolinking\', { paths: [require.resolve(\'expo\')] }))(process.argv.slice(1))',
+      "require(require.resolve(\'expo-modules-autolinking\', { paths: [require.resolve(\'${expoPath}\')] }))(process.argv.slice(1))",
       '--',
       command,
       '--platform',
diff --git a/node_modules/expo-modules-autolinking/scripts/ios/project_integrator.rb b/node_modules/expo-modules-autolinking/scripts/ios/project_integrator.rb
index 5d46f1e..fec4f34 100644
--- a/node_modules/expo-modules-autolinking/scripts/ios/project_integrator.rb
+++ b/node_modules/expo-modules-autolinking/scripts/ios/project_integrator.rb
@@ -215,6 +215,7 @@ module Expo
       args = autolinking_manager.base_command_args.map { |arg| "\"#{arg}\"" }
       platform = autolinking_manager.platform_name.downcase
       package_names = autolinking_manager.packages_to_generate.map { |package| "\"#{package.name}\"" }
+      expo_path = ENV['REACT_NATIVE_DIR'] ? "#{ENV['REACT_NATIVE_DIR']}/node_modules/expo" : "expo"

       <<~SUPPORT_SCRIPT
       #!/usr/bin/env bash
@@ -262,7 +263,7 @@ module Expo

       with_node \\
         --no-warnings \\
-        --eval "require(require.resolve(\'expo-modules-autolinking\', { paths: [require.resolve(\'expo/package.json\')] }))(process.argv.slice(1))" \\
+        --eval "require(require.resolve(\'expo-modules-autolinking\', { paths: [require.resolve(\'#{expo_path}/package.json\')] }))(process.argv.slice(1))" \\
         generate-modules-provider #{args.join(' ')} \\
         --target "#{modules_provider_path}" \\
         --platform "apple" \\
