diff --git a/node_modules/http-server/lib/http-server.js b/node_modules/http-server/lib/http-server.js
index dfe4c47..b1c3089 100644
--- a/node_modules/http-server/lib/http-server.js
+++ b/node_modules/http-server/lib/http-server.js
@@ -98,6 +98,8 @@ function HttpServer(options) {
     });
   }
 
+  this.headers['Document-Policy'] = 'js-profiling';
+
   if (options.cors) {
     this.headers['Access-Control-Allow-Origin'] = '*';
     this.headers['Access-Control-Allow-Headers'] = 'Origin, X-Requested-With, Content-Type, Accept, Range';
