diff --git a/node_modules/jest-expo/src/preset/setup.js b/node_modules/jest-expo/src/preset/setup.js
index 7a77203..d905ebe 100644
--- a/node_modules/jest-expo/src/preset/setup.js
+++ b/node_modules/jest-expo/src/preset/setup.js
@@ -293,5 +293,8 @@ try {
 // Installs web implementations of global things that are normally installed through JSI.
 require('expo-modules-core/src/web/index.web');
 
+jest.doMock('expo/src/winter/FormData', () => ({
+  installFormDataPatch: jest.fn(),
+}));
 // Ensure the environment globals are installed before the first test runs.
 require('expo/src/winter');
