diff --git a/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift b/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift
index e955eae..dcecae5 100644
--- a/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift
+++ b/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift
@@ -251,6 +251,12 @@ class ContainerView: RCTView {
         }
     }
     
+    override var frame: CGRect {
+        didSet {
+            animationView?.reactSetFrame(self.bounds)
+        }
+    }
+    
     // MARK: Private
     func replaceAnimationView(next: LottieAnimationView) {
         animationView?.removeFromSuperview()
