diff --git a/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgPackage.java b/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgPackage.java
index ce96152..f75cab5 100644
--- a/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgPackage.java
+++ b/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgPackage.java
@@ -22,7 +22,6 @@ import com.facebook.react.module.annotations.ReactModule;
 import com.facebook.react.module.annotations.ReactModuleList;
 import com.facebook.react.module.model.ReactModuleInfo;
 import com.facebook.react.module.model.ReactModuleInfoProvider;
-import com.facebook.react.turbomodule.core.interfaces.TurboModule;
 import com.facebook.react.uimanager.ViewManager;
 import java.util.ArrayList;
 import java.util.Collections;
@@ -241,7 +240,7 @@ public class SvgPackage extends TurboReactPackage implements ViewManagerOnDemand
   /** {@inheritDoc} */
   @Override
   public List<String> getViewManagerNames(ReactApplicationContext reactContext) {
-    return (List<String>) getViewManagersMap(reactContext).keySet();
+    return new ArrayList<>(getViewManagersMap(reactContext).keySet());
   }
 
   @Override
@@ -300,7 +299,7 @@ public class SvgPackage extends TurboReactPackage implements ViewManagerOnDemand
                     reactModule.needsEagerInit(),
                     reactModule.hasConstants(),
                     reactModule.isCxxModule(),
-                    TurboModule.class.isAssignableFrom(moduleClass)));
+                    true));
           }
 
           return reactModuleInfoMap;
diff --git a/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt b/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt
index 38799a1..4296879 100644
--- a/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt
+++ b/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt
@@ -49,6 +49,7 @@ target_link_libraries(
   react_render_graphics
   react_render_mapbuffer
   rrc_image
+  react_render_componentregistry
   rrc_view
   turbomodulejsijni
   yoga
@@ -62,7 +63,7 @@ target_compile_options(
   -DLOG_TAG=\"ReactNative\"
   -fexceptions
   -frtti
-  -std=c++17
+  -std=c++20
   -Wall
 )
 message(WARNING "CMAKE_CURRENT_SORUCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}")
diff --git a/node_modules/react-native-svg/src/fabric/NativeSvgRenderableModule.ts b/node_modules/react-native-svg/src/fabric/NativeSvgRenderableModule.ts
index b6a130e..0bffc94 100644
--- a/node_modules/react-native-svg/src/fabric/NativeSvgRenderableModule.ts
+++ b/node_modules/react-native-svg/src/fabric/NativeSvgRenderableModule.ts
@@ -2,7 +2,7 @@
 // its needed for codegen to work
 import type { TurboModule } from 'react-native';
 import { TurboModuleRegistry } from 'react-native';
-import type { Int32, Float } from 'react-native/Libraries/Types/CodegenTypes';
+import type { Double, Float } from 'react-native/Libraries/Types/CodegenTypes';
 
 type Rect = {
   x: Float;
@@ -26,13 +26,13 @@ type Point = {
 };
 
 export interface Spec extends TurboModule {
-  isPointInFill(tag: Int32 | null, options?: Object): boolean;
-  isPointInStroke(tag: Int32 | null, options?: Object): boolean;
-  getTotalLength(tag: Int32 | null): Float;
-  getPointAtLength(tag: Int32 | null, options?: Object): Point;
-  getBBox(tag: Int32 | null, options?: Object): Rect;
-  getCTM(tag: Int32 | null): Matrix;
-  getScreenCTM(tag: Int32 | null): Matrix;
+  isPointInFill(tag: Double | null, options?: Object): boolean;
+  isPointInStroke(tag: Double | null, options?: Object): boolean;
+  getTotalLength(tag: Double | null): Float;
+  getPointAtLength(tag: Double | null, options?: Object): Point;
+  getBBox(tag: Double | null, options?: Object): Rect;
+  getCTM(tag: Double | null): Matrix;
+  getScreenCTM(tag: Double | null): Matrix;
   getRawResource(name: string): Promise<string>;
 }
 
diff --git a/node_modules/react-native-svg/src/fabric/NativeSvgViewModule.ts b/node_modules/react-native-svg/src/fabric/NativeSvgViewModule.ts
index 0a57737..29a4419 100644
--- a/node_modules/react-native-svg/src/fabric/NativeSvgViewModule.ts
+++ b/node_modules/react-native-svg/src/fabric/NativeSvgViewModule.ts
@@ -2,11 +2,11 @@
 // its needed for codegen to work
 import type { TurboModule } from 'react-native';
 import { TurboModuleRegistry } from 'react-native';
-import type { Int32 } from 'react-native/Libraries/Types/CodegenTypes';
+import type { Double } from 'react-native/Libraries/Types/CodegenTypes';
 
 export interface Spec extends TurboModule {
   toDataURL(
-    tag: Int32 | null,
+    tag: Double | null,
     options?: Object,
     callback?: (base64: string) => void
   ): void;
