diff --git a/node_modules/react-native-performance/android/build.gradle b/node_modules/react-native-performance/android/build.gradle
index a6bbf7a..dd8a29c 100644
--- a/node_modules/react-native-performance/android/build.gradle
+++ b/node_modules/react-native-performance/android/build.gradle
@@ -100,19 +100,5 @@ repositories {
 }
 
 dependencies {
-    if (isNewArchitectureEnabled()) {
-        implementation project(":ReactAndroid")
-    } else {
-        implementation 'com.facebook.react:react-native:+'
-    }
-}
-
-if (isNewArchitectureEnabled()) {
-    react {
-        libraryName = "rnperformance"
-        codegenJavaPackageName = "com.oblador.performance"
-        reactNativeDir = rootProject.file("${nodeModules}/react-native")
-        jsRootDir = file("../src/")
-        codegenDir = rootProject.file("${nodeModules}/react-native-codegen")
-    }
+    implementation 'com.facebook.react:react-native:+'
 }
diff --git a/node_modules/react-native-performance/react-native-performance.podspec b/node_modules/react-native-performance/react-native-performance.podspec
index 48bc1ed..116e104 100644
--- a/node_modules/react-native-performance/react-native-performance.podspec
+++ b/node_modules/react-native-performance/react-native-performance.podspec
@@ -26,10 +26,6 @@ Pod::Spec.new do |s|
           "CLANG_CXX_LANGUAGE_STANDARD" => "c++17"
       }
 
-      s.dependency "React-Codegen"
-      s.dependency "RCT-Folly"
-      s.dependency "RCTRequired"
-      s.dependency "RCTTypeSafety"
-      s.dependency "ReactCommon/turbomodule/core"
+      install_modules_dependencies(s)
   end
 end
