diff --git a/node_modules/react-native-reanimated/lib/module/layoutReanimation/web/animationsManager.js b/node_modules/react-native-reanimated/lib/module/layoutReanimation/web/animationsManager.js
index fd6b3f5..886081c 100644
--- a/node_modules/react-native-reanimated/lib/module/layoutReanimation/web/animationsManager.js
+++ b/node_modules/react-native-reanimated/lib/module/layoutReanimation/web/animationsManager.js
@@ -102,9 +102,11 @@ export function tryActivateLayoutTransition(props, element, snapshot) {
   }
   const enteringAnimation = props.layout.enteringV?.presetName;
   const exitingAnimation = props.layout.exitingV?.presetName;
+  const deltaX = (snapshot.width - rect.width) / 2;
+  const deltaY = (snapshot.height - rect.height) / 2;
   const transitionData = {
-    translateX: snapshot.x - rect.x,
-    translateY: snapshot.y - rect.y,
+    translateX: snapshot.x - rect.x + deltaX,
+    translateY: snapshot.y - rect.y + deltaY,
     scaleX: snapshot.width / rect.width,
     scaleY: snapshot.height / rect.height,
     reversed: false,
