diff --git a/node_modules/react-native-keyboard-controller/src/components/KeyboardAvoidingView/index.tsx b/node_modules/react-native-keyboard-controller/src/components/KeyboardAvoidingView/index.tsx
index 9e73caf..0d5f143 100644
--- a/node_modules/react-native-keyboard-controller/src/components/KeyboardAvoidingView/index.tsx
+++ b/node_modules/react-native-keyboard-controller/src/components/KeyboardAvoidingView/index.tsx
@@ -81,7 +81,7 @@ const KeyboardAvoidingView = forwardRef<View, React.PropsWithChildren<Props>>(
     const onLayoutWorklet = useCallback((layout: LayoutRectangle) => {
       "worklet";
 
-      if (keyboard.isClosed.value) {
+      if (keyboard.isClosed.value || initialFrame.value === null) {
         initialFrame.value = layout;
       }
     }, []);
