diff --git a/node_modules/react-native/scripts/cocoapods/new_architecture.rb b/node_modules/react-native/scripts/cocoapods/new_architecture.rb
index ba75b01..e5eaee8 100644
--- a/node_modules/react-native/scripts/cocoapods/new_architecture.rb
+++ b/node_modules/react-native/scripts/cocoapods/new_architecture.rb
@@ -103,6 +103,7 @@ class NewArchitectureHelper
         compiler_flags = hash["compiler_flags"] ? hash["compiler_flags"] : ""
         current_config = hash["pod_target_xcconfig"] != nil ? hash["pod_target_xcconfig"] : {}
         current_headers = current_config["HEADER_SEARCH_PATHS"] != nil ? current_config["HEADER_SEARCH_PATHS"] : ""
+        current_cpp_flags = current_config["OTHER_CPLUSPLUSFLAGS"] != nil ? current_config["OTHER_CPLUSPLUSFLAGS"] : ""
 
         header_search_paths = ["\"$(PODS_ROOT)/boost\" \"$(PODS_ROOT)/Headers/Private/Yoga\""]
         if ENV['USE_FRAMEWORKS']
@@ -135,7 +136,7 @@ class NewArchitectureHelper
         spec.dependency "glog"
 
         if new_arch_enabled
-            current_config["OTHER_CPLUSPLUSFLAGS"] = @@new_arch_cpp_flags
+            current_config["OTHER_CPLUSPLUSFLAGS"] = current_cpp_flags.empty? ? @@new_arch_cpp_flags : "#{current_cpp_flags} #{@@new_arch_cpp_flags}"
             spec.dependency "React-RCTFabric" # This is for Fabric Component
             spec.dependency "React-Codegen"
 
