diff --git a/node_modules/react-native-reanimated/lib/module/reanimated2/layoutReanimation/web/componentUtils.js b/node_modules/react-native-reanimated/lib/module/reanimated2/layoutReanimation/web/componentUtils.js
index 1b73e15..09e9238 100644
--- a/node_modules/react-native-reanimated/lib/module/reanimated2/layoutReanimation/web/componentUtils.js
+++ b/node_modules/react-native-reanimated/lib/module/reanimated2/layoutReanimation/web/componentUtils.js
@@ -178,7 +178,8 @@ export function handleExitingAnimation(element, animationConfig) {
   dummy.reanimatedDummy = true;
   element.style.animationName = '';
   // We hide current element so only its copy with proper animation will be displayed
-  element.style.visibility = 'hidden';
+  // Temp fix for React StrictMode
+  // element.style.visibility = 'hidden';
 
   // After cloning the element, we want to move all children from original element to its clone. This is because original element
   // will be unmounted, therefore when this code executes in child component, parent will be either empty or removed soon.
