diff --git a/node_modules/react-native/ReactCommon/react/renderer/components/scrollview/ScrollViewShadowNode.cpp b/node_modules/react-native/ReactCommon/react/renderer/components/scrollview/ScrollViewShadowNode.cpp
index a8ecce5..6ad790e 100644
--- a/node_modules/react-native/ReactCommon/react/renderer/components/scrollview/ScrollViewShadowNode.cpp
+++ b/node_modules/react-native/ReactCommon/react/renderer/components/scrollview/ScrollViewShadowNode.cpp
@@ -66,7 +66,17 @@ void ScrollViewShadowNode::layout(LayoutContext layoutContext) {
 Point ScrollViewShadowNode::getContentOriginOffset() const {
   auto stateData = getStateData();
   auto contentOffset = stateData.contentOffset;
-  return {-contentOffset.x, -contentOffset.y + stateData.scrollAwayPaddingTop};
+  auto props = getConcreteProps();
+
+  float productX = 1.0f;
+  float productY = 1.0f;
+
+  for (const auto& operation : props.transform.operations) {
+    productX *= operation.x;
+    productY *= operation.y;
+  }
+
+  return {-contentOffset.x * productX, (-contentOffset.y + stateData.scrollAwayPaddingTop) * productY};
 }
 
 } // namespace facebook::react
