diff --git a/node_modules/react-native/Libraries/Components/StatusBar/StatusBar.js b/node_modules/react-native/Libraries/Components/StatusBar/StatusBar.js
index 2ac3d1f..5a400bf 100644
--- a/node_modules/react-native/Libraries/Components/StatusBar/StatusBar.js
+++ b/node_modules/react-native/Libraries/Components/StatusBar/StatusBar.js
@@ -479,8 +479,7 @@ class StatusBar extends React.Component<Props> {
         }
         // Activities are not translucent by default, so always set if true.
         if (
-          !oldProps ||
-          oldProps.translucent !== mergedProps.translucent ||
+          (oldProps && oldProps.translucent !== mergedProps.translucent) ||
           mergedProps.translucent
         ) {
           NativeStatusBarManagerAndroid.setTranslucent(mergedProps.translucent);
