diff --git a/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PlaidModule.kt b/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PlaidModule.kt
index bd8b2f4..e1d65ca 100644
--- a/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PlaidModule.kt
+++ b/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PlaidModule.kt
@@ -86,7 +86,7 @@ class PlaidModule internal constructor(reactContext: ReactApplicationContext) :
   }
 
   @ReactMethod
-  override fun create(
+  override fun create2(
     token: String,
     noLoadingState: Boolean,
     logLevel: String,
diff --git a/node_modules/react-native-plaid-link-sdk/src/fabric/NativePlaidLinkModuleAndroid.ts b/node_modules/react-native-plaid-link-sdk/src/fabric/NativePlaidLinkModuleAndroid.ts
index 1f912b9..18cb350 100644
--- a/node_modules/react-native-plaid-link-sdk/src/fabric/NativePlaidLinkModuleAndroid.ts
+++ b/node_modules/react-native-plaid-link-sdk/src/fabric/NativePlaidLinkModuleAndroid.ts
@@ -6,7 +6,7 @@ import {UnsafeObject} from './fabricUtils';
 import {LinkSuccess, LinkExit} from '../Types';
 
 export interface Spec extends TurboModule {
-    create(token: string, noLoadingState: boolean, logLevel: string): void;
+    create2(token: string, noLoadingState: boolean, logLevel: string): void;
     open(
         onSuccess: (result: UnsafeObject<LinkSuccess>) => void,
         onExit: (result: UnsafeObject<LinkExit>) => void,
diff --git a/node_modules/react-native-plaid-link-sdk/ios/RNLinksdk.mm b/node_modules/react-native-plaid-link-sdk/ios/RNLinksdk.mm
index 8a82562..72fc423 100644
--- a/node_modules/react-native-plaid-link-sdk/ios/RNLinksdk.mm
+++ b/node_modules/react-native-plaid-link-sdk/ios/RNLinksdk.mm
@@ -67,7 +67,7 @@ RCT_EXPORT_MODULE();
     self.hasObservers = NO;
 }
 
-RCT_EXPORT_METHOD(create:(NSString*)token noLoadingState:(BOOL)noLoadingState) {
+RCT_EXPORT_METHOD(create2:(NSString*)token noLoadingState:(BOOL)noLoadingState) {
     __weak RNLinksdk *weakSelf = self;
 
     void (^onSuccess)(PLKLinkSuccess *) = ^(PLKLinkSuccess *success) {
diff --git a/node_modules/react-native-plaid-link-sdk/src/fabric/NativePlaidLinkModuleiOS.ts b/node_modules/react-native-plaid-link-sdk/src/fabric/NativePlaidLinkModuleiOS.ts
index 44e3a52..6d08da9 100644
--- a/node_modules/react-native-plaid-link-sdk/src/fabric/NativePlaidLinkModuleiOS.ts
+++ b/node_modules/react-native-plaid-link-sdk/src/fabric/NativePlaidLinkModuleiOS.ts
@@ -6,7 +6,7 @@ import {UnsafeObject} from './fabricUtils';
 import {LinkSuccess, LinkExit, LinkError} from '../Types';
 
 export interface Spec extends TurboModule {
-    create(token: string, noLoadingState: boolean): void;
+    create2(token: string, noLoadingState: boolean): void;
     open(
         fullScreen: boolean,
         onSuccess: (result: UnsafeObject<LinkSuccess>) => void,
