diff --git a/node_modules/react-native-web/dist/exports/Modal/index.js b/node_modules/react-native-web/dist/exports/Modal/index.js
index a9a7c36..522ef93 100644
--- a/node_modules/react-native-web/dist/exports/Modal/index.js
+++ b/node_modules/react-native-web/dist/exports/Modal/index.js
@@ -88,13 +88,11 @@ var Modal = /*#__PURE__*/React.forwardRef((props, forwardedRef) => {
     onDismiss: onDismissCallback,
     onShow: onShowCallback,
     visible: visible
-  }, /*#__PURE__*/React.createElement(ModalFocusTrap, {
-    active: isActive
   }, /*#__PURE__*/React.createElement(ModalContent, _extends({}, rest, {
     active: isActive,
     onRequestClose: onRequestClose,
     ref: forwardedRef,
     transparent: transparent
-  }), children))));
+  }), children)));
 });
 export default Modal;
\ No newline at end of file
