diff --git a/node_modules/react-native-haptic-feedback/ios/RNHapticFeedback/RNHapticFeedback.h b/node_modules/react-native-haptic-feedback/ios/RNHapticFeedback/RNHapticFeedback.h
index c1498b9..250df1f 100644
--- a/node_modules/react-native-haptic-feedback/ios/RNHapticFeedback/RNHapticFeedback.h
+++ b/node_modules/react-native-haptic-feedback/ios/RNHapticFeedback/RNHapticFeedback.h
@@ -1,5 +1,5 @@
 #ifdef RCT_NEW_ARCH_ENABLED
-#import "RNHapticFeedbackSpec.h"
+#import <RNHapticFeedbackSpec/RNHapticFeedbackSpec.h>
 
 @interface RNHapticFeedback : NSObject <NativeHapticFeedbackSpec>
 #else
diff --git a/node_modules/react-native-haptic-feedback/ios/RNHapticFeedback/RNHapticFeedbackSpec.h b/node_modules/react-native-haptic-feedback/ios/RNHapticFeedback/RNHapticFeedbackSpec.h
deleted file mode 100644
index 6f0f81d..0000000
--- a/node_modules/react-native-haptic-feedback/ios/RNHapticFeedback/RNHapticFeedbackSpec.h
+++ /dev/null
@@ -1,15 +0,0 @@
-//
-//  RNHapticFeedbackSpec.h
-//  RNHapticFeedback
-//
-//  Created by Michael Kuczera on 05.08.24.
-//  Copyright © 2024 Facebook. All rights reserved.
-//
-#import <Foundation/Foundation.h>
-
-@protocol NativeHapticFeedbackSpec <NSObject>
-
-// Indicates whether the device supports haptic feedback
-- (Boolean)supportsHaptic;
-
-@end
diff --git a/node_modules/react-native-haptic-feedback/package.json b/node_modules/react-native-haptic-feedback/package.json
index 86dfaa4..9cec8e4 100644
--- a/node_modules/react-native-haptic-feedback/package.json
+++ b/node_modules/react-native-haptic-feedback/package.json
@@ -6,18 +6,7 @@
   "source": "src/index.ts",
   "main": "./lib/commonjs/index.js",
   "module": "./lib/module/index.js",
-  "exports": {
-    ".": {
-      "import": {
-        "types": "./lib/typescript/module/src/index.d.ts",
-        "default": "./lib/module/index.js"
-      },
-      "require": {
-        "types": "./lib/typescript/commonjs/src/index.d.ts",
-        "default": "./lib/commonjs/index.js"
-      }
-    }
-  },
+  "types": "./lib/typescript/module/src/index.d.ts",
   "scripts": {
     "typecheck": "tsc --noEmit --project tsconfig.test.json",
     "test": "jest",
