diff --git a/node_modules/react-native-reanimated/lib/module/platform-specific/RNRenderer.web.js b/node_modules/react-native-reanimated/lib/module/platform-specific/RNRenderer.web.js
index a9b2299..6a97aea 100644
--- a/node_modules/react-native-reanimated/lib/module/platform-specific/RNRenderer.web.js
+++ b/node_modules/react-native-reanimated/lib/module/platform-specific/RNRenderer.web.js
@@ -1,5 +1,6 @@
 'use strict';
 
 // RNRender is not used for web. An export is still defined to eliminate warnings from bundlers such as esbuild.
-export {};
+const RNRenderer = {};
+export { RNRenderer };
 //# sourceMappingURL=RNRenderer.web.js.map
\ No newline at end of file
diff --git a/node_modules/react-native-reanimated/src/platform-specific/RNRenderer.web.ts b/node_modules/react-native-reanimated/src/platform-specific/RNRenderer.web.ts
index f9eb5e3..b51ae1d 100644
--- a/node_modules/react-native-reanimated/src/platform-specific/RNRenderer.web.ts
+++ b/node_modules/react-native-reanimated/src/platform-specific/RNRenderer.web.ts
@@ -1,3 +1,4 @@
 'use strict';
 // RNRender is not used for web. An export is still defined to eliminate warnings from bundlers such as esbuild.
-export {};
+const RNRenderer = {};
+export { RNRenderer };
\ No newline at end of file
