diff --git a/node_modules/react-native-reanimated/Common/cpp/Fabric/ShadowTreeCloner.cpp b/node_modules/react-native-reanimated/Common/cpp/Fabric/ShadowTreeCloner.cpp
index f913ceb..3f58247 100644
--- a/node_modules/react-native-reanimated/Common/cpp/Fabric/ShadowTreeCloner.cpp
+++ b/node_modules/react-native-reanimated/Common/cpp/Fabric/ShadowTreeCloner.cpp
@@ -51,6 +51,7 @@ ShadowNode::Unshared cloneShadowTreeWithNewProps(
     newChildNode = parentNode.clone({
         ShadowNodeFragment::propsPlaceholder(),
         std::make_shared<ShadowNode::ListOfShared>(children),
+        parentNode.getState()
     });
   }
 
