diff --git a/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx b/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
index 7d2448b..f8205ea 100644
--- a/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
+++ b/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
@@ -11,6 +11,7 @@ import type {
   PagerProps,
   Route,
 } from './types';
+import type { NativeProps } from 'react-native-pager-view/lib/typescript/specs/PagerViewNativeComponent';
 import { useAnimatedValue } from './useAnimatedValue';
 
 const AnimatedViewPager = Animated.createAnimatedComponent(ViewPager);
@@ -18,6 +19,7 @@ const AnimatedViewPager = Animated.createAnimatedComponent(ViewPager);
 type Props<T extends Route> = PagerProps & {
   onIndexChange: (index: number) => void;
   navigationState: NavigationState<T>;
+  onPageSelected?: NativeProps['onPageSelected'];
   children: (
     props: EventEmitterProps & {
       // Animated value which represents the state of current index
@@ -43,6 +45,7 @@ export function PagerViewAdapter<T extends Route>({
   children,
   style,
   animationEnabled,
+  onPageSelected,
   ...rest
 }: Props<T>) {
   const { index } = navigationState;
@@ -166,6 +169,7 @@ export function PagerViewAdapter<T extends Route>({
           position.setValue(index);
           setForceRender((fr) => fr+1)
           onIndexChange(index);
+          onPageSelected?.(e);
         }}
         onPageScrollStateChanged={onPageScrollStateChanged}
         scrollEnabled={swipeEnabled}
diff --git a/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx b/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx
index dbe7ed6..1600e98 100644
--- a/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx
+++ b/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx
@@ -18,12 +18,14 @@ import type {
   PagerProps,
   Route,
 } from './types';
+import type { NativeProps } from 'react-native-pager-view/lib/typescript/specs/PagerViewNativeComponent';
 import { useAnimatedValue } from './useAnimatedValue';
 
 type Props<T extends Route> = PagerProps & {
   layout: Layout;
   onIndexChange: (index: number) => void;
   navigationState: NavigationState<T>;
+  onPageSelected?: NativeProps['onPageSelected'];
   children: (
     props: EventEmitterProps & {
       // Animated value which represents the state of current index
diff --git a/node_modules/react-native-tab-view/src/TabView.tsx b/node_modules/react-native-tab-view/src/TabView.tsx
index bb1f531..1c57b14 100644
--- a/node_modules/react-native-tab-view/src/TabView.tsx
+++ b/node_modules/react-native-tab-view/src/TabView.tsx
@@ -17,6 +17,7 @@ import type {
   Route,
   SceneRendererProps,
 } from './types';
+import type { NativeProps } from 'react-native-pager-view/lib/typescript/specs/PagerViewNativeComponent';
 
 export type Props<T extends Route> = PagerProps & {
   onIndexChange: (index: number) => void;
@@ -33,6 +34,7 @@ export type Props<T extends Route> = PagerProps & {
   sceneContainerStyle?: StyleProp<ViewStyle>;
   pagerStyle?: StyleProp<ViewStyle>;
   style?: StyleProp<ViewStyle>;
+  onPageSelected?: NativeProps['onPageSelected'];
 };
 
 export function TabView<T extends Route>({
@@ -54,6 +56,7 @@ export function TabView<T extends Route>({
   tabBarPosition = 'top',
   animationEnabled = true,
   overScrollMode,
+  onPageSelected,
 }: Props<T>) {
   const [layout, setLayout] = React.useState({
     width: 0,
@@ -92,6 +95,7 @@ export function TabView<T extends Route>({
         animationEnabled={animationEnabled}
         overScrollMode={overScrollMode}
         style={pagerStyle}
+        onPageSelected={onPageSelected}
       >
         {({ position, render, addEnterListener, jumpTo }) => {
           // All of the props here must not change between re-renders
