diff --git a/node_modules/react-native-keyboard-controller/src/components/KeyboardAvoidingView/hooks.ts b/node_modules/react-native-keyboard-controller/src/components/KeyboardAvoidingView/hooks.ts
index 676eafc..6d84beb 100644
--- a/node_modules/react-native-keyboard-controller/src/components/KeyboardAvoidingView/hooks.ts
+++ b/node_modules/react-native-keyboard-controller/src/components/KeyboardAvoidingView/hooks.ts
@@ -33,6 +33,10 @@ export const useKeyboardAnimation = () => {
 
         isClosed.value = e.height === 0;
 
+        if (e.height > 0) {
+          heightWhenOpened.value = e.height;
+        }
+
         // `height` update happens in `onMove` handler
         // in `onEnd` we need to update only if `onMove`
         // wasn't called (i. e. duration === 0)
