diff --git a/node_modules/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitMarker.cpp b/node_modules/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitMarker.cpp
index 3404e89..b545cb6 100644
--- a/node_modules/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitMarker.cpp
+++ b/node_modules/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitMarker.cpp
@@ -9,7 +9,7 @@ namespace reanimated {
 thread_local bool ReanimatedCommitMarker::reanimatedCommitFlag_{false};
 
 ReanimatedCommitMarker::ReanimatedCommitMarker() {
-  react_native_assert(reanimatedCommitFlag_ != true);
+  // react_native_assert(reanimatedCommitFlag_ != true);
   reanimatedCommitFlag_ = true;
 }

