diff --git a/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java b/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
index 840b5f5..a0afd83 100644
--- a/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
+++ b/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
@@ -357,6 +357,8 @@ public class ReactEditText extends AppCompatEditText {
             ContentResolver cr = getReactContext(this).getContentResolver();
             type = cr.getType(itemUri);
             data = itemUri.toString();
+            mPasteWatcher.onPaste(type, data);
+            return true;
           }
         }
         if (type != null && data != null) {
