diff --git a/node_modules/react-native-reanimated/src/reanimated2/initializers.ts b/node_modules/react-native-reanimated/src/reanimated2/initializers.ts
index 2c82793..74e36b6 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/initializers.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/initializers.ts
@@ -153,7 +153,7 @@ export function initializeUIRuntime() {
     };
   }
 
-  const capturableConsole = console;
+  const capturableConsole = { ...console };
   runOnUIImmediately(() => {
     'worklet';
     // setup error handler
