diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultNewArchitectureEntryPoint.kt b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultNewArchitectureEntryPoint.kt
index 38b3be0..edaee98 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultNewArchitectureEntryPoint.kt
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultNewArchitectureEntryPoint.kt
@@ -51,6 +51,8 @@ public object DefaultNewArchitectureEntryPoint {
             // similar bugs related to pressable.
             override fun enableEventEmitterRetentionDuringGesturesOnAndroid(): Boolean =
                 fabricEnabled
+
+            override fun allowRecursiveCommitsWithSynchronousMountOnAndroid(): Boolean = true
           })
     }
 
