diff --git a/node_modules/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitMarker.cpp b/node_modules/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitMarker.cpp
index 3404e89..b545cb6 100644
--- a/node_modules/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitMarker.cpp
+++ b/node_modules/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitMarker.cpp
@@ -9,7 +9,7 @@ namespace reanimated {
 thread_local bool ReanimatedCommitMarker::reanimatedCommitFlag_{false};
 
 ReanimatedCommitMarker::ReanimatedCommitMarker() {
-  react_native_assert(reanimatedCommitFlag_ != true);
+  // react_native_assert(reanimatedCommitFlag_ != true);
   reanimatedCommitFlag_ = true;
 }
 
diff --git a/node_modules/react-native-reanimated/lib/module/reanimated2/UpdateProps.js b/node_modules/react-native-reanimated/lib/module/reanimated2/UpdateProps.js
index e69c581..78b7034 100644
--- a/node_modules/react-native-reanimated/lib/module/reanimated2/UpdateProps.js
+++ b/node_modules/react-native-reanimated/lib/module/reanimated2/UpdateProps.js
@@ -7,14 +7,11 @@ import { isFabric, isJest, shouldBeUseWeb } from './PlatformChecker';
 import { runOnUIImmediately } from './threads';
 let updateProps;
 if (shouldBeUseWeb()) {
-  updateProps = (_, updates, maybeViewRef, isAnimatedProps) => {
+  updateProps = (viewDescriptorsSet, updates, maybeViewRef, isAnimatedProps) => {
     'worklet';
-
-    if (maybeViewRef) {
-      maybeViewRef.items.forEach((item, _index) => {
-        _updatePropsJS(updates, item, isAnimatedProps);
-      });
-    }
+    viewDescriptorsSet.value.forEach((viewDescriptor) => {
+      _updatePropsJS(updates, {_component: viewDescriptor.tag}, isAnimatedProps);
+    })
   };
 } else {
   updateProps = (viewDescriptors, updates) => {
