diff --git a/node_modules/react-native-tab-view/lib/commonjs/PagerViewAdapter.js b/node_modules/react-native-tab-view/lib/commonjs/PagerViewAdapter.js
index f439cfe..5fc4765 100644
--- a/node_modules/react-native-tab-view/lib/commonjs/PagerViewAdapter.js
+++ b/node_modules/react-native-tab-view/lib/commonjs/PagerViewAdapter.js
@@ -19,6 +19,7 @@ function PagerViewAdapter(_ref) {
     swipeEnabled = true,
     navigationState,
     onIndexChange,
+    onTabSelect,
     onSwipeStart,
     onSwipeEnd,
     children,
@@ -120,6 +121,7 @@ function PagerViewAdapter(_ref) {
         const index = e.nativeEvent.position;
         indexRef.current = index;
         onIndexChange(index);
+        onTabSelect === null || onTabSelect === void 0 || onTabSelect(index);
       },
       onPageScrollStateChanged: onPageScrollStateChanged,
       scrollEnabled: swipeEnabled
diff --git a/node_modules/react-native-tab-view/lib/commonjs/PanResponderAdapter.js b/node_modules/react-native-tab-view/lib/commonjs/PanResponderAdapter.js
index f483cb6..4cbaad9 100644
--- a/node_modules/react-native-tab-view/lib/commonjs/PanResponderAdapter.js
+++ b/node_modules/react-native-tab-view/lib/commonjs/PanResponderAdapter.js
@@ -25,6 +25,7 @@ function PanResponderAdapter(_ref) {
     swipeEnabled = true,
     navigationState,
     onIndexChange,
+    onTabSelect,
     onSwipeStart,
     onSwipeEnd,
     children,
@@ -40,10 +41,17 @@ function PanResponderAdapter(_ref) {
   const navigationStateRef = React.useRef(navigationState);
   const layoutRef = React.useRef(layout);
   const onIndexChangeRef = React.useRef(onIndexChange);
+  const onTabSelectRef = React.useRef(onTabSelect);
   const currentIndexRef = React.useRef(index);
   const pendingIndexRef = React.useRef();
   const swipeVelocityThreshold = 0.15;
   const swipeDistanceThreshold = layout.width / 1.75;
+  const onTabSelectHandler = React.useCallback((tabIndex = index) => {
+    _reactNative.InteractionManager.runAfterInteractions(() => {
+      var _onTabSelectRef$curre;
+      (_onTabSelectRef$curre = onTabSelectRef.current) === null || _onTabSelectRef$curre === void 0 || _onTabSelectRef$curre.call(onTabSelectRef, tabIndex);
+    });
+  }, []);
   const jumpToIndex = React.useCallback(function (index) {
     let animate = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : animationEnabled;
     const offset = -index * layoutRef.current.width;
@@ -62,6 +70,7 @@ function PanResponderAdapter(_ref) {
         } = _ref2;
         if (finished) {
           onIndexChangeRef.current(index);
+          onTabSelectHandler(index);
           pendingIndexRef.current = undefined;
         }
       });
@@ -69,6 +78,7 @@ function PanResponderAdapter(_ref) {
     } else {
       panX.setValue(offset);
       onIndexChangeRef.current(index);
+      onTabSelectHandler(index);
       pendingIndexRef.current = undefined;
     }
   }, [animationEnabled, panX]);
@@ -76,11 +86,13 @@ function PanResponderAdapter(_ref) {
     navigationStateRef.current = navigationState;
     layoutRef.current = layout;
     onIndexChangeRef.current = onIndexChange;
+    onTabSelectRef.current = onTabSelect;
   });
   React.useEffect(() => {
     const offset = -navigationStateRef.current.index * layout.width;
     panX.setValue(offset);
-  }, [layout.width, panX]);
+    onTabSelectHandler();
+  }, [layout.width, panX, onTabSelectHandler]);
   React.useEffect(() => {
     if (keyboardDismissMode === 'auto') {
       _reactNative.Keyboard.dismiss();
diff --git a/node_modules/react-native-tab-view/lib/commonjs/TabView.js b/node_modules/react-native-tab-view/lib/commonjs/TabView.js
index 571c9c6..6206602 100644
--- a/node_modules/react-native-tab-view/lib/commonjs/TabView.js
+++ b/node_modules/react-native-tab-view/lib/commonjs/TabView.js
@@ -15,6 +15,7 @@ function _extends() { _extends = Object.assign ? Object.assign.bind() : function
 function TabView(_ref) {
   let {
     onIndexChange,
+    onTabSelect,
     navigationState,
     renderScene,
     initialLayout,
@@ -69,6 +70,7 @@ function TabView(_ref) {
     onSwipeStart: onSwipeStart,
     onSwipeEnd: onSwipeEnd,
     onIndexChange: jumpToIndex,
+    onTabSelect: onTabSelect,
     animationEnabled: animationEnabled,
     overScrollMode: overScrollMode,
     style: pagerStyle
diff --git a/node_modules/react-native-tab-view/lib/module/PagerViewAdapter.js b/node_modules/react-native-tab-view/lib/module/PagerViewAdapter.js
index d09a316..17c6f70 100644
--- a/node_modules/react-native-tab-view/lib/module/PagerViewAdapter.js
+++ b/node_modules/react-native-tab-view/lib/module/PagerViewAdapter.js
@@ -10,6 +10,7 @@ export function PagerViewAdapter(_ref) {
     swipeEnabled = true,
     navigationState,
     onIndexChange,
+    onTabSelect,
     onSwipeStart,
     onSwipeEnd,
     children,
@@ -111,6 +112,7 @@ export function PagerViewAdapter(_ref) {
         const index = e.nativeEvent.position;
         indexRef.current = index;
         onIndexChange(index);
+        onTabSelect === null || onTabSelect === void 0 || onTabSelect(index);
       },
       onPageScrollStateChanged: onPageScrollStateChanged,
       scrollEnabled: swipeEnabled
diff --git a/node_modules/react-native-tab-view/lib/module/PanResponderAdapter.js b/node_modules/react-native-tab-view/lib/module/PanResponderAdapter.js
index 7caafd7..0523bd0 100644
--- a/node_modules/react-native-tab-view/lib/module/PanResponderAdapter.js
+++ b/node_modules/react-native-tab-view/lib/module/PanResponderAdapter.js
@@ -1,6 +1,6 @@
 function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
 import * as React from 'react';
-import { Animated, I18nManager, Keyboard, PanResponder, StyleSheet, View } from 'react-native';
+import { Animated, I18nManager, InteractionManager, Keyboard, PanResponder, StyleSheet, View } from 'react-native';
 import { useAnimatedValue } from './useAnimatedValue';
 const DEAD_ZONE = 12;
 const DefaultTransitionSpec = {
@@ -17,6 +17,7 @@ export function PanResponderAdapter(_ref) {
     swipeEnabled = true,
     navigationState,
     onIndexChange,
+    onTabSelect,
     onSwipeStart,
     onSwipeEnd,
     children,
@@ -32,10 +33,17 @@ export function PanResponderAdapter(_ref) {
   const navigationStateRef = React.useRef(navigationState);
   const layoutRef = React.useRef(layout);
   const onIndexChangeRef = React.useRef(onIndexChange);
+  const onTabSelectRef = React.useRef(onTabSelect);
   const currentIndexRef = React.useRef(index);
   const pendingIndexRef = React.useRef();
   const swipeVelocityThreshold = 0.15;
   const swipeDistanceThreshold = layout.width / 1.75;
+  const onTabSelectHandler = React.useCallback((tabIndex = index) => {
+    InteractionManager.runAfterInteractions(() => {
+      var _onTabSelectRef$curre;
+      (_onTabSelectRef$curre = onTabSelectRef.current) === null || _onTabSelectRef$curre === void 0 || _onTabSelectRef$curre.call(onTabSelectRef, tabIndex);
+    });
+  }, []);
   const jumpToIndex = React.useCallback(function (index) {
     let animate = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : animationEnabled;
     const offset = -index * layoutRef.current.width;
@@ -54,6 +62,7 @@ export function PanResponderAdapter(_ref) {
         } = _ref2;
         if (finished) {
           onIndexChangeRef.current(index);
+          onTabSelectHandler(index);
           pendingIndexRef.current = undefined;
         }
       });
@@ -61,6 +70,7 @@ export function PanResponderAdapter(_ref) {
     } else {
       panX.setValue(offset);
       onIndexChangeRef.current(index);
+      onTabSelectHandler(index);
       pendingIndexRef.current = undefined;
     }
   }, [animationEnabled, panX]);
@@ -68,11 +78,13 @@ export function PanResponderAdapter(_ref) {
     navigationStateRef.current = navigationState;
     layoutRef.current = layout;
     onIndexChangeRef.current = onIndexChange;
+    onTabSelectRef.current = onTabSelect;
   });
   React.useEffect(() => {
     const offset = -navigationStateRef.current.index * layout.width;
     panX.setValue(offset);
-  }, [layout.width, panX]);
+    onTabSelectHandler();
+  }, [layout.width, panX, onTabSelectHandler]);
   React.useEffect(() => {
     if (keyboardDismissMode === 'auto') {
       Keyboard.dismiss();
diff --git a/node_modules/react-native-tab-view/lib/module/TabView.js b/node_modules/react-native-tab-view/lib/module/TabView.js
index ba28170..adfd3e3 100644
--- a/node_modules/react-native-tab-view/lib/module/TabView.js
+++ b/node_modules/react-native-tab-view/lib/module/TabView.js
@@ -7,6 +7,7 @@ import { TabBar } from './TabBar';
 export function TabView(_ref) {
   let {
     onIndexChange,
+    onTabSelect,
     navigationState,
     renderScene,
     initialLayout,
@@ -61,6 +62,7 @@ export function TabView(_ref) {
     onSwipeStart: onSwipeStart,
     onSwipeEnd: onSwipeEnd,
     onIndexChange: jumpToIndex,
+    onTabSelect: onTabSelect,
     animationEnabled: animationEnabled,
     overScrollMode: overScrollMode,
     style: pagerStyle
diff --git a/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx b/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
index 7d2448b..17531cb 100644
--- a/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
+++ b/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
@@ -17,6 +17,7 @@ const AnimatedViewPager = Animated.createAnimatedComponent(ViewPager);
 
 type Props<T extends Route> = PagerProps & {
   onIndexChange: (index: number) => void;
+  onTabSelect?: (index: number) => void;
   navigationState: NavigationState<T>;
   children: (
     props: EventEmitterProps & {
@@ -38,6 +39,7 @@ export function PagerViewAdapter<T extends Route>({
   swipeEnabled = true,
   navigationState,
   onIndexChange,
+  onTabSelect,
   onSwipeStart,
   onSwipeEnd,
   children,
@@ -166,6 +168,7 @@ export function PagerViewAdapter<T extends Route>({
           position.setValue(index);
           setForceRender((fr) => fr+1)
           onIndexChange(index);
+          onTabSelect?.(index);
         }}
         onPageScrollStateChanged={onPageScrollStateChanged}
         scrollEnabled={swipeEnabled}
diff --git a/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx b/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx
index dbe7ed6..2afa90b 100644
--- a/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx
+++ b/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx
@@ -3,6 +3,7 @@ import {
   Animated,
   GestureResponderEvent,
   I18nManager,
+  InteractionManager,
   Keyboard,
   PanResponder,
   PanResponderGestureState,
@@ -23,6 +24,7 @@ import { useAnimatedValue } from './useAnimatedValue';
 type Props<T extends Route> = PagerProps & {
   layout: Layout;
   onIndexChange: (index: number) => void;
+  onTabSelect?: (index: number) => void;
   navigationState: NavigationState<T>;
   children: (
     props: EventEmitterProps & {
@@ -55,6 +57,7 @@ export function PanResponderAdapter<T extends Route>({
   swipeEnabled = true,
   navigationState,
   onIndexChange,
+  onTabSelect,
   onSwipeStart,
   onSwipeEnd,
   children,
@@ -70,6 +73,7 @@ export function PanResponderAdapter<T extends Route>({
   const navigationStateRef = React.useRef(navigationState);
   const layoutRef = React.useRef(layout);
   const onIndexChangeRef = React.useRef(onIndexChange);
+  const onTabSelectRef = React.useRef(onTabSelect);
 
   const currentIndexRef = React.useRef(index);
   const pendingIndexRef = React.useRef<number>();
@@ -77,6 +81,12 @@ export function PanResponderAdapter<T extends Route>({
   const swipeVelocityThreshold = 0.15;
   const swipeDistanceThreshold = layout.width / 1.75;
 
+  const onTabSelectHandler = React.useCallback((tabIndex = index) => {
+    InteractionManager.runAfterInteractions(() => {
+      onTabSelectRef.current?.(tabIndex);
+    });
+  }, []);
+
   const jumpToIndex = React.useCallback(
     (index: number, animate = animationEnabled) => {
       const offset = -index * layoutRef.current.width;
@@ -93,6 +103,7 @@ export function PanResponderAdapter<T extends Route>({
         ]).start(({ finished }) => {
           if (finished) {
             onIndexChangeRef.current(index);
+            onTabSelectHandler(index);
             pendingIndexRef.current = undefined;
           }
         });
@@ -100,6 +111,7 @@ export function PanResponderAdapter<T extends Route>({
       } else {
         panX.setValue(offset);
         onIndexChangeRef.current(index);
+        onTabSelectHandler(index);
         pendingIndexRef.current = undefined;
       }
     },
@@ -110,13 +122,15 @@ export function PanResponderAdapter<T extends Route>({
     navigationStateRef.current = navigationState;
     layoutRef.current = layout;
     onIndexChangeRef.current = onIndexChange;
+    onTabSelectRef.current = onTabSelect;
   });
 
   React.useEffect(() => {
     const offset = -navigationStateRef.current.index * layout.width;
 
     panX.setValue(offset);
-  }, [layout.width, panX]);
+    onTabSelectHandler();
+  }, [layout.width, panX, onTabSelectHandler]);
 
   React.useEffect(() => {
     if (keyboardDismissMode === 'auto') {
diff --git a/node_modules/react-native-tab-view/src/TabView.tsx b/node_modules/react-native-tab-view/src/TabView.tsx
index bb1f531..3752ea9 100644
--- a/node_modules/react-native-tab-view/src/TabView.tsx
+++ b/node_modules/react-native-tab-view/src/TabView.tsx
@@ -20,6 +20,7 @@ import type {
 
 export type Props<T extends Route> = PagerProps & {
   onIndexChange: (index: number) => void;
+  onTabSelect?: (index: number) => void;
   navigationState: NavigationState<T>;
   renderScene: (props: SceneRendererProps & { route: T }) => React.ReactNode;
   renderLazyPlaceholder?: (props: { route: T }) => React.ReactNode;
@@ -37,6 +38,7 @@ export type Props<T extends Route> = PagerProps & {
 
 export function TabView<T extends Route>({
   onIndexChange,
+  onTabSelect,
   navigationState,
   renderScene,
   initialLayout,
@@ -89,6 +91,7 @@ export function TabView<T extends Route>({
         onSwipeStart={onSwipeStart}
         onSwipeEnd={onSwipeEnd}
         onIndexChange={jumpToIndex}
+        onTabSelect={onTabSelect}
         animationEnabled={animationEnabled}
         overScrollMode={overScrollMode}
         style={pagerStyle}
