diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java
index b767513..8947486 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java
@@ -143,6 +143,7 @@ public class FabricEventDispatcher implements EventDispatcher, LifecycleEventLis
   @Override
   public void onHostResume() {
     scheduleDispatchOfBatchedEvents();
+    mCurrentFrameCallback.resume();
   }
 
   @Override
@@ -212,6 +213,10 @@ public class FabricEventDispatcher implements EventDispatcher, LifecycleEventLis
       }
     }
 
+    public void resume() {
+      mShouldStop = false;
+    }
+
     public void stop() {
       mShouldStop = true;
     }
