diff --git a/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.mm b/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.mm
index 6d159ef..e254f50 100644
--- a/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.mm
+++ b/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.mm
@@ -163,7 +163,7 @@ - (BOOL)fabricEnabled
 
 - (BOOL)bridgelessEnabled
 {
-  return [self newArchEnabled];
+  return NO;
 }
 
 - (NSURL *)bundleURL