diff --git a/node_modules/react-native-pdf/fabric/RNPDFPdfNativeComponent.js b/node_modules/react-native-pdf/fabric/RNPDFPdfNativeComponent.js
index 596d796..4e47061 100644
--- a/node_modules/react-native-pdf/fabric/RNPDFPdfNativeComponent.js
+++ b/node_modules/react-native-pdf/fabric/RNPDFPdfNativeComponent.js
@@ -22,6 +22,7 @@
    enablePaging: ?boolean,
    enableRTL: ?boolean,
    enableAnnotationRendering: ?boolean,
+   enableDoubleTapZoom: ?boolean,
    showsHorizontalScrollIndicator: ?boolean,
    showsVerticalScrollIndicator: ?boolean,
    enableAntialiasing: ?boolean,
