diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
index df62f49..b134908 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
@@ -1084,12 +1084,18 @@ public class ReactEditText extends AppCompatEditText {
   public void onAttachedToWindow() {
     super.onAttachedToWindow();
 
+    int selectionStart = getSelectionStart();
+    int selectionEnd = getSelectionEnd();
+
     // Used to ensure that text is selectable inside of removeClippedSubviews
     // See https://github.com/facebook/react-native/issues/6805 for original
     // fix that was ported to here.
 
     super.setTextIsSelectable(true);
 
+    // Restore the selection since `setTextIsSelectable` changed it.
+    setSelection(selectionStart, selectionEnd);
+
     if (mContainsImages) {
       Spanned text = getText();
       TextInlineImageSpan[] spans = text.getSpans(0, text.length(), TextInlineImageSpan.class);
