diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/MaintainVisibleScrollPositionHelper.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/MaintainVisibleScrollPositionHelper.java
index b2cff8b..097c1d3 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/MaintainVisibleScrollPositionHelper.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/MaintainVisibleScrollPositionHelper.java
@@ -203,7 +203,12 @@ class MaintainVisibleScrollPositionHelper<ScrollViewT extends ViewGroup & HasSmo
     }
     Rect frame = new Rect();
     firstVisibleView.getHitRect(frame);
-    mPrevFirstVisibleFrame = frame;
+
+    if (frame.width() > 0 || frame.height() > 0) {
+      mPrevFirstVisibleFrame = frame;
+    } else {
+      mPrevFirstVisibleFrame = null;
+    }
   }
 
   // UIManagerListener
