diff --git a/node_modules/react-native/ReactAndroid/src/main/jni/react/fabric/Binding.h b/node_modules/react-native/ReactAndroid/src/main/jni/react/fabric/Binding.h
index d4266bc..af8aeeb 100644
--- a/node_modules/react-native/ReactAndroid/src/main/jni/react/fabric/Binding.h
+++ b/node_modules/react-native/ReactAndroid/src/main/jni/react/fabric/Binding.h
@@ -34,7 +34,12 @@ class ReactNativeConfig;
 class Scheduler;
 class SurfaceHandlerBinding;
 
-class Binding : public jni::HybridClass<Binding>,
+struct JBinding : public jni::JavaClass<JBinding> {
+    constexpr static auto kJavaDescriptor =
+            "Lcom/facebook/react/fabric/Binding;";
+};
+
+class Binding : public jni::HybridClass<Binding, JBinding>,
                 public SchedulerDelegate,
                 public LayoutAnimationStatusDelegate {
  public:
diff --git a/node_modules/react-native/ReactAndroid/src/main/jni/react/fabric/JFabricUIManager.cpp b/node_modules/react-native/ReactAndroid/src/main/jni/react/fabric/JFabricUIManager.cpp
index a2ccdbf..289ac58 100644
--- a/node_modules/react-native/ReactAndroid/src/main/jni/react/fabric/JFabricUIManager.cpp
+++ b/node_modules/react-native/ReactAndroid/src/main/jni/react/fabric/JFabricUIManager.cpp
@@ -13,8 +13,8 @@ namespace facebook::react {
 
 Binding* JFabricUIManager::getBinding() {
   static const auto bindingField =
-      javaClassStatic()->getField<Binding::javaobject>("mBinding");
+          javaClassStatic()->getField<JBinding::javaobject>("mBinding");
 
-  return getFieldValue(bindingField)->cthis();
+    return jni::static_ref_cast<Binding::javaobject>(getFieldValue(bindingField))->cthis();
 }
 } // namespace facebook::react
