diff --git a/node_modules/react-native-svg/apple/Elements/RNSVGImage.mm b/node_modules/react-native-svg/apple/Elements/RNSVGImage.mm
index 62b961f..4898760 100644
--- a/node_modules/react-native-svg/apple/Elements/RNSVGImage.mm
+++ b/node_modules/react-native-svg/apple/Elements/RNSVGImage.mm
@@ -151,6 +151,23 @@ - (void)didReceiveImage:(UIImage *)image metadata:(id)metadata fromObserver:(voi
   });
 }
 
+- (void)didReceiveFailure:(nonnull NSError *)error fromObserver:(nonnull const void *)observer
+{
+  if (_image) {
+    CGImageRelease(_image);
+  }
+  _image = nil;
+}
+
+- (void)didReceiveProgress:(float)progress
+                    loaded:(int64_t)loaded
+                     total:(int64_t)total
+              fromObserver:(nonnull const void *)observer
+{
+}
+
+#pragma mark - RCTImageResponseDelegate - < RN 0.75
+
 - (void)didReceiveProgress:(float)progress fromObserver:(void const *)observer
 {
 }
@@ -183,6 +200,7 @@ - (void)prepareForRecycle
   _imageSize = CGSizeZero;
   _reloadImageCancellationBlock = nil;
 }
+
 #endif // RCT_NEW_ARCH_ENABLED
 
 - (void)setSrc:(RCTImageSource *)src
@@ -218,7 +236,7 @@ - (void)setSrc:(RCTImageSource *)src
 #if TARGET_OS_OSX // [macOS]
                            sourceLoaded = [src imageSourceWithSize:image.size scale:1];
 #else
-                            sourceLoaded = [src imageSourceWithSize:image.size scale:image.scale];
+                sourceLoaded = [src imageSourceWithSize:image.size scale:image.scale];
 #endif
                            NSDictionary *dict = @{
                              @"uri" : sourceLoaded.request.URL.absoluteString,
