diff --git a/node_modules/react-native-reanimated/mock.js b/node_modules/react-native-reanimated/mock.js
index 68b20d2..b088001 100644
--- a/node_modules/react-native-reanimated/mock.js
+++ b/node_modules/react-native-reanimated/mock.js
@@ -41,6 +41,9 @@ const Reanimated = {
   createAnimatedComponent: (Component) => Component,
   addWhitelistedUIProps: NOOP,
   addWhitelistedNativeProps: NOOP,
+
+  // used by react-navigation fork
+  isConfigured: () => true,
 };
 
 module.exports = {
