diff --git a/node_modules/react-native/scripts/codegen/generate-artifacts-executor.js b/node_modules/react-native/scripts/codegen/generate-artifacts-executor.js
index 025f80c..d276c69 100644
--- a/node_modules/react-native/scripts/codegen/generate-artifacts-executor.js
+++ b/node_modules/react-native/scripts/codegen/generate-artifacts-executor.js
@@ -454,7 +454,7 @@ function findCodegenEnabledLibraries(
   codegenConfigFilename,
   codegenConfigKey,
 ) {
-  const pkgJson = readPackageJSON(appRootDir);
+  const pkgJson = readPackageJSON(path.join(appRootDir, process.env.REACT_NATIVE_DIR ? 'react-native' : ''));
   const dependencies = {...pkgJson.dependencies, ...pkgJson.devDependencies};
   const libraries = [];
 
