diff --git a/node_modules/react-native-reanimated/lib/module/reanimated2/index.js b/node_modules/react-native-reanimated/lib/module/reanimated2/index.js
index 0027008..2aea2ff 100644
--- a/node_modules/react-native-reanimated/lib/module/reanimated2/index.js
+++ b/node_modules/react-native-reanimated/lib/module/reanimated2/index.js
@@ -1,15 +1,15 @@
 import './publicGlobals';
-export { runOnJS, runOnUI, createWorkletRuntime, WorkletRuntime, makeMutable, makeShareableCloneRecursive, isReanimated3, isConfigured, enableLayoutAnimations, getViewProp } from './core';
-export { useAnimatedProps, useEvent, useHandler, useWorkletCallback, useSharedValue, useReducedMotion, useAnimatedStyle, useAnimatedGestureHandler, GestureHandlers, useAnimatedReaction, AnimatedRef, useAnimatedRef, useAnimatedScrollHandler, ScrollHandler, ScrollHandlers, useDerivedValue, DerivedValue, useAnimatedSensor, useFrameCallback, FrameCallback, useAnimatedKeyboard, useScrollViewOffset } from './hook';
-export { DelayAnimation, RepeatAnimation, SequenceAnimation, StyleLayoutAnimation, cancelAnimation, defineAnimation, withTiming, WithTimingConfig, TimingAnimation, withSpring, WithSpringConfig, SpringAnimation, withDecay, WithDecayConfig, DecayAnimation, withDelay, withRepeat, withSequence } from './animation';
-export { Extrapolation, ExtrapolationConfig, ExtrapolationType, interpolate, clamp } from './interpolation';
-export { Extrapolate, InterpolationOptions, interpolateColor, ColorSpace, InterpolateConfig, useInterpolateConfig, InterpolateRGB, InterpolateHSV } from './interpolateColor';
-export { EasingFunction, EasingFn, EasingFunctionFactory, EasingFactoryFn, Easing } from './Easing';
+export { runOnJS, runOnUI, createWorkletRuntime, makeMutable, makeShareableCloneRecursive, isReanimated3, isConfigured, enableLayoutAnimations, getViewProp } from './core';
+export { useAnimatedProps, useEvent, useHandler, useWorkletCallback, useSharedValue, useReducedMotion, useAnimatedStyle, useAnimatedGestureHandler, useAnimatedReaction, useAnimatedRef, useAnimatedScrollHandler, useDerivedValue, useAnimatedSensor, useFrameCallback, useAnimatedKeyboard, useScrollViewOffset } from './hook';
+export { cancelAnimation, defineAnimation, withTiming, withSpring, withDecay, withDelay, withRepeat, withSequence } from './animation';
+export { interpolate, clamp } from './interpolation';
+export { Extrapolate, interpolateColor, useInterpolateConfig } from './interpolateColor';
+export { Easing } from './Easing';
 export { measure, dispatchCommand, scrollTo, setGestureState } from './NativeMethods';
 export { setNativeProps } from './SetNativeProps';
-export { isColor, processColor, ParsedColorArray, convertToRGBA } from './Colors';
+export { isColor, processColor, convertToRGBA } from './Colors';
 export { createAnimatedPropAdapter } from './PropAdapters';
-export { BaseAnimationBuilder, ComplexAnimationBuilder, Keyframe, LayoutAnimation, EntryAnimationsValues, ExitAnimationsValues, EntryExitAnimationFunction, LayoutAnimationsValues, LayoutAnimationFunction, ILayoutAnimationBuilder, IEntryExitAnimationBuilder,
+export { BaseAnimationBuilder, ComplexAnimationBuilder, Keyframe,
 // Flip
 FlipInXUp, FlipInYLeft, FlipInXDown, FlipInYRight, FlipInEasyX, FlipInEasyY, FlipOutXUp, FlipOutYLeft, FlipOutXDown, FlipOutYRight, FlipOutEasyX, FlipOutEasyY,
 // Stretch
@@ -34,9 +34,7 @@ RollInLeft, RollInRight, RollOutLeft, RollOutRight,
 Layout, LinearTransition, FadingTransition, SequencedTransition, JumpingTransition, CurvedTransition, EntryExitTransition, combineTransition,
 // SET
 SharedTransition, SharedTransitionType } from './layoutReanimation';
-export { getRelativeCoords, ComponentCoords, isSharedValue } from './utils';
-export { StyleProps, SharedValue, AnimatableValueObject, AnimatableValue, AnimationObject, Animation, SensorType, IOSReferenceFrame, SensorConfig, AnimatedSensor, AnimationCallback, Value3D, ValueRotation, InterfaceOrientation, KeyboardState, AnimatedKeyboardInfo, MeasuredDimensions, AnimatedKeyboardOptions, ReduceMotion } from './commonTypes';
-export { FrameInfo } from './frameCallback';
+export { getRelativeCoords, isSharedValue } from './utils';
+export { SensorType, IOSReferenceFrame, InterfaceOrientation, KeyboardState, ReduceMotion } from './commonTypes';
 export { getUseOfValueInStyleWarning } from './pluginUtils';
-export { withReanimatedTimer, advanceAnimationByTime, advanceAnimationByFrame, setUpTests, getAnimatedStyle } from './jestUtils';
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts b/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
index 4df2481..cca7f1a 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
@@ -40,5 +40,5 @@ export function isReducedMotion() {
     ? isWindowAvailable()
       ? !window.matchMedia('(prefers-reduced-motion: no-preference)').matches
       : false
-    : global._REANIMATED_IS_REDUCED_MOTION ?? false;
+    : (global as any)._REANIMATED_IS_REDUCED_MOTION ?? false;
 }
