diff --git a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/CameraSession.kt b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/CameraSession.kt
index 9a0414f..fc158f2 100644
--- a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/CameraSession.kt
+++ b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/CameraSession.kt
@@ -548,8 +548,7 @@ class CameraSession(
     mutex.withLock {
       val session = captureSession
       if (session == null) {
-        // Not yet ready. Start session first, then it will update repeating request.
-        startRunning()
+        // Not yet ready.
         return
       }
 
