diff --git a/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt b/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt
index 7ef8b36..f4d44ff 100644
--- a/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt
+++ b/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt
@@ -74,7 +74,7 @@ class EdgeToEdgeReactViewGroup(
   }
 
   override fun onConfigurationChanged(newConfig: Configuration?) {
-    this.reApplyWindowInsets()
+    // this.reApplyWindowInsets()
   }
   // endregion
 
@@ -124,12 +124,12 @@ class EdgeToEdgeReactViewGroup(
   }
 
   private fun goToEdgeToEdge(edgeToEdge: Boolean) {
-    reactContext.currentActivity?.let {
-      WindowCompat.setDecorFitsSystemWindows(
-        it.window,
-        !edgeToEdge,
-      )
-    }
+    // reactContext.currentActivity?.let {
+    //   WindowCompat.setDecorFitsSystemWindows(
+    //     it.window,
+    //     !edgeToEdge,
+    //   )
+    // }
   }
 
   private fun setupKeyboardCallbacks() {
@@ -182,16 +182,16 @@ class EdgeToEdgeReactViewGroup(
   // region State managers
   private fun enable() {
     this.goToEdgeToEdge(true)
-    this.setupWindowInsets()
+    // this.setupWindowInsets()
     this.setupKeyboardCallbacks()
-    modalAttachedWatcher.enable()
+    // modalAttachedWatcher.enable()
   }
 
   private fun disable() {
     this.goToEdgeToEdge(false)
-    this.setupWindowInsets()
+    // this.setupWindowInsets()
     this.removeKeyboardCallbacks()
-    modalAttachedWatcher.disable()
+    // modalAttachedWatcher.disable()
   }
   // endregion
 
@@ -219,7 +219,7 @@ class EdgeToEdgeReactViewGroup(
   fun forceStatusBarTranslucent(isStatusBarTranslucent: Boolean) {
     if (active && this.isStatusBarTranslucent != isStatusBarTranslucent) {
       this.isStatusBarTranslucent = isStatusBarTranslucent
-      this.reApplyWindowInsets()
+      // this.reApplyWindowInsets()
     }
   }
   // endregion
