diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java
index 26dc163..4dd8a47 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java
@@ -18,6 +18,8 @@ import android.graphics.Paint;
 import android.graphics.PorterDuff;
 import android.graphics.drawable.Drawable;
 import android.os.Build;
+import android.os.Handler;
+import android.os.Looper;
 import android.text.Editable;
 import android.text.InputFilter;
 import android.text.InputType;
@@ -1231,12 +1233,14 @@ public class ReactTextInputManager extends BaseViewManager<ReactEditText, Layout
         mPreviousContentHeight = contentHeight;
         mPreviousContentWidth = contentWidth;
 
-        mEventDispatcher.dispatchEvent(
-            new ReactContentSizeChangedEvent(
-                mSurfaceId,
-                mEditText.getId(),
-                PixelUtil.toDIPFromPixel(contentWidth),
-                PixelUtil.toDIPFromPixel(contentHeight)));
+        // fixes `onContentSizeChanged` not being fired in JS on mount
+        Handler handler = new Handler(Looper.getMainLooper());
+        handler.postDelayed(() -> mEventDispatcher.dispatchEvent(
+                new ReactContentSizeChangedEvent(
+                        mSurfaceId,
+                        mEditText.getId(),
+                        PixelUtil.toDIPFromPixel(mPreviousContentWidth),
+                        PixelUtil.toDIPFromPixel(mPreviousContentHeight))), 1);
       }
     }
   }
