diff --git a/node_modules/react-native-web/dist/exports/Modal/index.js b/node_modules/react-native-web/dist/exports/Modal/index.js
index d5df021..e2c46cf 100644
--- a/node_modules/react-native-web/dist/exports/Modal/index.js
+++ b/node_modules/react-native-web/dist/exports/Modal/index.js
@@ -86,13 +86,11 @@ var Modal = /*#__PURE__*/React.forwardRef((props, forwardedRef) => {
     onDismiss: onDismissCallback,
     onShow: onShowCallback,
     visible: visible
-  }, /*#__PURE__*/React.createElement(ModalFocusTrap, {
-    active: isActive
   }, /*#__PURE__*/React.createElement(ModalContent, _extends({}, rest, {
     active: isActive,
     onRequestClose: onRequestClose,
     ref: forwardedRef,
     transparent: transparent
-  }), children))));
+  }), children)));
 });
 export default Modal;
\ No newline at end of file
diff --git a/node_modules/react-native-web/src/exports/Text/index.js b/node_modules/react-native-web/src/exports/Text/index.js
index 071ae10..e43042c 100644
--- a/node_modules/react-native-web/src/exports/Text/index.js
+++ b/node_modules/react-native-web/src/exports/Text/index.js
@@ -219,7 +219,7 @@ const styles = StyleSheet.create({
   textMultiLine: {
     display: '-webkit-box',
     maxWidth: '100%',
-    overflow: 'hidden',
+    overflow: 'clip',
     textOverflow: 'ellipsis',
     WebkitBoxOrient: 'vertical'
   },
