diff --git a/node_modules/react-native-svg/common/cpp/react/renderer/components/rnsvg/RNSVGImageState.h b/node_modules/react-native-svg/common/cpp/react/renderer/components/rnsvg/RNSVGImageState.h
index 746b801..ac571a4 100644
--- a/node_modules/react-native-svg/common/cpp/react/renderer/components/rnsvg/RNSVGImageState.h
+++ b/node_modules/react-native-svg/common/cpp/react/renderer/components/rnsvg/RNSVGImageState.h
@@ -14,6 +14,7 @@
 #ifdef ANDROID
 #include <react/renderer/mapbuffer/MapBuffer.h>
 #include <react/renderer/mapbuffer/MapBufferBuilder.h>
+#include <folly/dynamic.h>
 #endif
 
 namespace facebook {
