diff --git a/node_modules/react-fast-pdf/dist/PDFPreviewer.js b/node_modules/react-fast-pdf/dist/PDFPreviewer.js
index 4407807..ea3964d 100644
--- a/node_modules/react-fast-pdf/dist/PDFPreviewer.js
+++ b/node_modules/react-fast-pdf/dist/PDFPreviewer.js
@@ -28,7 +28,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 Object.defineProperty(exports, "__esModule", { value: true });
 // @ts-expect-error - This line imports a module from 'pdfjs-dist' package which lacks TypeScript typings.
 // eslint-disable-next-line import/no-extraneous-dependencies
-const pdf_worker_1 = __importDefault(require("pdfjs-dist/legacy/build/pdf.worker"));
+const pdf_worker_1 = __importDefault(require("pdfjs-dist/legacy/build/pdf.worker.mjs"));
 const react_1 = __importStar(require("react"));
 const times_1 = __importDefault(require("lodash/times"));
 const prop_types_1 = __importDefault(require("prop-types"));
