diff --git a/node_modules/react-native-reanimated/src/mock.ts b/node_modules/react-native-reanimated/src/mock.ts
index 3d8e3f8..5eba613 100644
--- a/node_modules/react-native-reanimated/src/mock.ts
+++ b/node_modules/react-native-reanimated/src/mock.ts
@@ -87,7 +87,12 @@ const hook = {
   useAnimatedReaction: NOOP,
   useAnimatedRef: () => ({ current: null }),
   useAnimatedScrollHandler: NOOP_FACTORY,
-  useDerivedValue: <Value>(processor: () => Value) => ({ value: processor() }),
+  // https://github.com/software-mansion/react-native-reanimated/pull/6809
+  useDerivedValue: <Value>(processor: () => Value) => {
+    const result = processor();
+
+    return { value: result, get: () => result };
+  },
   useAnimatedSensor: () => ({
     sensor: {
       value: {
