diff --git a/node_modules/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.mm b/node_modules/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.mm
index 20807aa..31aec69 100644
--- a/node_modules/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.mm
+++ b/node_modules/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.mm
@@ -294,6 +294,15 @@ - (BOOL)canPerformAction:(SEL)action withSender:(id)sender
   return [super canPerformAction:action withSender:sender];
 }
 
+-(void)buildMenuWithBuilder:(id<UIMenuBuilder>)builder {
+    if (_contextMenuHidden) {
+        if (@available(iOS 17.0, *)) {
+            [builder removeMenuForIdentifier:UIMenuAutoFill];
+        }
+    }
+    [super buildMenuWithBuilder:builder];
+}
+
 #pragma mark - Dictation
 
 - (void)dictationRecordingDidEnd
diff --git a/node_modules/react-native/Libraries/Text/TextInput/Singleline/RCTUITextField.mm b/node_modules/react-native/Libraries/Text/TextInput/Singleline/RCTUITextField.mm
index 507df43..1461bb4 100644
--- a/node_modules/react-native/Libraries/Text/TextInput/Singleline/RCTUITextField.mm
+++ b/node_modules/react-native/Libraries/Text/TextInput/Singleline/RCTUITextField.mm
@@ -150,6 +150,15 @@ - (BOOL)canPerformAction:(SEL)action withSender:(id)sender
   return [super canPerformAction:action withSender:sender];
 }
 
+-(void)buildMenuWithBuilder:(id<UIMenuBuilder>)builder {
+    if (_contextMenuHidden) {
+        if (@available(iOS 17.0, *)) {
+            [builder removeMenuForIdentifier:UIMenuAutoFill];
+        }
+    }
+    [super buildMenuWithBuilder:builder];
+}
+
 #pragma mark - Dictation
 
 - (void)dictationRecordingDidEnd
