diff --git a/node_modules/react-native-tab-view/lib/module/TabView.js b/node_modules/react-native-tab-view/lib/module/TabView.js
index ba28170..4c607a2 100644
--- a/node_modules/react-native-tab-view/lib/module/TabView.js
+++ b/node_modules/react-native-tab-view/lib/module/TabView.js
@@ -40,6 +40,9 @@ export function TabView(_ref) {
       height,
       width
     } = e.nativeEvent.layout;
+    if(!!width || !!height) {
+      return;
+    }
     setLayout(prevLayout => {
       if (prevLayout.width === width && prevLayout.height === height) {
         return prevLayout;
diff --git a/node_modules/react-native-tab-view/src/TabView.tsx b/node_modules/react-native-tab-view/src/TabView.tsx
index bb1f531..d70f9ba 100644
--- a/node_modules/react-native-tab-view/src/TabView.tsx
+++ b/node_modules/react-native-tab-view/src/TabView.tsx
@@ -70,6 +70,10 @@ export function TabView<T extends Route>({
   const handleLayout = (e: LayoutChangeEvent) => {
     const { height, width } = e.nativeEvent.layout;
 
+    if(!!width || !!height) {
+      return;
+    }
+
     setLayout((prevLayout) => {
       if (prevLayout.width === width && prevLayout.height === height) {
         return prevLayout;
