diff --git a/node_modules/react-compiler-healthcheck/dist/index.js b/node_modules/react-compiler-healthcheck/dist/index.js
index c2d96fd..93be1fb 100755
--- a/node_modules/react-compiler-healthcheck/dist/index.js
+++ b/node_modules/react-compiler-healthcheck/dist/index.js
@@ -69174,10 +69174,10 @@ var reactCompilerCheck = {
       const extractFileName = (output) => output.fnLoc.filename;
       const successfulFiles = SucessfulCompilation.map(extractFileName);
       const unsuccessfulFiles = [...new Set([...OtherFailures, ...ActionableFailures].map(extractFileName))];
-      console.log({
+      console.log(JSON.stringify({
         success: successfulFiles,
         failure: unsuccessfulFiles,
-      });
+      }));
     }
     
     if (verbose) {
