diff --git a/node_modules/react-native/ReactCommon/react/renderer/core/ShadowNode.cpp b/node_modules/react-native/ReactCommon/react/renderer/core/ShadowNode.cpp
index a9ca4d7..69e335f 100644
--- a/node_modules/react-native/ReactCommon/react/renderer/core/ShadowNode.cpp
+++ b/node_modules/react-native/ReactCommon/react/renderer/core/ShadowNode.cpp
@@ -363,7 +363,8 @@ ShadowNode::Unshared ShadowNode::cloneTree(
     children[childIndex] = childNode;
 
     childNode = parentNode.clone(
-        {.children = std::make_shared<ShadowNode::ListOfShared>(children)});
+        {.children = std::make_shared<ShadowNode::ListOfShared>(children),
+         .state = parentNode.getState()});
   }
 
   return std::const_pointer_cast<ShadowNode>(childNode);
