diff --git a/node_modules/react-native-reanimated/src/component/PerformanceMonitor.tsx b/node_modules/react-native-reanimated/src/component/PerformanceMonitor.tsx
index ff8fc8a..025cca4 100644
--- a/node_modules/react-native-reanimated/src/component/PerformanceMonitor.tsx
+++ b/node_modules/react-native-reanimated/src/component/PerformanceMonitor.tsx
@@ -3,7 +3,6 @@
 import React, { useEffect, useRef } from 'react';
 import { StyleSheet, TextInput, View } from 'react-native';

-import { addWhitelistedNativeProps } from '../ConfigHelper';
 import { createAnimatedComponent } from '../createAnimatedComponent';
 import type { FrameInfo } from '../frameCallback';
 import { useAnimatedProps, useFrameCallback, useSharedValue } from '../hook';
@@ -46,7 +45,6 @@ function createCircularDoublesBuffer(size: number) {
 }

 const DEFAULT_BUFFER_SIZE = 20;
-addWhitelistedNativeProps({ text: true });
 const AnimatedTextInput = createAnimatedComponent(TextInput);

 function loopAnimationFrame(fn: (lastTime: number, time: number) => void) {
