diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java
index ce8520e..e2b22b0 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java
@@ -122,6 +122,7 @@ public class FabricEventDispatcher implements EventDispatcher, LifecycleEventLis
   @Override
   public void onHostResume() {
     maybePostFrameCallbackFromNonUI();
+    mCurrentFrameCallback.resume();
   }
 
   @Override
@@ -190,6 +191,10 @@ public class FabricEventDispatcher implements EventDispatcher, LifecycleEventLis
       mShouldStop = true;
     }
 
+    public void resume() {
+      mShouldStop = false;
+    }
+
     public void maybePost() {
       if (!mIsPosted) {
         mIsPosted = true;
