diff --git a/node_modules/react-native-draggable-flatlist/.github/FUNDING.yml b/node_modules/react-native-draggable-flatlist/.github/FUNDING.yml
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/.github/ISSUE_TEMPLATE/bug_report.md b/node_modules/react-native-draggable-flatlist/.github/ISSUE_TEMPLATE/bug_report.md
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/.github/ISSUE_TEMPLATE/feature_request.md b/node_modules/react-native-draggable-flatlist/.github/ISSUE_TEMPLATE/feature_request.md
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/.prettierignore b/node_modules/react-native-draggable-flatlist/.prettierignore
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/.vscode/settings.json b/node_modules/react-native-draggable-flatlist/.vscode/settings.json
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/LICENSE.txt b/node_modules/react-native-draggable-flatlist/LICENSE.txt
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/README.md b/node_modules/react-native-draggable-flatlist/README.md
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/babel.config.js b/node_modules/react-native-draggable-flatlist/babel.config.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/jest-setup.js b/node_modules/react-native-draggable-flatlist/jest-setup.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/CellDecorators.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/CellDecorators.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/CellDecorators.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/CellDecorators.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/CellRendererComponent.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/CellRendererComponent.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/CellRendererComponent.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/CellRendererComponent.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/DraggableFlatList.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/DraggableFlatList.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/DraggableFlatList.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/DraggableFlatList.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/NestableDraggableFlatList.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/NestableDraggableFlatList.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/NestableDraggableFlatList.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/NestableDraggableFlatList.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/NestableScrollContainer.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/NestableScrollContainer.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/NestableScrollContainer.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/NestableScrollContainer.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/PlaceholderItem.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/PlaceholderItem.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/PlaceholderItem.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/PlaceholderItem.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/RowItem.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/RowItem.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/RowItem.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/RowItem.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/ScrollOffsetListener.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/ScrollOffsetListener.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/components/ScrollOffsetListener.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/components/ScrollOffsetListener.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/constants.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/constants.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/constants.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/constants.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/animatedValueContext.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/animatedValueContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/animatedValueContext.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/animatedValueContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/cellContext.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/cellContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/cellContext.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/cellContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/draggableFlatListContext.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/draggableFlatListContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/draggableFlatListContext.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/draggableFlatListContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/nestableScrollContainerContext.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/nestableScrollContainerContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/nestableScrollContainerContext.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/nestableScrollContainerContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/propsContext.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/propsContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/propsContext.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/propsContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/refContext.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/refContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/context/refContext.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/context/refContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useAutoScroll.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useAutoScroll.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useAutoScroll.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useAutoScroll.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useCellTranslate.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useCellTranslate.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useCellTranslate.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useCellTranslate.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useNestedAutoScroll.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useNestedAutoScroll.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useNestedAutoScroll.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useNestedAutoScroll.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useOnCellActiveAnimation.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useOnCellActiveAnimation.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useOnCellActiveAnimation.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useOnCellActiveAnimation.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useStableCallback.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useStableCallback.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useStableCallback.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/hooks/useStableCallback.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/index.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/index.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/index.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/index.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/types.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/types.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/types.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/types.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/utils.js b/node_modules/react-native-draggable-flatlist/lib/commonjs/utils.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/commonjs/utils.js.map b/node_modules/react-native-draggable-flatlist/lib/commonjs/utils.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/CellDecorators.js b/node_modules/react-native-draggable-flatlist/lib/module/components/CellDecorators.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/CellDecorators.js.map b/node_modules/react-native-draggable-flatlist/lib/module/components/CellDecorators.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/CellRendererComponent.js b/node_modules/react-native-draggable-flatlist/lib/module/components/CellRendererComponent.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/CellRendererComponent.js.map b/node_modules/react-native-draggable-flatlist/lib/module/components/CellRendererComponent.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/DraggableFlatList.js b/node_modules/react-native-draggable-flatlist/lib/module/components/DraggableFlatList.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/DraggableFlatList.js.map b/node_modules/react-native-draggable-flatlist/lib/module/components/DraggableFlatList.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/NestableDraggableFlatList.js b/node_modules/react-native-draggable-flatlist/lib/module/components/NestableDraggableFlatList.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/NestableDraggableFlatList.js.map b/node_modules/react-native-draggable-flatlist/lib/module/components/NestableDraggableFlatList.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/NestableScrollContainer.js b/node_modules/react-native-draggable-flatlist/lib/module/components/NestableScrollContainer.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/NestableScrollContainer.js.map b/node_modules/react-native-draggable-flatlist/lib/module/components/NestableScrollContainer.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/PlaceholderItem.js b/node_modules/react-native-draggable-flatlist/lib/module/components/PlaceholderItem.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/PlaceholderItem.js.map b/node_modules/react-native-draggable-flatlist/lib/module/components/PlaceholderItem.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/RowItem.js b/node_modules/react-native-draggable-flatlist/lib/module/components/RowItem.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/RowItem.js.map b/node_modules/react-native-draggable-flatlist/lib/module/components/RowItem.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/ScrollOffsetListener.js b/node_modules/react-native-draggable-flatlist/lib/module/components/ScrollOffsetListener.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/components/ScrollOffsetListener.js.map b/node_modules/react-native-draggable-flatlist/lib/module/components/ScrollOffsetListener.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/constants.js b/node_modules/react-native-draggable-flatlist/lib/module/constants.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/constants.js.map b/node_modules/react-native-draggable-flatlist/lib/module/constants.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/animatedValueContext.js b/node_modules/react-native-draggable-flatlist/lib/module/context/animatedValueContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/animatedValueContext.js.map b/node_modules/react-native-draggable-flatlist/lib/module/context/animatedValueContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/cellContext.js b/node_modules/react-native-draggable-flatlist/lib/module/context/cellContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/cellContext.js.map b/node_modules/react-native-draggable-flatlist/lib/module/context/cellContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/draggableFlatListContext.js b/node_modules/react-native-draggable-flatlist/lib/module/context/draggableFlatListContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/draggableFlatListContext.js.map b/node_modules/react-native-draggable-flatlist/lib/module/context/draggableFlatListContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/nestableScrollContainerContext.js b/node_modules/react-native-draggable-flatlist/lib/module/context/nestableScrollContainerContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/nestableScrollContainerContext.js.map b/node_modules/react-native-draggable-flatlist/lib/module/context/nestableScrollContainerContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/propsContext.js b/node_modules/react-native-draggable-flatlist/lib/module/context/propsContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/propsContext.js.map b/node_modules/react-native-draggable-flatlist/lib/module/context/propsContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/refContext.js b/node_modules/react-native-draggable-flatlist/lib/module/context/refContext.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/context/refContext.js.map b/node_modules/react-native-draggable-flatlist/lib/module/context/refContext.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useAutoScroll.js b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useAutoScroll.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useAutoScroll.js.map b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useAutoScroll.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useCellTranslate.js b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useCellTranslate.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useCellTranslate.js.map b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useCellTranslate.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useNestedAutoScroll.js b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useNestedAutoScroll.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useNestedAutoScroll.js.map b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useNestedAutoScroll.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useOnCellActiveAnimation.js b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useOnCellActiveAnimation.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useOnCellActiveAnimation.js.map b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useOnCellActiveAnimation.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useStableCallback.js b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useStableCallback.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/hooks/useStableCallback.js.map b/node_modules/react-native-draggable-flatlist/lib/module/hooks/useStableCallback.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/index.js b/node_modules/react-native-draggable-flatlist/lib/module/index.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/index.js.map b/node_modules/react-native-draggable-flatlist/lib/module/index.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/types.js b/node_modules/react-native-draggable-flatlist/lib/module/types.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/types.js.map b/node_modules/react-native-draggable-flatlist/lib/module/types.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/utils.js b/node_modules/react-native-draggable-flatlist/lib/module/utils.js
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/module/utils.js.map b/node_modules/react-native-draggable-flatlist/lib/module/utils.js.map
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/components/CellDecorators.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/components/CellDecorators.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/components/CellRendererComponent.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/components/CellRendererComponent.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/components/DraggableFlatList.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/components/DraggableFlatList.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/components/NestableDraggableFlatList.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/components/NestableDraggableFlatList.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/components/NestableScrollContainer.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/components/NestableScrollContainer.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/components/PlaceholderItem.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/components/PlaceholderItem.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/components/RowItem.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/components/RowItem.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/components/ScrollOffsetListener.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/components/ScrollOffsetListener.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/constants.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/constants.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/context/animatedValueContext.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/context/animatedValueContext.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/context/cellContext.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/context/cellContext.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/context/draggableFlatListContext.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/context/draggableFlatListContext.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/context/nestableScrollContainerContext.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/context/nestableScrollContainerContext.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/context/propsContext.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/context/propsContext.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/context/refContext.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/context/refContext.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useAutoScroll.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useAutoScroll.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useCellTranslate.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useCellTranslate.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useNestedAutoScroll.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useNestedAutoScroll.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useOnCellActiveAnimation.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useOnCellActiveAnimation.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useStableCallback.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/hooks/useStableCallback.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/index.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/index.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/types.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/types.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/lib/typescript/utils.d.ts b/node_modules/react-native-draggable-flatlist/lib/typescript/utils.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/components/CellDecorators.tsx b/node_modules/react-native-draggable-flatlist/src/components/CellDecorators.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/components/CellRendererComponent.tsx b/node_modules/react-native-draggable-flatlist/src/components/CellRendererComponent.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/components/DraggableFlatList.tsx b/node_modules/react-native-draggable-flatlist/src/components/DraggableFlatList.tsx
old mode 100644
new mode 100755
index d7d98c2..2f59c7a
--- a/node_modules/react-native-draggable-flatlist/src/components/DraggableFlatList.tsx
+++ b/node_modules/react-native-draggable-flatlist/src/components/DraggableFlatList.tsx
@@ -295,7 +295,7 @@ function DraggableFlatListInner<T>(props: DraggableFlatListProps<T>) {
       const springTo = placeholderOffset.value - activeCellOffset.value;
       touchTranslate.value = withSpring(
         springTo,
-        animationConfigRef.current,
+        animationConfigRef.value,
         () => {
           runOnJS(onDragEnd)({
             from: activeIndexAnim.value,
diff --git a/node_modules/react-native-draggable-flatlist/src/components/NestableDraggableFlatList.tsx b/node_modules/react-native-draggable-flatlist/src/components/NestableDraggableFlatList.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/components/NestableScrollContainer.tsx b/node_modules/react-native-draggable-flatlist/src/components/NestableScrollContainer.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/components/PlaceholderItem.tsx b/node_modules/react-native-draggable-flatlist/src/components/PlaceholderItem.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/components/RowItem.tsx b/node_modules/react-native-draggable-flatlist/src/components/RowItem.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/components/ScrollOffsetListener.tsx b/node_modules/react-native-draggable-flatlist/src/components/ScrollOffsetListener.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/constants.ts b/node_modules/react-native-draggable-flatlist/src/constants.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/context/animatedValueContext.tsx b/node_modules/react-native-draggable-flatlist/src/context/animatedValueContext.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/context/cellContext.tsx b/node_modules/react-native-draggable-flatlist/src/context/cellContext.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/context/draggableFlatListContext.tsx b/node_modules/react-native-draggable-flatlist/src/context/draggableFlatListContext.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/context/nestableScrollContainerContext.tsx b/node_modules/react-native-draggable-flatlist/src/context/nestableScrollContainerContext.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/context/propsContext.tsx b/node_modules/react-native-draggable-flatlist/src/context/propsContext.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/context/refContext.tsx b/node_modules/react-native-draggable-flatlist/src/context/refContext.tsx
old mode 100644
new mode 100755
index ea21575..dc6b095
--- a/node_modules/react-native-draggable-flatlist/src/context/refContext.tsx
+++ b/node_modules/react-native-draggable-flatlist/src/context/refContext.tsx
@@ -1,14 +1,14 @@
 import React, { useContext } from "react";
 import { useMemo, useRef } from "react";
 import { FlatList } from "react-native-gesture-handler";
-import Animated, { WithSpringConfig } from "react-native-reanimated";
+import Animated, { type SharedValue, useSharedValue, WithSpringConfig } from "react-native-reanimated";
 import { DEFAULT_PROPS } from "../constants";
 import { useProps } from "./propsContext";
 import { CellData, DraggableFlatListProps } from "../types";
 
 type RefContextValue<T> = {
   propsRef: React.MutableRefObject<DraggableFlatListProps<T>>;
-  animationConfigRef: React.MutableRefObject<WithSpringConfig>;
+  animationConfigRef: SharedValue<WithSpringConfig>;
   cellDataRef: React.MutableRefObject<Map<string, CellData>>;
   keyToIndexRef: React.MutableRefObject<Map<string, number>>;
   containerRef: React.RefObject<Animated.View>;
@@ -54,8 +54,7 @@ function useSetupRefs<T>({
     ...DEFAULT_PROPS.animationConfig,
     ...animationConfig,
   } as WithSpringConfig;
-  const animationConfigRef = useRef(animConfig);
-  animationConfigRef.current = animConfig;
+  const animationConfigRef = useSharedValue(animConfig);
 
   const cellDataRef = useRef(new Map<string, CellData>());
   const keyToIndexRef = useRef(new Map<string, number>());
diff --git a/node_modules/react-native-draggable-flatlist/src/hooks/useAutoScroll.tsx b/node_modules/react-native-draggable-flatlist/src/hooks/useAutoScroll.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/hooks/useCellTranslate.tsx b/node_modules/react-native-draggable-flatlist/src/hooks/useCellTranslate.tsx
old mode 100644
new mode 100755
index ce4ab68..efea240
--- a/node_modules/react-native-draggable-flatlist/src/hooks/useCellTranslate.tsx
+++ b/node_modules/react-native-draggable-flatlist/src/hooks/useCellTranslate.tsx
@@ -101,7 +101,7 @@ export function useCellTranslate({ cellIndex, cellSize, cellOffset }: Params) {
       ? activeCellSize.value * (isAfterActive ? -1 : 1)
       : 0;
 
-    return withSpring(translationAmt, animationConfigRef.current);
+    return withSpring(translationAmt, animationConfigRef.value);
   }, [activeKey, cellIndex]);
 
   return translate;
diff --git a/node_modules/react-native-draggable-flatlist/src/hooks/useNestedAutoScroll.tsx b/node_modules/react-native-draggable-flatlist/src/hooks/useNestedAutoScroll.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/hooks/useOnCellActiveAnimation.ts b/node_modules/react-native-draggable-flatlist/src/hooks/useOnCellActiveAnimation.ts
old mode 100644
new mode 100755
index 7c20587..33042e9
--- a/node_modules/react-native-draggable-flatlist/src/hooks/useOnCellActiveAnimation.ts
+++ b/node_modules/react-native-draggable-flatlist/src/hooks/useOnCellActiveAnimation.ts
@@ -1,8 +1,9 @@
-import { useRef } from "react";
-import Animated, {
+
+import {
   useDerivedValue,
   withSpring,
   WithSpringConfig,
+  useSharedValue,
 } from "react-native-reanimated";
 import { DEFAULT_ANIMATION_CONFIG } from "../constants";
 import { useAnimatedValues } from "../context/animatedValueContext";
@@ -15,8 +16,7 @@ type Params = {
 export function useOnCellActiveAnimation(
   { animationConfig }: Params = { animationConfig: {} }
 ) {
-  const animationConfigRef = useRef(animationConfig);
-  animationConfigRef.current = animationConfig;
+  const animationConfigRef = useSharedValue(animationConfig);
 
   const isActive = useIsActive();
 
@@ -26,7 +26,7 @@ export function useOnCellActiveAnimation(
     const toVal = isActive && isTouchActiveNative.value ? 1 : 0;
     return withSpring(toVal, {
       ...DEFAULT_ANIMATION_CONFIG,
-      ...animationConfigRef.current,
+      ...animationConfigRef.value,
     });
   }, [isActive]);
 
diff --git a/node_modules/react-native-draggable-flatlist/src/hooks/useStableCallback.ts b/node_modules/react-native-draggable-flatlist/src/hooks/useStableCallback.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/index.tsx b/node_modules/react-native-draggable-flatlist/src/index.tsx
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/types.ts b/node_modules/react-native-draggable-flatlist/src/types.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/src/utils.ts b/node_modules/react-native-draggable-flatlist/src/utils.ts
old mode 100644
new mode 100755
diff --git a/node_modules/react-native-draggable-flatlist/tsconfig.json b/node_modules/react-native-draggable-flatlist/tsconfig.json
old mode 100644
new mode 100755
