diff --git a/node_modules/react-native-pager-view/ios/Fabric/RNCPagerViewComponentView.mm b/node_modules/react-native-pager-view/ios/Fabric/RNCPagerViewComponentView.mm
index 83fa497..a8f792a 100644
--- a/node_modules/react-native-pager-view/ios/Fabric/RNCPagerViewComponentView.mm
+++ b/node_modules/react-native-pager-view/ios/Fabric/RNCPagerViewComponentView.mm
@@ -2,7 +2,7 @@
 
 #import <Foundation/Foundation.h>
 #import "RNCPagerViewComponentView.h"
-#import <RNCViewPager/RNCViewPagerComponentDescriptor.h>
+#import <react/renderer/components/RNCViewPager/RNCViewPagerComponentDescriptor.h>
 #import <react/renderer/components/pagerview/EventEmitters.h>
 #import <react/renderer/components/pagerview/Props.h>
 #import <react/renderer/components/pagerview/RCTComponentViewHelpers.h>
