diff --git a/node_modules/react-native/Libraries/Animated/nodes/AnimatedProps.js b/node_modules/react-native/Libraries/Animated/nodes/AnimatedProps.js
index feeaa19..9b69879 100644
--- a/node_modules/react-native/Libraries/Animated/nodes/AnimatedProps.js
+++ b/node_modules/react-native/Libraries/Animated/nodes/AnimatedProps.js
@@ -79,6 +79,7 @@ export default class AnimatedProps extends AnimatedNode {
   #nodeKeys: $ReadOnlyArray<string>;
   #nodes: $ReadOnlyArray<AnimatedNode>;
   #props: {[string]: mixed};
+  #connectedToView: boolean;
 
   constructor(
     inputProps: {[string]: mixed},
@@ -91,6 +92,7 @@ export default class AnimatedProps extends AnimatedNode {
     this.#nodes = nodes;
     this.#props = props;
     this.#callback = callback;
+    this.#connectedToView = false;
   }
 
   __getValue(): Object {
@@ -196,6 +198,8 @@ export default class AnimatedProps extends AnimatedNode {
 
       if (this.#animatedView) {
         this.__connectAnimatedView();
+      } else if (!this.#connectedToView && this._animatedView) {
+        this.__connectAnimatedView();
       }
     }
   }
@@ -220,6 +224,7 @@ export default class AnimatedProps extends AnimatedNode {
         throw new Error('Unable to locate attached view in the native tree');
       }
     }
+    this.#connectedToView = true;
     NativeAnimatedHelper.API.connectAnimatedNodeToView(
       this.__getNativeTag(),
       nativeViewTag,
@@ -236,6 +241,7 @@ export default class AnimatedProps extends AnimatedNode {
         throw new Error('Unable to locate attached view in the native tree');
       }
     }
+    this.#connectedToView = false;
     NativeAnimatedHelper.API.disconnectAnimatedNodeFromView(
       this.__getNativeTag(),
       nativeViewTag,
