diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultNewArchitectureEntryPoint.kt b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultNewArchitectureEntryPoint.kt
index f90ad02..44a8e36 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultNewArchitectureEntryPoint.kt
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultNewArchitectureEntryPoint.kt
@@ -50,6 +50,8 @@ public object DefaultNewArchitectureEntryPoint {
           override fun enableEventEmitterRetentionDuringGesturesOnAndroid(): Boolean =
               bridgelessEnabled || fabricEnabled
 
+          override fun enableAndroidLineHeightCentering(): Boolean = false
+
           override fun useTurboModules(): Boolean = bridgelessEnabled || turboModulesEnabled
         })
 
