diff --git a/node_modules/react-native-reanimated/lib/module/animation/util.js b/node_modules/react-native-reanimated/lib/module/animation/util.js
index 2bc36ca..4f70add 100644
--- a/node_modules/react-native-reanimated/lib/module/animation/util.js
+++ b/node_modules/react-native-reanimated/lib/module/animation/util.js
@@ -357,7 +357,7 @@ function cancelAnimationNative(sharedValue) {
   'worklet';
 
   // setting the current value cancels the animation if one is currently running
-  if (_WORKLET) {
+  if (_WORKLET || SHOULD_BE_USE_WEB) {
     sharedValue.value = sharedValue.value; // eslint-disable-line no-self-assign
   } else {
     runOnUI(() => {
diff --git a/node_modules/react-native-reanimated/src/animation/util.ts b/node_modules/react-native-reanimated/src/animation/util.ts
index e0176f3..8d44aa5 100644
--- a/node_modules/react-native-reanimated/src/animation/util.ts
+++ b/node_modules/react-native-reanimated/src/animation/util.ts
@@ -562,7 +562,7 @@ export function defineAnimation<
 function cancelAnimationNative<TValue>(sharedValue: SharedValue<TValue>): void {
   'worklet';
   // setting the current value cancels the animation if one is currently running
-  if (_WORKLET) {
+  if (_WORKLET || SHOULD_BE_USE_WEB) {
     sharedValue.value = sharedValue.value; // eslint-disable-line no-self-assign
   } else {
     runOnUI(() => {
