diff --git a/node_modules/react-native/ReactAndroid/build.gradle b/node_modules/react-native/ReactAndroid/build.gradle
index 78c57eb..ec147fd 100644
--- a/node_modules/react-native/ReactAndroid/build.gradle
+++ b/node_modules/react-native/ReactAndroid/build.gradle
@@ -125,6 +125,19 @@ final def preparePrefab = tasks.register("preparePrefab", PreparePrefabHeadersTa
                 "rrc_root",
                 new Pair("../ReactCommon/react/renderer/components/root/", "react/renderer/components/root/")
             ),
+            new PrefabPreprocessingEntry(
+                "rrc_text",
+                [
+                    new Pair("../ReactCommon/react/renderer/components/text/", "react/renderer/components/text/"),
+                    new Pair("../ReactCommon/react/renderer/attributedstring", "react/renderer/attributedstring"),
+                ]
+            ),
+            new PrefabPreprocessingEntry(
+                "rrc_textinput",
+                [
+                    new Pair("../ReactCommon/react/renderer/components/textinput/androidtextinput", ""),
+                ]
+            ),
             new PrefabPreprocessingEntry(
                 "rrc_view",
                 [
@@ -132,6 +145,13 @@ final def preparePrefab = tasks.register("preparePrefab", PreparePrefabHeadersTa
                     new Pair("../ReactCommon/react/renderer/components/view/platform/android/", ""),
                 ]
             ),
+            new PrefabPreprocessingEntry(
+                "react_render_textlayoutmanager",
+                [
+                    new Pair("../ReactCommon/react/renderer/textlayoutmanager/", "react/renderer/textlayoutmanager/"),
+                    new Pair("../ReactCommon/react/renderer/textlayoutmanager/platform/android/", ""),
+                ]
+            ),
             new PrefabPreprocessingEntry(
                 "rrc_legacyviewmanagerinterop",
                 new Pair("../ReactCommon/react/renderer/components/legacyviewmanagerinterop/", "react/renderer/components/legacyviewmanagerinterop/")
@@ -559,6 +579,9 @@ android {
                     "glog",
                     "fabricjni",
                     "react_render_mapbuffer",
+                    "react_render_textlayoutmanager",
+                    "rrc_textinput",
+                    "rrc_text",
                     "yoga",
                     "folly_runtime",
                     "react_nativemodule_core",
@@ -683,6 +706,15 @@ android {
         rrc_root {
             headers(new File(prefabHeadersDir, "rrc_root").absolutePath)
         }
+        rrc_text {
+            headers(new File(prefabHeadersDir, "rrc_text").absolutePath)
+        }
+        rrc_textinput {
+            headers(new File(prefabHeadersDir, "rrc_textinput").absolutePath)
+        }
+        react_render_textlayoutmanager {
+            headers(new File(prefabHeadersDir, "react_render_textlayoutmanager").absolutePath)
+        }
         rrc_view {
             headers(new File(prefabHeadersDir, "rrc_view").absolutePath)
         }
diff --git a/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake b/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake
index d49fa9e..3607c69 100644
--- a/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake
+++ b/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake
@@ -78,11 +78,14 @@ add_library(jsi ALIAS ReactAndroid::jsi)
 add_library(glog ALIAS ReactAndroid::glog)
 add_library(fabricjni ALIAS ReactAndroid::fabricjni)
 add_library(react_render_mapbuffer ALIAS ReactAndroid::react_render_mapbuffer)
+add_library(react_render_textlayoutmanager ALIAS ReactAndroid::react_render_textlayoutmanager)
 add_library(yoga ALIAS ReactAndroid::yoga)
 add_library(folly_runtime ALIAS ReactAndroid::folly_runtime)
 add_library(react_nativemodule_core ALIAS ReactAndroid::react_nativemodule_core)
 add_library(react_render_imagemanager ALIAS ReactAndroid::react_render_imagemanager)
 add_library(rrc_image ALIAS ReactAndroid::rrc_image)
+add_library(rrc_text ALIAS ReactAndroid::rrc_text)
+add_library(rrc_textinput ALIAS ReactAndroid::rrc_textinput)
 add_library(rrc_legacyviewmanagerinterop ALIAS ReactAndroid::rrc_legacyviewmanagerinterop)
 
 find_package(fbjni REQUIRED CONFIG)
@@ -105,8 +108,11 @@ target_link_libraries(${CMAKE_PROJECT_NAME}
         react_render_graphics               # prefab ready
         react_render_imagemanager           # prefab ready
         react_render_mapbuffer              # prefab ready
+        react_render_textlayoutmanager      # prefab ready
         rrc_image                           # prefab ready
         rrc_view                            # prefab ready
+        rrc_text                            # prefab ready
+        rrc_textinput                       # prefab ready
         rrc_legacyviewmanagerinterop        # prefab ready
         runtimeexecutor                     # prefab ready
         turbomodulejsijni                   # prefab ready
