diff --git a/node_modules/react-native-screens/src/components/Screen.tsx b/node_modules/react-native-screens/src/components/Screen.tsx
index 3f9a1cb..45767f7 100644
--- a/node_modules/react-native-screens/src/components/Screen.tsx
+++ b/node_modules/react-native-screens/src/components/Screen.tsx
@@ -79,6 +79,7 @@ export class InnerScreen extends React.Component<ScreenProps> {
       // Due to how Yoga resolves layout, we need to have different components for modal nad non-modal screens
       const AnimatedScreen =
         Platform.OS === 'android' ||
+        stackPresentation === undefined ||
         stackPresentation === 'push' ||
         stackPresentation === 'containedModal' ||
         stackPresentation === 'containedTransparentModal'
