diff --git a/node_modules/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm b/node_modules/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm
index e78e8ae..f6465f3 100644
--- a/node_modules/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/RCTComponentViewRegistry.mm
@@ -104,18 +104,18 @@ - (RCTComponentViewDescriptor)_dequeueComponentViewWithComponentHandle:(Componen
 - (void)_enqueueComponentViewWithComponentHandle:(ComponentHandle)componentHandle
                          componentViewDescriptor:(RCTComponentViewDescriptor)componentViewDescriptor
 {
-  RCTAssertMainQueue();
-  auto &recycledViews = _recyclePool[componentHandle];
-
-  if (recycledViews.size() > RCTComponentViewRegistryRecyclePoolMaxSize) {
-    return;
-  }
-
-  RCTAssert(
-      componentViewDescriptor.view.superview == nil, @"RCTComponentViewRegistry: Attempt to recycle a mounted view.");
-  [componentViewDescriptor.view prepareForRecycle];
-
-  recycledViews.push_back(componentViewDescriptor);
+//  RCTAssertMainQueue();
+//  auto &recycledViews = _recyclePool[componentHandle];
+//
+//  if (recycledViews.size() > RCTComponentViewRegistryRecyclePoolMaxSize) {
+//    return;
+//  }
+//
+//  RCTAssert(
+//      componentViewDescriptor.view.superview == nil, @"RCTComponentViewRegistry: Attempt to recycle a mounted view.");
+//  [componentViewDescriptor.view prepareForRecycle];
+//
+//  recycledViews.push_back(componentViewDescriptor);
 }
 
 - (void)handleApplicationDidReceiveMemoryWarningNotification
