diff --git a/node_modules/react-native-web/src/exports/StyleSheet/compiler/index.js b/node_modules/react-native-web/src/exports/StyleSheet/compiler/index.js
index 639f532..524c13f 100644
--- a/node_modules/react-native-web/src/exports/StyleSheet/compiler/index.js
+++ b/node_modules/react-native-web/src/exports/StyleSheet/compiler/index.js
@@ -385,13 +385,14 @@ function createAtomicRules(identifier: string, property, value): Rules {
         finalValue = 'auto!important';
         if (value === 'box-only') {
           const block = createDeclarationBlock({ pointerEvents: 'none' });
-          rules.push(`${selector}>*${block}`);
+          rules.push(`${selector} *${block}`);
         }
       } else if (value === 'none' || value === 'box-none') {
         finalValue = 'none!important';
         if (value === 'box-none') {
           const block = createDeclarationBlock({ pointerEvents: 'auto' });
-          rules.push(`${selector}>*${block}`);
+          rules.push(`${selector} *${block}`);
         }
       }
       const block = createDeclarationBlock({ pointerEvents: finalValue });
