diff --git a/node_modules/react-native/ReactAndroid/publish.gradle b/node_modules/react-native/ReactAndroid/publish.gradle
index 32287a7..e425c3d 100644
--- a/node_modules/react-native/ReactAndroid/publish.gradle
+++ b/node_modules/react-native/ReactAndroid/publish.gradle
@@ -13,7 +13,9 @@ def signingKey = findProperty("SIGNING_KEY")
 def signingPwd = findProperty("SIGNING_PWD")
 
 def reactAndroidProjectDir = project(':packages:react-native:ReactAndroid').projectDir
-def mavenTempLocalUrl = "file:///tmp/maven-local"
+def mavenTempLocalUrl = "https://maven.pkg.github.com/mateuuszzzzz/App"
+
+def rnVersion = System.getenv("RN_VERSION")
 
 publishing {
     publications {
@@ -26,12 +28,14 @@ publishing {
                 }
             }
 
+            groupId = "com.example.react"
+
             // We populate the publishing version using the project version,
             // appending -SNAPSHOT if on nightly or prerelase.
             if (isSnapshot) {
-                version = this.version + "-SNAPSHOT"
+                version = rnVersion + "-SNAPSHOT"
             } else {
-                version = this.version
+                version = rnVersion
             }
 
             pom {
@@ -39,6 +43,12 @@ publishing {
                 description = "A framework for building native apps with React"
                 url = "https://github.com/facebook/react-native"
                 
+                properties = [
+                    "patchesHash": System.getenv("PATCHES_HASH"),
+                    "commitHash": System.getenv("COMMIT_HASH"),
+                    "pullRequestNumber": System.getenv("PULL_REQUEST_NUMBER")
+                ]
+
                 developers {
                     developer {
                         id = "facebook"
@@ -67,6 +77,10 @@ publishing {
         maven {
             name = "mavenTempLocal"
             url = mavenTempLocalUrl
+            credentials {
+                username = System.getenv("GH_PUBLISH_ACTOR")
+                password = System.getenv("GH_PUBLISH_TOKEN")
+            }
         }
     }
 
