diff --git a/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt b/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt
index 6e566fc..07ccb3c 100644
--- a/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt
+++ b/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt
@@ -117,12 +117,12 @@ class EdgeToEdgeReactViewGroup(private val reactContext: ThemedReactContext) : R
   }
 
   private fun goToEdgeToEdge(edgeToEdge: Boolean) {
-    reactContext.currentActivity?.let {
-      WindowCompat.setDecorFitsSystemWindows(
-        it.window,
-        !edgeToEdge,
-      )
-    }
+    // reactContext.currentActivity?.let {
+    //   WindowCompat.setDecorFitsSystemWindows(
+    //     it.window,
+    //     !edgeToEdge,
+    //   )
+    // }
   }
 
   private fun setupKeyboardCallbacks() {
@@ -169,16 +169,16 @@ class EdgeToEdgeReactViewGroup(private val reactContext: ThemedReactContext) : R
   // region State managers
   private fun enable() {
     this.goToEdgeToEdge(true)
-    this.setupWindowInsets()
+    // this.setupWindowInsets()
     this.setupKeyboardCallbacks()
-    modalAttachedWatcher.enable()
+    // modalAttachedWatcher.enable()
   }
 
   private fun disable() {
     this.goToEdgeToEdge(false)
-    this.setupWindowInsets()
+    // this.setupWindowInsets()
     this.removeKeyboardCallbacks()
-    modalAttachedWatcher.disable()
+    // modalAttachedWatcher.disable()
   }
   // endregion
 
@@ -206,7 +206,7 @@ class EdgeToEdgeReactViewGroup(private val reactContext: ThemedReactContext) : R
   fun forceStatusBarTranslucent(isStatusBarTranslucent: Boolean) {
     if (active && this.isStatusBarTranslucent != isStatusBarTranslucent) {
       this.isStatusBarTranslucent = isStatusBarTranslucent
-      this.setupWindowInsets()
+      // this.setupWindowInsets()
       this.requestApplyInsetsWhenAttached()
     }
   }
