diff --git a/node_modules/react-native-quick-sqlite/ios/QuickSQLite.mm b/node_modules/react-native-quick-sqlite/ios/QuickSQLite.mm
index c2857a6..7647115 100644
--- a/node_modules/react-native-quick-sqlite/ios/QuickSQLite.mm
+++ b/node_modules/react-native-quick-sqlite/ios/QuickSQLite.mm
@@ -31,9 +31,19 @@ RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(install) {
   auto &runtime = *jsiRuntime;
   auto callInvoker = bridge.jsCallInvoker;
 
-  // Get iOS app's document directory (to safely store database .sqlite3 file)
-  NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, true);
-  NSString *documentPath = [paths objectAtIndex:0];
+  NSString *appGroupID = @"group.com.chat.expensify.chat";
+  NSString *documentPath; 
+
+  if (appGroupID != nil) {
+    // Get the app groups container storage url
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSURL *storeUrl = [fileManager containerURLForSecurityApplicationGroupIdentifier:appGroupID];
+    documentPath = [storeUrl path];
+  } else {
+    // Get iOS app's document directory (to safely store database .sqlite3 file)
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, true);
+    documentPath = [paths objectAtIndex:0];
+  }
 
   osp::install(runtime, callInvoker,[documentPath UTF8String]);
   return @true;
