diff --git a/node_modules/react-native-release-profiler/lib/typescript/src/index.d.ts b/node_modules/react-native-release-profiler/lib/typescript/src/index.d.ts
index 32abf21..39b2de3 100644
--- a/node_modules/react-native-release-profiler/lib/typescript/src/index.d.ts
+++ b/node_modules/react-native-release-profiler/lib/typescript/src/index.d.ts
@@ -1,3 +1,3 @@
 export declare function startProfiling(): Boolean;
-export declare function stopProfiling(saveToDownloads?: boolean): Promise<string>;
+export declare function stopProfiling(saveToDownloads?: boolean, fileName?: string): Promise<string>;
 //# sourceMappingURL=index.d.ts.map
\ No newline at end of file
diff --git a/node_modules/react-native-release-profiler/src/index.tsx b/node_modules/react-native-release-profiler/src/index.tsx
index 9484c92..1b9f115 100644
--- a/node_modules/react-native-release-profiler/src/index.tsx
+++ b/node_modules/react-native-release-profiler/src/index.tsx
@@ -21,6 +21,6 @@ export function startProfiling(): Boolean {
   return ReleaseProfiler.startProfiling();
 }
 
-export function stopProfiling(saveToDownloads = false): Promise<string> {
+export function stopProfiling(saveToDownloads = false, _fileName = ""): Promise<string> {
   return ReleaseProfiler.stopProfiling(saveToDownloads);
 }
