diff --git a/node_modules/react-pdf/dist/esm/Document.js b/node_modules/react-pdf/dist/esm/Document.js
index ac7ca31..56bc766 100644
--- a/node_modules/react-pdf/dist/esm/Document.js
+++ b/node_modules/react-pdf/dist/esm/Document.js
@@ -240,6 +240,7 @@ const Document = forwardRef(function Document(_a, ref) {
             pdfDispatch({ type: 'REJECT', error });
         });
         return () => {
+            loadingTask._worker.destroy();
             loadingTask.destroy();
         };
     }
