diff --git a/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts b/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
index 4df2481..2be308f 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
@@ -1,4 +1,5 @@
 import { Platform } from 'react-native';
+import './globals.d.ts'
 
 export function isJest(): boolean {
   return !!process.env.JEST_WORKER_ID;
