diff --git a/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/modules/KeyboardControllerModuleImpl.kt b/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/modules/KeyboardControllerModuleImpl.kt
index 472ee09..1ef2459 100644
--- a/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/modules/KeyboardControllerModuleImpl.kt
+++ b/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/modules/KeyboardControllerModuleImpl.kt
@@ -24,17 +24,24 @@ class KeyboardControllerModuleImpl(
 
   fun dismiss() {
     val activity = mReactContext.currentActivity
-    val view: View? = FocusedInputHolder.get()
+    val view: View? = FocusedInputHolder.get() ?: activity?.currentFocus
 
     if (view != null) {
-      val imm = activity?.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
-      imm?.hideSoftInputFromWindow(view.windowToken, 0)
+      UiThreadUtil.runOnUiThread {
+        val imm = activity?.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
+        imm?.hideSoftInputFromWindow(view.windowToken, 0)
+        view.clearFocus()
+      }
     }
   }
 
   fun setFocusTo(direction: String) {
     if (direction == "current") {
-      return FocusedInputHolder.focus()
+      UiThreadUtil.runOnUiThread {
+        FocusedInputHolder.focus()
+      }
+
+      return
     }
 
     val view: View? = FocusedInputHolder.get()
