diff --git a/node_modules/react-native/ReactCommon/react/renderer/components/textinput/CMakeLists.txt b/node_modules/react-native/ReactCommon/react/renderer/components/textinput/CMakeLists.txt
index e8d12d5..8652636 100644
--- a/node_modules/react-native/ReactCommon/react/renderer/components/textinput/CMakeLists.txt
+++ b/node_modules/react-native/ReactCommon/react/renderer/components/textinput/CMakeLists.txt
@@ -15,7 +15,7 @@ add_compile_options(
         -DLOG_TAG=\"Fabric\")
 
 file(GLOB rrc_textinput_SRC CONFIGURE_DEPENDS platform/android/react/renderer/components/androidtextinput/*.cpp)
-add_library(rrc_textinput STATIC ${rrc_textinput_SRC})
+add_library(rrc_textinput SHARED ${rrc_textinput_SRC})
 
 target_include_directories(rrc_textinput
         PUBLIC
