diff --git a/node_modules/react-native-pager-view/src/PagerView.tsx b/node_modules/react-native-pager-view/src/PagerView.tsx
index 62faa74..e3fb020 100644
--- a/node_modules/react-native-pager-view/src/PagerView.tsx
+++ b/node_modules/react-native-pager-view/src/PagerView.tsx
@@ -20,17 +20,6 @@ import LEGACY_PagerViewNativeComponent, {
   Commands as LEGACY_PagerViewNativeCommands,
 } from './specs/LEGACY_PagerViewNativeComponent';
 
-// The Fabric component for PagerView uses a work around present also in ScrollView:
-// https://github.com/callstack/react-native-pager-view/blob/master/ios/Fabric/RNCPagerViewComponentView.mm#L362-L368
-// That workaround works only if we add these lines in to make sure that the RCTEventEmitter is registered properly
-// in the JS callable modules.
-// NOTE: This is a workaround as we would like to get rid of these lines below. But for the time being, as the cut date for
-// 0.74 approaches, we need to keep these lines.
-// As soon as we figure out how to move forward, we will provide guidance and/or submit a PR to fix this.
-if (Platform.OS === 'ios') {
-  require('react-native/Libraries/Renderer/shims/ReactNative'); // Force side effects to prevent T55744311
-}
-
 /**
  * Container that allows to flip left and right between child views. Each
  * child view of the `PagerView` will be treated as a separate page
