diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java
index 4c3387b..284720f 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/FabricEventDispatcher.java
@@ -38,10 +38,10 @@ public class FabricEventDispatcher implements EventDispatcher, LifecycleEventLis
 
   @Override
   public void dispatchEvent(Event event) {
-    event.dispatchModern(mReactEventEmitter);
     for (EventDispatcherListener listener : mListeners) {
       listener.onEventDispatch(event);
     }
+    event.dispatchModern(mReactEventEmitter);
 
     event.dispose();
     maybePostFrameCallbackFromNonUI();
