diff --git a/node_modules/react-native-config/android/dotenv.gradle b/node_modules/react-native-config/android/dotenv.gradle
index 2225375..48f94ca 100644
--- a/node_modules/react-native-config/android/dotenv.gradle
+++ b/node_modules/react-native-config/android/dotenv.gradle
@@ -41,7 +41,8 @@ def loadDotEnv(flavor = getCurrentFlavor()) {
     def env = [:]
     println("Reading env from: $envFile")
 
-    File f = new File("$project.rootDir/../$envFile");
+    def reactNativeProjectRoot = project.hasProperty('reactNativeProject') ? project.reactNativeProject : ".."
+    File f = new File("$project.rootDir/$reactNativeProjectRoot/$envFile");
     if (!f.exists()) {
         f = new File("$envFile");
     }
diff --git a/node_modules/react-native-config/ios/ReactNativeConfig/GeneratedDotEnv.m b/node_modules/react-native-config/ios/ReactNativeConfig/GeneratedDotEnv.m
index 59df625..04a2f3d 100644
--- a/node_modules/react-native-config/ios/ReactNativeConfig/GeneratedDotEnv.m
+++ b/node_modules/react-native-config/ios/ReactNativeConfig/GeneratedDotEnv.m
@@ -1 +1 @@
-  #define DOT_ENV @{ @"ENV":@"dev",@"API_URL":@"http://localhost" };
+  #define DOT_ENV @{  };
diff --git a/node_modules/react-native-config/ios/ReactNativeConfig/RNCConfigModule.h b/node_modules/react-native-config/ios/ReactNativeConfig/RNCConfigModule.h
index 5341aca..c550e69 100644
--- a/node_modules/react-native-config/ios/ReactNativeConfig/RNCConfigModule.h
+++ b/node_modules/react-native-config/ios/ReactNativeConfig/RNCConfigModule.h
@@ -6,10 +6,10 @@
 
 #if __has_include(<React/RCTBridgeModule.h>)
 #import <React/RCTBridgeModule.h>
-#elif __has_include("React/RCTBridgeModule.h")
-#import "React/RCTBridgeModule.h"
+#elif __has_include(<React/RCTBridgeModule.h>)
+#import <React/RCTBridgeModule.h>
 #else
-#import "RCTBridgeModule.h"
+#import <React/RCTBridgeModule.h>
 #endif
 
 @interface RNCConfigModule : NSObject <RCTBridgeModule>
diff --git a/node_modules/react-native-config/react-native-config.podspec b/node_modules/react-native-config/react-native-config.podspec
index 56bce4a..44603dd 100644
--- a/node_modules/react-native-config/react-native-config.podspec
+++ b/node_modules/react-native-config/react-native-config.podspec
@@ -4,6 +4,7 @@ require 'json'
 
 package = JSON.parse(File.read(File.join(__dir__, 'package.json')))
 
+REACT_NATIVE_DIR = ENV["REACT_NATIVE_DIR"] || ".."
 fabric_enabled = ENV['RCT_NEW_ARCH_ENABLED'] == '1'
 
 Pod::Spec.new do |s|
@@ -23,7 +24,7 @@ Pod::Spec.new do |s|
     name: 'Config codegen',
     script: %(
 set -ex
-HOST_PATH="$SRCROOT/../.."
+HOST_PATH="$SRCROOT/../#{REACT_NATIVE_DIR}"
 "${PODS_TARGET_SRCROOT}/ios/ReactNativeConfig/BuildDotenvConfig.rb" "$HOST_PATH" "${PODS_TARGET_SRCROOT}/ios/ReactNativeConfig"
 ),
     execution_position: :before_compile,
@@ -65,7 +66,7 @@ HOST_PATH="$SRCROOT/../.."
       name: 'Config codegen',
       script: %(
         set -ex
-        HOST_PATH="$SRCROOT/../.."
+        HOST_PATH="$SRCROOT/../#{REACT_NATIVE_DIR}"
         "${PODS_TARGET_SRCROOT}/ios/ReactNativeConfig/BuildDotenvConfig.rb" "$HOST_PATH" "${PODS_TARGET_SRCROOT}/ios/ReactNativeConfig"
         ),
       execution_position: :before_compile,
