diff --git a/node_modules/react-native-reanimated/RNReanimated.podspec b/node_modules/react-native-reanimated/RNReanimated.podspec
index a5aea4f..788a223 100644
--- a/node_modules/react-native-reanimated/RNReanimated.podspec
+++ b/node_modules/react-native-reanimated/RNReanimated.podspec
@@ -82,8 +82,8 @@ Pod::Spec.new do |s|
     ].join(' '),
     "FRAMEWORK_SEARCH_PATHS" => '"${PODS_CONFIGURATION_BUILD_DIR}/React-hermes"',
     "CLANG_CXX_LANGUAGE_STANDARD" => "c++17",
-    "GCC_PREPROCESSOR_DEFINITIONS[config=Debug]" => gcc_debug_definitions,
-    "GCC_PREPROCESSOR_DEFINITIONS[config=Release]" => '$(inherited) NDEBUG=1',
+    "GCC_PREPROCESSOR_DEFINITIONS[config=*Debug*]" => gcc_debug_definitions,
+    "GCC_PREPROCESSOR_DEFINITIONS[config=*Release*]" => '$(inherited) NDEBUG=1',
   }
   s.compiler_flags = "#{folly_flags} #{boost_compiler_flags}"
   s.xcconfig = {
