diff --git a/node_modules/react-native-web/dist/modules/AccessibilityUtil/propsToAccessibilityComponent.js b/node_modules/react-native-web/dist/modules/AccessibilityUtil/propsToAccessibilityComponent.js
index 9c9a533..7794181 100644
--- a/node_modules/react-native-web/dist/modules/AccessibilityUtil/propsToAccessibilityComponent.js
+++ b/node_modules/react-native-web/dist/modules/AccessibilityUtil/propsToAccessibilityComponent.js
@@ -27,7 +27,8 @@ var roleComponents = {
   navigation: 'nav',
   paragraph: 'p',
   region: 'section',
-  strong: 'strong'
+  strong: 'strong',
+  switch: 'button'
 };
 var emptyObject = {};
 var propsToAccessibilityComponent = function propsToAccessibilityComponent(props) {
diff --git a/node_modules/react-native-web/dist/modules/useResponderEvents/ResponderSystem.js b/node_modules/react-native-web/dist/modules/useResponderEvents/ResponderSystem.js
index 0aec2d6..a71aec2 100644
--- a/node_modules/react-native-web/dist/modules/useResponderEvents/ResponderSystem.js
+++ b/node_modules/react-native-web/dist/modules/useResponderEvents/ResponderSystem.js
@@ -133,7 +133,7 @@ to return true:wantsResponderID|                            |
 
 import createResponderEvent from './createResponderEvent';
 import { isCancelish, isEndish, isMoveish, isScroll, isSelectionChange, isStartish } from './ResponderEventTypes';
-import { getLowestCommonAncestor, getResponderPaths, hasTargetTouches, hasValidSelection, isPrimaryPointerDown, setResponderId } from './utils';
+import { getLowestCommonAncestor, getResponderPaths, hasTargetTouches, hasValidSelection, isPrimaryOrSecondaryPointerDown, setResponderId } from './utils';
 import { ResponderTouchHistoryStore } from './ResponderTouchHistoryStore';
 import canUseDOM from '../canUseDom';
 
@@ -225,7 +225,7 @@ function eventListener(domEvent) {
     }
     return;
   }
-  var isStartEvent = isStartish(eventType) && isPrimaryPointerDown(domEvent);
+  var isStartEvent = isStartish(eventType) && isPrimaryOrSecondaryPointerDown(domEvent);
   var isMoveEvent = isMoveish(eventType);
   var isEndEvent = isEndish(eventType);
   var isScrollEvent = isScroll(eventType);
diff --git a/node_modules/react-native-web/dist/modules/useResponderEvents/utils.js b/node_modules/react-native-web/dist/modules/useResponderEvents/utils.js
index 7382cdd..d88f6c0 100644
--- a/node_modules/react-native-web/dist/modules/useResponderEvents/utils.js
+++ b/node_modules/react-native-web/dist/modules/useResponderEvents/utils.js
@@ -148,14 +148,14 @@ export function hasValidSelection(domEvent) {
 /**
  * Events are only valid if the primary button was used without specific modifier keys.
  */
-export function isPrimaryPointerDown(domEvent) {
+export function isPrimaryOrSecondaryPointerDown(domEvent) {
   var altKey = domEvent.altKey,
     button = domEvent.button,
     buttons = domEvent.buttons,
     ctrlKey = domEvent.ctrlKey,
     type = domEvent.type;
   var isTouch = type === 'touchstart' || type === 'touchmove';
-  var isPrimaryMouseDown = type === 'mousedown' && (button === 0 || buttons === 1);
+  var isPrimaryMouseDown = type === 'mousedown' && (button === 0 || buttons === 1 || buttons === 2);
   var isPrimaryMouseMove = type === 'mousemove' && buttons === 1;
   var noModifiers = altKey === false && ctrlKey === false;
   if (isTouch || isPrimaryMouseDown && noModifiers || isPrimaryMouseMove && noModifiers) {
diff --git a/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js b/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
index 7f6c880..b05da08 100644
--- a/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
+++ b/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
@@ -78,14 +78,6 @@ function scrollEventThrottleOrDefault(scrollEventThrottle) {
 function windowSizeOrDefault(windowSize) {
   return windowSize !== null && windowSize !== void 0 ? windowSize : 21;
 }
-function findLastWhere(arr, predicate) {
-  for (var i = arr.length - 1; i >= 0; i--) {
-    if (predicate(arr[i])) {
-      return arr[i];
-    }
-  }
-  return null;
-}
 
 /**
  * Base implementation for the more convenient [`<FlatList>`](https://reactnative.dev/docs/flatlist)
@@ -1107,7 +1099,8 @@ class VirtualizedList extends StateSafePureComponent {
       _keylessItemComponentName = '';
       var spacerKey = this._getSpacerKey(!horizontal);
       var renderRegions = this.state.renderMask.enumerateRegions();
-      var lastSpacer = findLastWhere(renderRegions, r => r.isSpacer);
+      var lastRegion = renderRegions[renderRegions.length - 1];
+      var lastSpacer = lastRegion?.isSpacer ? lastRegion : null;
       for (var _iterator = _createForOfIteratorHelperLoose(renderRegions), _step; !(_step = _iterator()).done;) {
         var section = _step.value;
         if (section.isSpacer) {
