diff --git a/node_modules/rock/dist/src/lib/plugins/remoteCache.js b/node_modules/rock/dist/src/lib/plugins/remoteCache.js
index c90e1bd..9618a98 100644
--- a/node_modules/rock/dist/src/lib/plugins/remoteCache.js
+++ b/node_modules/rock/dist/src/lib/plugins/remoteCache.js
@@ -107,20 +107,24 @@ ${output
                 let uploadedArtifact;
                 const appFileName = path.basename(binaryPath);
                 const appName = appFileName.replace(/\.[^/.]+$/, '');
+                const randomHex = Math.floor(Math.random() * 0xffffffffff)
+                    .toString(16)
+                    .padStart(10, '0');
+                const temporaryUploadArtifactName = `${artifactName}-${randomHex}`;
                 const uploadContent = {
                     messagePrefix: 'build',
                     artifactName: undefined,
                 };
                 if (args.adHoc && isArtifactIPA) {
                     uploadContent.messagePrefix = 'IPA, index.html and manifest.plist';
-                    uploadContent.artifactName = `ad-hoc/${artifactName}/${appName}.ipa`;
+                    uploadContent.artifactName = `ad-hoc/${temporaryUploadArtifactName}/${appName}.ipa`;
                 }
                 else if (args.adHoc && isArtifactAPK) {
                     uploadContent.messagePrefix = 'APK, index.html';
-                    uploadContent.artifactName = `ad-hoc/${artifactName}/${appName}.apk`;
+                    uploadContent.artifactName = `ad-hoc/${temporaryUploadArtifactName}/${appName}.apk`;
                 }
                 const { name, url, getResponse } = await remoteBuildCache.upload({
-                    artifactName,
+                    artifactName: temporaryUploadArtifactName,
                     uploadArtifactName: uploadContent.artifactName,
                 });
                 const uploadMessage = `${uploadContent.messagePrefix} to ${color.bold(remoteBuildCache.name)}`;
@@ -134,13 +138,13 @@ ${output
                 if (args.adHoc && isArtifactIPA) {
                     const { version, bundleIdentifier } = await getInfoPlistFromIpa(binaryPath);
                     const { url: urlIndexHtml, getResponse: getResponseIndexHtml } = await remoteBuildCache.upload({
-                        artifactName,
-                        uploadArtifactName: `ad-hoc/${artifactName}/index.html`,
+                        artifactName: temporaryUploadArtifactName,
+                        uploadArtifactName: `ad-hoc/${temporaryUploadArtifactName}/index.html`,
                     });
                     getResponseIndexHtml(Buffer.from(templateIndexHtmlIOS({ appName, bundleIdentifier, version })), 'text/html');
                     const { getResponse: getResponseManifestPlist } = await remoteBuildCache.upload({
-                        artifactName,
-                        uploadArtifactName: `ad-hoc/${artifactName}/manifest.plist`,
+                        artifactName: temporaryUploadArtifactName,
+                        uploadArtifactName: `ad-hoc/${temporaryUploadArtifactName}/manifest.plist`,
                     });
                     getResponseManifestPlist((baseUrl) => Buffer.from(templateManifestPlist({
                         appName,
@@ -157,8 +161,8 @@ ${output
                 if (args.adHoc && isArtifactAPK) {
                     const { version, packageName } = await getManifestFromApk(binaryPath);
                     const { url: urlIndexHtml, getResponse: getResponseIndexHtml } = await remoteBuildCache.upload({
-                        artifactName,
-                        uploadArtifactName: `ad-hoc/${artifactName}/index.html`,
+                        artifactName: temporaryUploadArtifactName,
+                        uploadArtifactName: `ad-hoc/${temporaryUploadArtifactName}/index.html`,
                     });
                     getResponseIndexHtml(Buffer.from(templateIndexHtmlAndroid({ appName, packageName, version })), 'text/html');
                     // For ad-hoc distribution, we want the url to point to the index.html for easier installation
