diff --git a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js
index 9885b25..ae4fe46 100644
--- a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js
+++ b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js
@@ -396,6 +396,7 @@ export class CardStack extends React.Component {
               extrapolate: 'clamp'
             }) : STATE_TRANSITIONING_OR_BELOW_TOP;
           }
+          const shouldNotDetachScreen = route?.dontDetachScreen && isScreenActive !== STATE_ON_TOP;
           const {
             headerShown = true,
             headerTransparent,
@@ -416,7 +417,7 @@ export class CardStack extends React.Component {
           return /*#__PURE__*/_jsx(MaybeScreen, {
             style: [StyleSheet.absoluteFill],
             enabled: detachInactiveScreens,
-            active: isScreenActive,
+            active: shouldNotDetachScreen ? STATE_TRANSITIONING_OR_BELOW_TOP : isScreenActive,
             freezeOnBlur: freezeOnBlur,
             shouldFreeze: isScreenActive === STATE_INACTIVE && !isPreloaded,
             homeIndicatorHidden: autoHideHomeIndicator,
@@ -451,7 +452,7 @@ export class CardStack extends React.Component {
               onTransitionStart: onTransitionStart,
               onTransitionEnd: onTransitionEnd,
               isNextScreenTransparent: isNextScreenTransparent,
-              detachCurrentScreen: detachCurrentScreen,
+              detachCurrentScreen: shouldNotDetachScreen ? false : detachCurrentScreen,
               preloaded: isPreloaded
             })
           }, route.key);
diff --git a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardContainer.js b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardContainer.js
index c2240e8..63b95e9 100644
--- a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardContainer.js
+++ b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardContainer.js
@@ -3,7 +3,7 @@
 import { getHeaderTitle, HeaderBackContext, HeaderHeightContext, HeaderShownContext } from '@react-navigation/elements';
 import { useLinkBuilder, useLocale, useTheme } from '@react-navigation/native';
 import * as React from 'react';
-import { StyleSheet, View } from 'react-native';
+import { Platform, StyleSheet, View } from 'react-native';
 import { ModalPresentationContext } from "../../utils/ModalPresentationContext.js";
 import { useKeyboardManager } from "../../utils/useKeyboardManager.js";
 import { Card } from "./Card.js";
@@ -229,7 +229,8 @@ function CardContainerInner({
       display:
       // Hide unfocused screens when animation isn't enabled
       // This is also necessary for a11y on web
-      animation === 'none' && isNextScreenTransparent === false && detachCurrentScreen !== false && !focused ? 'none' : 'flex'
+      animation === 'none' && isNextScreenTransparent === false && detachCurrentScreen !== false && !focused ? 'none' : 'flex',
+      zIndex: Platform.OS === 'web' ? 'auto' : undefined,
     }, StyleSheet.absoluteFill],
     children: /*#__PURE__*/_jsx(View, {
       style: styles.container,
