diff --git a/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js b/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js
index 1183c51..303635c 100644
--- a/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js
+++ b/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js
@@ -160,7 +160,7 @@ export function NativeBottomTabView({
     ios: PlatformColor('label'),
     default: colors.text
   });
-  const activeIndicatorColor = currentOptions?.tabBarActiveIndicatorColor ?? typeof activeTintColor === 'string' ? Color(activeTintColor)?.alpha(0.1).string() : undefined;
+  const activeIndicatorColor = currentOptions?.tabBarActiveIndicatorColor ?? (typeof activeTintColor === 'string' ? Color(activeTintColor)?.alpha(0.1).string() : undefined);
   const onTransitionStart = ({
     closing,
     route
@@ -253,11 +253,11 @@ export function NativeBottomTabView({
           tabBarItemBadgeBackgroundColor: badgeBackgroundColor
         };
         const normalTabItemAppearance = {
-          tabBarItemTitleFontColor: inactiveTintColor ?? fontColor,
+          tabBarItemTitleFontColor: currentOptions.tabBarInactiveLabelColor ?? inactiveTintColor ?? fontColor,
           tabBarItemIconColor: inactiveTintColor
         };
         const selectedTabItemAppearance = {
-          tabBarItemTitleFontColor: activeTintColor,
+          tabBarItemTitleFontColor: currentOptions.tabBarActiveLabelColor ?? activeTintColor,
           tabBarItemIconColor: activeTintColor
         };
         const icon = typeof tabBarIcon === 'function' ? getPlatformIcon(tabBarIcon({
diff --git a/node_modules/@react-navigation/bottom-tabs/lib/typescript/src/unstable/types.d.ts b/node_modules/@react-navigation/bottom-tabs/lib/typescript/src/unstable/types.d.ts
index 5a502c2..a071119 100644
--- a/node_modules/@react-navigation/bottom-tabs/lib/typescript/src/unstable/types.d.ts
+++ b/node_modules/@react-navigation/bottom-tabs/lib/typescript/src/unstable/types.d.ts
@@ -156,6 +156,18 @@ export type NativeBottomTabNavigationOptions = NativeHeaderOptions & {
      * @platform android
      */
     tabBarInactiveTintColor?: ColorValue;
+    /**
+     * Color for the label in the active tab, overriding `tabBarActiveTintColor` for the label only.
+     *
+     * @platform android
+     */
+    tabBarActiveLabelColor?: ColorValue;
+    /**
+     * Color for the label in the inactive tabs, overriding `tabBarInactiveTintColor` for the label only.
+     *
+     * @platform android
+     */
+    tabBarInactiveLabelColor?: ColorValue;
     /**
      * Background color of the active indicator.
      *
diff --git a/node_modules/@react-navigation/bottom-tabs/src/unstable/NativeBottomTabView.native.tsx b/node_modules/@react-navigation/bottom-tabs/src/unstable/NativeBottomTabView.native.tsx
index 287dea3..0f06b2b 100644
--- a/node_modules/@react-navigation/bottom-tabs/src/unstable/NativeBottomTabView.native.tsx
+++ b/node_modules/@react-navigation/bottom-tabs/src/unstable/NativeBottomTabView.native.tsx
@@ -262,10 +262,10 @@ export function NativeBottomTabView({ state, navigation, descriptors }: Props) {
     });
 
   const activeIndicatorColor =
-    (currentOptions?.tabBarActiveIndicatorColor ??
-    typeof activeTintColor === 'string')
+    currentOptions?.tabBarActiveIndicatorColor ??
+    (typeof activeTintColor === 'string'
       ? Color(activeTintColor)?.alpha(0.1).string()
-      : undefined;
+      : undefined);
 
   const onTransitionStart = ({
     closing,
@@ -387,13 +387,17 @@ export function NativeBottomTabView({ state, navigation, descriptors }: Props) {
 
           const normalTabItemAppearance: TabsScreenItemStateAppearanceAndroid =
             {
-              tabBarItemTitleFontColor: inactiveTintColor ?? fontColor,
+              tabBarItemTitleFontColor:
+                currentOptions.tabBarInactiveLabelColor ??
+                inactiveTintColor ??
+                fontColor,
               tabBarItemIconColor: inactiveTintColor,
             };
 
           const selectedTabItemAppearance: TabsScreenItemStateAppearanceAndroid =
             {
-              tabBarItemTitleFontColor: activeTintColor,
+              tabBarItemTitleFontColor:
+                currentOptions.tabBarActiveLabelColor ?? activeTintColor,
               tabBarItemIconColor: activeTintColor,
             };
 
diff --git a/node_modules/@react-navigation/bottom-tabs/src/unstable/types.tsx b/node_modules/@react-navigation/bottom-tabs/src/unstable/types.tsx
index a2471f5..d9df793 100644
--- a/node_modules/@react-navigation/bottom-tabs/src/unstable/types.tsx
+++ b/node_modules/@react-navigation/bottom-tabs/src/unstable/types.tsx
@@ -200,6 +200,20 @@ export type NativeBottomTabNavigationOptions = NativeHeaderOptions & {
    */
   tabBarInactiveTintColor?: ColorValue;
 
+  /**
+   * Color for the label in the active tab, overriding `tabBarActiveTintColor` for the label only.
+   *
+   * @platform android
+   */
+  tabBarActiveLabelColor?: ColorValue;
+
+  /**
+   * Color for the label in the inactive tabs, overriding `tabBarInactiveTintColor` for the label only.
+   *
+   * @platform android
+   */
+  tabBarInactiveLabelColor?: ColorValue;
+
   /**
    * Background color of the active indicator.
    *
