diff --git a/node_modules/@react-navigation/core/lib/module/getPathFromState.js b/node_modules/@react-navigation/core/lib/module/getPathFromState.js
index 0d1ca6a..5c9d316 100644
--- a/node_modules/@react-navigation/core/lib/module/getPathFromState.js
+++ b/node_modules/@react-navigation/core/lib/module/getPathFromState.js
@@ -156,7 +156,7 @@ export function getPathFromState(state, options) {
 
           // Valid characters according to
           // https://datatracker.ietf.org/doc/html/rfc3986#section-3.3 (see pchar definition)
-          return String(value).replace(/[^A-Za-z0-9\-._~!$&'()*+,;=:@]/g, char => encodeURIComponent(char));
+          return Array.from(value).map(char => /[^A-Za-z0-9\-._~!$&'()*+,;=:@]/g.test(char) ? encodeURIComponent(char) : char).join('');
         }
         return encodeURIComponent(segment);
       }).join('/');
