diff --git a/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js b/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js
index 1393e4c..2d06ae9 100644
--- a/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js
+++ b/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js
@@ -537,14 +537,15 @@ export function useNavigationBuilder(createRouter, options, convertCustomScreenO
     // @ts-expect-error: this should have both core and custom events, but too much work right now
     emitter
   }, convertCustomScreenOptions);
+  const validState = state?.preloadedRoutes?.some(preloadedRoute => !state.routeNames.includes(preloadedRoute.name)) ? {...state, preloadedRoutes: []} : state;
   useCurrentRender({
-    state,
+    state: validState,
     navigation,
     descriptors
   });
   const NavigationContent = useComponent(children => {
     const element = layout != null ? layout({
-      state,
+      state: validState,
       descriptors,
       navigation,
       children
@@ -552,7 +553,7 @@ export function useNavigationBuilder(createRouter, options, convertCustomScreenO
     return /*#__PURE__*/_jsx(NavigationHelpersContext.Provider, {
       value: navigation,
       children: /*#__PURE__*/_jsx(NavigationStateListenerProvider, {
-        state: state,
+        state: validState,
         children: /*#__PURE__*/_jsx(PreventRemoveProvider, {
           children: element
         })
@@ -560,7 +561,7 @@ export function useNavigationBuilder(createRouter, options, convertCustomScreenO
     });
   });
   return {
-    state,
+    state: validState,
     navigation,
     describe,
     descriptors,
diff --git a/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx b/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx
index 835bc07..6e8481e 100644
--- a/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx
+++ b/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx
@@ -910,9 +910,11 @@ export function useNavigationBuilder<
     // @ts-expect-error: this should have both core and custom events, but too much work right now
     emitter,
   });
+
+  const validState = state?.preloadedRoutes?.some(preloadedRoute => !state.routeNames.includes(preloadedRoute.name)) ? {...state, preloadedRoutes: []} : state;

   useCurrentRender({
-    state,
+    state: validState,
     navigation,
     descriptors,
   });
@@ -921,7 +923,7 @@ export function useNavigationBuilder<
     const element =
       layout != null
         ? layout({
-            state,
+            state: validState,
             descriptors,
             navigation,
             children,
@@ -930,7 +932,7 @@ export function useNavigationBuilder<

     return (
       <NavigationHelpersContext.Provider value={navigation}>
-        <NavigationStateListenerProvider state={state}>
+        <NavigationStateListenerProvider state={validState}>
           <PreventRemoveProvider>{element}</PreventRemoveProvider>
         </NavigationStateListenerProvider>
       </NavigationHelpersContext.Provider>
@@ -938,7 +940,7 @@ export function useNavigationBuilder<
   });

   return {
-    state,
+    state: validState,
     navigation,
     describe,
     descriptors,
