diff --git a/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js b/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js
index 0000000..c93a4f5 100644
--- a/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js
+++ b/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js
@@ -575,15 +575,16 @@ export function useNavigationBuilder(createRouter, options, convertCustomScreenO
     router,
     // @ts-expect-error: this should have both core and custom events, but too much work right now
     emitter
   }, convertCustomScreenOptions);
+  const validState = state?.preloadedRoutes?.some(preloadedRoute => !state.routeNames.includes(preloadedRoute.name)) ? {...state, preloadedRoutes: []} : state;
   useCurrentRender({
-    state,
+    state: validState,
     navigation,
     descriptors
   });
   const NavigationContent = useComponent(children => {
     const element = layout != null ? layout({
-      state,
+      state: validState,
       descriptors,
       navigation,
       children
@@ -605,7 +606,7 @@ export function useNavigationBuilder(createRouter, options, convertCustomScreenO
     });
   });
   return {
-    state,
+    state: validState,
     navigation,
     describe,
     descriptors,
diff --git a/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx b/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx
index 0000000..3bab87a 100644
--- a/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx
+++ b/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx
@@ -985,10 +985,14 @@ export function useNavigationBuilder<
     router,
     // @ts-expect-error: this should have both core and custom events, but too much work right now
     emitter,
   }, convertCustomScreenOptions);
+
+  const validState = state?.preloadedRoutes?.some((preloadedRoute) => !state.routeNames.includes(preloadedRoute.name))
+    ? {...state, preloadedRoutes: []}
+    : state;
 
   useCurrentRender({
-    state,
+    state: validState,
     navigation,
     descriptors,
   });
@@ -997,7 +1001,7 @@ export function useNavigationBuilder<
     const element =
       layout != null
         ? layout({
-            state,
+            state: validState,
             descriptors,
             navigation,
             children,
@@ -1020,7 +1024,7 @@ export function useNavigationBuilder<
   });

   return {
-    state,
+    state: validState,
     navigation,
     describe,
     descriptors,
