diff --git a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardA11yWrapper.js b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardA11yWrapper.js
index 44c8f10..39f5065 100644
--- a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardA11yWrapper.js
+++ b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardA11yWrapper.js
@@ -30,7 +30,8 @@ export const CardA11yWrapper = /*#__PURE__*/React.forwardRef(({
       // Hide unfocused screens when animation isn't enabled
       // This is also necessary for a11y on web
       // @ts-expect-error visibility is only available on web
-      visibility: isHidden ? 'hidden' : 'visible'
+      visibility: isHidden ? 'hidden' : 'visible',
+      zIndex: Platform.OS === 'web' ? 'auto' : undefined
     }]
     // Make sure this view is not removed on the new architecture, as it causes focus loss during navigation on Android.
     // This can happen when the view flattening results in different trees - due to `overflow` style changing in a parent.
diff --git a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js
index a1b2c3d..2e9bb70 100644
--- a/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js
+++ b/node_modules/@react-navigation/stack/lib/module/views/Stack/CardStack.js
@@ -428,10 +428,11 @@ export class CardStack extends React.Component {
           const isNextScreenTransparent = scenes[index + 1]?.descriptor.options.presentation === 'transparentModal';
           const detachCurrentScreen = scenes[index + 1]?.descriptor.options.detachPreviousScreen !== false;
           const activityState = isPreloaded ? STATE_INACTIVE : activeStates[index];
+          const shouldNotDetachScreen = route?.dontDetachScreen && activityState !== STATE_ON_TOP;
           return /*#__PURE__*/_jsx(MaybeScreen, {
             style: [StyleSheet.absoluteFill],
             enabled: detachInactiveScreens,
-            active: activityState,
+            active: shouldNotDetachScreen ? STATE_TRANSITIONING_OR_BELOW_TOP : activityState,
             freezeOnBlur: freezeOnBlur,
             shouldFreeze: activityState === STATE_INACTIVE && !isPreloaded,
             homeIndicatorHidden: autoHideHomeIndicator,
@@ -466,7 +467,7 @@ export class CardStack extends React.Component {
               onTransitionStart: onTransitionStart,
               onTransitionEnd: onTransitionEnd,
               isNextScreenTransparent: isNextScreenTransparent,
-              detachCurrentScreen: detachCurrentScreen,
+              detachCurrentScreen: shouldNotDetachScreen ? false : detachCurrentScreen,
               preloaded: isPreloaded
             })
           }, route.key);
