diff --git a/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js b/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js
index d7028e1..769b798 100644
--- a/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js
+++ b/node_modules/@react-navigation/core/lib/module/useNavigationBuilder.js
@@ -473,14 +473,15 @@ export function useNavigationBuilder(createRouter, options, convertCustomScreenO
     // @ts-expect-error: this should have both core and custom events, but too much work right now
     emitter
   }, convertCustomScreenOptions);
+  const validState = state?.preloadedRoutes?.some(preloadedRoute => !state.routeNames.includes(preloadedRoute.name)) ? {...state, preloadedRoutes: []} : state;
   useCurrentRender({
-    state,
+    state: validState,
     navigation,
     descriptors
   });
   const NavigationContent = useComponent(children => {
     const element = layout != null ? layout({
-      state,
+      state: validState,
       descriptors,
       navigation,
       children
@@ -493,7 +494,7 @@ export function useNavigationBuilder(createRouter, options, convertCustomScreenO
     });
   });
   return {
-    state,
+    state: validState,
     navigation,
     describe,
     descriptors,
diff --git a/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx b/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx
index 835bc07..6e8481e 100644
--- a/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx
+++ b/node_modules/@react-navigation/core/src/useNavigationBuilder.tsx
@@ -821,8 +821,10 @@ export function useNavigationBuilder<
     convertCustomScreenOptions
   );
 
+  const validState = state?.preloadedRoutes?.some(preloadedRoute => !state.routeNames.includes(preloadedRoute.name)) ? {...state, preloadedRoutes: []} : state;
+
   useCurrentRender({
-    state,
+    state: validState,
     navigation,
     descriptors,
   });
@@ -831,7 +833,7 @@ export function useNavigationBuilder<
     const element =
       layout != null
         ? layout({
-            state,
+            state: validState,
             descriptors,
             navigation,
             children,
@@ -846,7 +848,7 @@ export function useNavigationBuilder<
   });
 
   return {
-    state,
+    state: validState,
     navigation,
     describe,
     descriptors,
