diff --git a/node_modules/react-native-modal/dist/modal.d.ts b/node_modules/react-native-modal/dist/modal.d.ts
index bd6419e..029762c 100644
--- a/node_modules/react-native-modal/dist/modal.d.ts
+++ b/node_modules/react-native-modal/dist/modal.d.ts
@@ -46,6 +46,7 @@ declare const defaultProps: {
     scrollOffsetMax: number;
     scrollHorizontal: boolean;
     statusBarTranslucent: boolean;
+    navigationBarTranslucent: boolean;
     supportedOrientations: ("landscape" | "portrait" | "portrait-upside-down" | "landscape-left" | "landscape-right")[];
 };
 export declare type ModalProps = ViewProps & {
@@ -137,6 +138,7 @@ export declare class ReactNativeModal extends React.Component<ModalProps, State>
         scrollOffsetMax: number;
         scrollHorizontal: boolean;
         statusBarTranslucent: boolean;
+        navigationBarTranslucent: boolean;
         supportedOrientations: ("landscape" | "portrait" | "portrait-upside-down" | "landscape-left" | "landscape-right")[];
     };
     state: State;
diff --git a/node_modules/react-native-modal/dist/modal.js b/node_modules/react-native-modal/dist/modal.js
index 2e70c0c..7a39705 100644
--- a/node_modules/react-native-modal/dist/modal.js
+++ b/node_modules/react-native-modal/dist/modal.js
@@ -39,6 +39,7 @@ const defaultProps = {
     scrollOffsetMax: 0,
     scrollHorizontal: false,
     statusBarTranslucent: false,
+    navigationBarTranslucent: false,
     supportedOrientations: ['portrait', 'landscape'],
 };
 const extractAnimationFromProps = (props) => ({
@@ -69,6 +70,7 @@ export class ReactNativeModal extends React.Component {
         this.interactionHandle = null;
         this.getDeviceHeight = () => this.props.deviceHeight || this.state.deviceHeight;
         this.getDeviceWidth = () => this.props.deviceWidth || this.state.deviceWidth;
+        this.backHandlerEventSubscription = null;
         this.onBackButtonPress = () => {
             if (this.props.onBackButtonPress && this.props.isVisible) {
                 this.props.onBackButtonPress();
@@ -467,13 +469,15 @@ export class ReactNativeModal extends React.Component {
             document?.body?.addEventListener?.('keyup', this.handleEscape, true);
             return;
         }
-        BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
+        this.backHandlerEventSubscription = BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
     }
     componentWillUnmount() {
         if (Platform.OS === 'web') {
             document?.body?.removeEventListener?.('keyup', this.handleEscape, true);
         } else {
-            BackHandler.removeEventListener('hardwareBackPress', this.onBackButtonPress);
+            if (this.backHandlerEventSubscription) {
+                this.backHandlerEventSubscription.remove();
+            }
         }
         if (this.didUpdateDimensionsEmitter) {
             this.didUpdateDimensionsEmitter.remove();
