diff --git a/node_modules/eslint-config-airbnb-typescript/lib/shared.js b/node_modules/eslint-config-airbnb-typescript/lib/shared.js
index 306550b..574bc49 100644
--- a/node_modules/eslint-config-airbnb-typescript/lib/shared.js
+++ b/node_modules/eslint-config-airbnb-typescript/lib/shared.js
@@ -1,10 +1,10 @@
 /* eslint-disable import/no-extraneous-dependencies */
-const { rules: baseBestPracticesRules } = require('eslint-config-airbnb-base/rules/best-practices');
-const { rules: baseErrorsRules } = require('eslint-config-airbnb-base/rules/errors');
-const { rules: baseES6Rules } = require('eslint-config-airbnb-base/rules/es6');
-const { rules: baseImportsRules } = require('eslint-config-airbnb-base/rules/imports');
-const { rules: baseStyleRules } = require('eslint-config-airbnb-base/rules/style');
-const { rules: baseVariablesRules } = require('eslint-config-airbnb-base/rules/variables');
+const { rules: baseBestPracticesRules } = require('eslint-config-expensify/rules/base/best-practices').default[0];
+const { rules: baseErrorsRules } = require('eslint-config-expensify/rules/base/errors').default[0];
+const { rules: baseES6Rules } = require('eslint-config-expensify/rules/base/es6').default[0];
+const { rules: baseImportsRules } = require('eslint-config-expensify/rules/base/imports').default[0];
+const { rules: baseStyleRules } = require('eslint-config-expensify/rules/base/style').default[0];
+const { rules: baseVariablesRules } = require('eslint-config-expensify/rules/base/variables').default[0];
 /* eslint-enable import/no-extraneous-dependencies */
 
 module.exports = {
