diff --git a/node_modules/expo-image/android/src/main/java/expo/modules/image/ResourceIdHelper.kt b/node_modules/expo-image/android/src/main/java/expo/modules/image/ResourceIdHelper.kt
index b1085fe..aac94db 100644
--- a/node_modules/expo-image/android/src/main/java/expo/modules/image/ResourceIdHelper.kt
+++ b/node_modules/expo-image/android/src/main/java/expo/modules/image/ResourceIdHelper.kt
@@ -32,7 +32,7 @@ object ResourceIdHelper {
   }
 
   fun getResourceUri(context: Context, name: String): Uri? {
-    val drawableUri = ResourceDrawableIdHelper.getResourceDrawableUri(context, name)
+    val drawableUri = ResourceDrawableIdHelper.instance.getResourceDrawableUri(context, name)
     if (drawableUri != Uri.EMPTY) {
       return drawableUri
     }
