diff --git a/node_modules/expo/src/winter/FormData.ts b/node_modules/expo/src/winter/FormData.ts
index 8f86d9c..2df9be4 100644
--- a/node_modules/expo/src/winter/FormData.ts
+++ b/node_modules/expo/src/winter/FormData.ts
@@ -74,7 +74,7 @@ function normalizeArgs(
 ): [string, File | string] {
   if (value instanceof Blob) {
     // @ts-expect-error: `Blob.data.blobId` is react-native's proprietary property.
-    value.name = blobFilename ?? blob.name ?? 'blob';
+    value.name = blobFilename ?? value.name ?? 'blob';
   } else if (typeof value !== 'object') {
     value = String(value);
   }
