diff --git a/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm b/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm
index a4be628..d8f304e 100644
--- a/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm
+++ b/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm
@@ -40,6 +40,11 @@ - (instancetype) initWithFrame:(CGRect)frame
 
 #pragma mark - RCTComponentViewProtocol
 
++ (BOOL)shouldBeRecycled
+{
+    return NO;
+}
+
 + (ComponentDescriptorProvider)componentDescriptorProvider
 {
     return concreteComponentDescriptorProvider<LottieAnimationViewComponentDescriptor>();
