
diff --git a/node_modules/expo-video/build/VideoPlayer.web.js b/node_modules/expo-video/build/VideoPlayer.web.js
index 2c7224a..16df18b 100644
--- a/node_modules/expo-video/build/VideoPlayer.web.js
+++ b/node_modules/expo-video/build/VideoPlayer.web.js
@@ -305,14 +305,16 @@ export default class VideoPlayerWeb extends globalThis.expo.SharedObject {
                 mountedVideo.play();
             });
         };
-        video.onpause = () => {
+        video.onpause = (e) => {
             this._emitOnce(video, 'playingChange', {
                 isPlaying: false,
                 oldIsPlaying: this.playing,
             });
             this.playing = false;
             this._mountedVideos.forEach((mountedVideo) => {
-                mountedVideo.pause();
+                if (e.target !== mountedVideo) {
+                    mountedVideo.pause();
+                }
             });
         };
         video.onvolumechange = () => {
