diff --git a/node_modules/react-native-onyx/dist/types.d.ts b/node_modules/react-native-onyx/dist/types.d.ts
index cb91605..f974c46 100644
--- a/node_modules/react-native-onyx/dist/types.d.ts
+++ b/node_modules/react-native-onyx/dist/types.d.ts
@@ -1,4 +1,4 @@
-import type { Merge } from 'type-fest';
+import type { Merge, ReadonlyDeep } from 'type-fest';
 import type OnyxUtils from './OnyxUtils';
 import type { OnyxMethod } from './OnyxUtils';
 import type { FastMergeReplaceNullPatch } from './utils';
@@ -219,7 +219,7 @@ type OnyxCollectionInputValue<TOnyxValue> = OnyxInputValue<Record<string, TOnyxV
  * Setting a key to `null` will remove the key from the store.
  * `undefined` is not allowed for setting values, because it will have no effect on the data.
  */
-type OnyxInput<TKey extends OnyxKey> = OnyxInputValue<NullishDeep<KeyValueMapping[TKey]>>;
+type OnyxInput<TKey extends OnyxKey> = ReadonlyDeep<OnyxInputValue<NullishDeep<KeyValueMapping[TKey]>>>;
 /**
  * Represents a mapping object where each `OnyxKey` maps to either a value of its corresponding type in `KeyValueMapping` or `null`.
  *
