diff --git a/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt b/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt
index 3689418..a236663 100644
--- a/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt
+++ b/node_modules/react-native-keyboard-controller/android/src/main/java/com/reactnativekeyboardcontroller/views/EdgeToEdgeReactViewGroup.kt
@@ -118,13 +118,21 @@ class EdgeToEdgeReactViewGroup(
           if (!shouldApplyBottomPadding) 0 else insets.getInsets(WindowInsetsCompat.Type.ime()).bottom
 
         params.setMargins(
-          navBarInsets.left,
+          if (this.isNavigationBarTranslucent) {
+            0
+          } else {
+            navBarInsets.left
+          },
           if (this.isStatusBarTranslucent) {
             0
           } else {
             systemBarInsets.top
           },
-          navBarInsets.right,
+          if (this.isNavigationBarTranslucent) {
+            0
+          } else {
+            navBarInsets.right
+          },
           if (this.isNavigationBarTranslucent) {
             keyboardInsets
           } else {
