diff --git a/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt b/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt
index 49a53af..98b690d 100644
--- a/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt
+++ b/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt
@@ -57,6 +57,7 @@ class AudioModule : Module() {
   private var focusAcquired = false
   private var interruptionMode: InterruptionMode? = null
   private var allowsBackgroundRecording = false
+  private var playsInSilentMode = true
 
   private var audioFocusRequest: AudioFocusRequest? = null
   private val audioFocusChangeListener = AudioManager.OnAudioFocusChangeListener { focusChange ->
@@ -184,6 +185,7 @@ class AudioModule : Module() {
     AsyncFunction("setAudioModeAsync") { mode: AudioMode ->
       staysActiveInBackground = mode.shouldPlayInBackground
       interruptionMode = mode.interruptionMode
+      playsInSilentMode = mode.playsInSilentMode
       updatePlaySoundThroughEarpiece(mode.shouldRouteThroughEarpiece ?: false)
       allowsBackgroundRecording = mode.allowsBackgroundRecording
 
@@ -390,6 +392,9 @@ class AudioModule : Module() {
           Log.e(TAG, "Audio has been disabled. Re-enable to start playing")
           return@Function
         }
+        if (!playsInSilentMode && audioManager.getRingerMode() != AudioManager.RINGER_MODE_NORMAL) {
+          return@Function
+        }
         runOnMain {
           if (!focusAcquired) {
             requestAudioFocus()
diff --git a/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioRecords.kt b/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioRecords.kt
index 092af43..b969e86 100644
--- a/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioRecords.kt
+++ b/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioRecords.kt
@@ -16,7 +16,8 @@ class AudioMode(
   @Field val shouldPlayInBackground: Boolean = false,
   @Field val shouldRouteThroughEarpiece: Boolean?,
   @Field val interruptionMode: InterruptionMode?,
-  @Field val allowsBackgroundRecording: Boolean = false
+  @Field val allowsBackgroundRecording: Boolean = false,
+  @Field val playsInSilentMode: Boolean = true
 ) : Record
 
 // Data class because we want `equals`
