diff --git a/node_modules/react-native/Libraries/AppDelegate/RCTAppSetupUtils.mm b/node_modules/react-native/Libraries/AppDelegate/RCTAppSetupUtils.mm
index fabfff3..cefa0fa 100644
--- a/node_modules/react-native/Libraries/AppDelegate/RCTAppSetupUtils.mm
+++ b/node_modules/react-native/Libraries/AppDelegate/RCTAppSetupUtils.mm
@@ -103,11 +103,16 @@ void RCTAppSetupPrepareApp(UIApplication *application, BOOL turboModuleEnabled)
         initWithHandlersProvider:^NSArray<id<RCTURLRequestHandler>> *(RCTModuleRegistry *moduleRegistry) {
           NSArray *URLRequestHandlerModules =
               extractModuleConformingToProtocol(moduleRegistry, @protocol(RCTURLRequestHandler));
-          return [@[
+          NSMutableArray<id<RCTURLRequestHandler>> *defaultHandlers = [NSMutableArray arrayWithObjects:
             [RCTHTTPRequestHandler new],
             [RCTDataRequestHandler new],
             [RCTFileRequestHandler new],
-            [moduleRegistry moduleForName:"BlobModule"],
-          ] arrayByAddingObjectsFromArray:URLRequestHandlerModules];
+            nil
+          ];
+          id blobModule = [moduleRegistry moduleForName:"BlobModule"];
+          if (blobModule != nil) {
+            [defaultHandlers addObject:blobModule];
+          }
+          return [defaultHandlers arrayByAddingObjectsFromArray:URLRequestHandlerModules];
         }];
   }
