diff --git a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
--- a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
@@ -1314,8 +1314,9 @@
                       "mixed", "a checkbox, radio button, or other widget which is both checked and unchecked")];
   }
-  if (accessibilityState.expanded.value_or(false)) {
-    [valueComponents
-        addObject:RCTLocalizedString("expanded", "a menu, dialog, accordian panel, or other widget which is expanded")];
-  }
+  if (accessibilityState.expanded.has_value()) {
+    [valueComponents addObject:accessibilityState.expanded.value()
+        ? RCTLocalizedString("expanded", "a menu, dialog, accordian panel, or other widget which is expanded")
+        : RCTLocalizedString("collapsed", "a menu, dialog, accordian panel, or other widget which is collapsed")];
+  }
 
   if (accessibilityState.busy) {
