diff --git a/node_modules/react-native/ReactCommon/react/runtime/hermes/HermesInstance.cpp b/node_modules/react-native/ReactCommon/react/runtime/hermes/HermesInstance.cpp
index d09f3a7..9018067 100644
--- a/node_modules/react-native/ReactCommon/react/runtime/hermes/HermesInstance.cpp
+++ b/node_modules/react-native/ReactCommon/react/runtime/hermes/HermesInstance.cpp
@@ -126,6 +126,8 @@ std::unique_ptr<JSRuntime> HermesInstance::createJSRuntime(
   assert(msgQueueThread != nullptr);
 
   auto gcConfig = ::hermes::vm::GCConfig::Builder()
+                      .withInitHeapSize(150ll * 1024 * 1024)
+                      .withShouldReleaseUnused(::hermes::vm::kReleaseUnusedNone)
                       // Default to 3GB
                       .withMaxHeapSize(3072 << 20)
                       .withName("RNBridgeless");
