diff --git a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
index 71dd62f..f732944 100644
--- a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
@@ -659,6 +659,8 @@ const CGFloat BACKGROUND_COLOR_ZPOSITION = -1024.0f;
   _filterLayer = nil;
   [self clearExistingBackgroundImageLayers];
 
+  self.autoresizingMask = UIViewAutoresizingNone;
+
   _propKeysManagedByAnimated_DO_NOT_USE_THIS_IS_BROKEN = nil;
   _eventEmitter.reset();
   _isJSResponder = NO;
