diff --git a/node_modules/react-native/.DS_Store b/node_modules/react-native/.DS_Store
new file mode 100644
index 0000000..60811a9
Binary files /dev/null and b/node_modules/react-native/.DS_Store differ
diff --git a/node_modules/react-native/ReactAndroid/.DS_Store b/node_modules/react-native/ReactAndroid/.DS_Store
new file mode 100644
index 0000000..8bfcb73
Binary files /dev/null and b/node_modules/react-native/ReactAndroid/.DS_Store differ
diff --git a/node_modules/react-native/ReactAndroid/publish.gradle b/node_modules/react-native/ReactAndroid/publish.gradle
index 32287a7..5607be3 100644
--- a/node_modules/react-native/ReactAndroid/publish.gradle
+++ b/node_modules/react-native/ReactAndroid/publish.gradle
@@ -13,7 +13,11 @@ def signingKey = findProperty("SIGNING_KEY")
 def signingPwd = findProperty("SIGNING_PWD")
 
 def reactAndroidProjectDir = project(':packages:react-native:ReactAndroid').projectDir
-def mavenTempLocalUrl = "file:///tmp/maven-local"
+def mavenURL = "https://maven.pkg.github.com/Expensify/App"
+
+def patchedVersion = System.getenv("PATCHED_VERSION")
+def isHybridBuild = System.getenv("IS_HYBRID_BUILD")?.toBoolean()
+def publishingGroupId = isHybridBuild ? "com.expensify.react-hybrid" : "com.expensify.react-standalone"
 
 publishing {
     publications {
@@ -26,12 +30,14 @@ publishing {
                 }
             }
 
+            groupId = publishingGroupId
+
             // We populate the publishing version using the project version,
             // appending -SNAPSHOT if on nightly or prerelase.
             if (isSnapshot) {
-                version = this.version + "-SNAPSHOT"
+                version = patchedVersion + "-SNAPSHOT"
             } else {
-                version = this.version
+                version = patchedVersion
             }
 
             pom {
@@ -39,10 +45,14 @@ publishing {
                 description = "A framework for building native apps with React"
                 url = "https://github.com/facebook/react-native"
 
+                properties = [
+                    "patchesHash": System.getenv("PATCHES_HASH"),
+                ]
+
                 developers {
                     developer {
-                        id = "facebook"
-                        name = "Facebook"
+                        id = "expensify"
+                        name = "Expensify"
                     }
                 }
 
@@ -65,8 +75,11 @@ publishing {
 
     repositories {
         maven {
-            name = "mavenTempLocal"
-            url = mavenTempLocalUrl
+            url = mavenURL
+            credentials {
+                username = System.getenv("GH_PUBLISH_ACTOR")
+                password = System.getenv("GH_PUBLISH_TOKEN")
+            }
         }
     }
 
