diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
index 5b3ffa4..ff56df0 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
@@ -139,6 +139,7 @@ public class ReactEditText extends AppCompatEditText {
   private boolean mContextMenuHidden = false;
   private boolean mDidAttachToWindow = false;
   private boolean mSelectTextOnFocus = false;
+  private boolean hasSelectedTextOnFocus = false;
   private @Nullable String mPlaceholder = null;
   private Overflow mOverflow = Overflow.VISIBLE;
 
@@ -260,11 +261,11 @@ public class ReactEditText extends AppCompatEditText {
   @Override
   protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
     onContentSizeChange();
-    if (mSelectTextOnFocus && isFocused()) {
+    if (mSelectTextOnFocus && isFocused() && !hasSelectedTextOnFocus) {
       // Explicitly call this method to select text when layout is drawn
       selectAll();
       // Prevent text on being selected for next layout pass
-      mSelectTextOnFocus = false;
+      hasSelectedTextOnFocus = true;
     }
   }
 
@@ -398,7 +399,7 @@ public class ReactEditText extends AppCompatEditText {
     if (getShowSoftInputOnFocus()) {
       showSoftKeyboard();
     }
-
+    hasSelectedTextOnFocus = true;
     return focused;
   }
 
