diff --git a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
--- a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
@@ -1440,9 +1440,16 @@
         addObject:RCTLocalizedString(
                       "mixed", "a checkbox, radio button, or other widget which is both checked and unchecked")];
   }
-  if (accessibilityState.expanded.value_or(false)) {
-    [valueComponents
-        addObject:RCTLocalizedString("expanded", "a menu, dialog, accordian panel, or other widget which is expanded")];
+  if (accessibilityState.expanded.has_value()) {
+    if (accessibilityState.expanded.value()) {
+      [valueComponents
+          addObject:RCTLocalizedString(
+                        "expanded", "a menu, dialog, accordian panel, or other widget which is expanded")];
+    } else {
+      [valueComponents
+          addObject:RCTLocalizedString(
+                        "collapsed", "a menu, dialog, accordian panel, or other widget which is collapsed")];
+    }
   }

   if (accessibilityState.busy) {
