diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.kt b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.kt
index ffba031..87e18c2 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.kt
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.kt
@@ -146,6 +146,7 @@ public open class ReactEditText public constructor(context: Context) : AppCompat
   private var contextMenuHidden = false
   private var didAttachToWindow = false
   private var selectTextOnFocus = false
+  private var hasSelectedTextOnFocus = false;
   private var placeholder: String? = null
   private var overflow = Overflow.VISIBLE
 
@@ -291,11 +292,11 @@ public open class ReactEditText public constructor(context: Context) : AppCompat
 
   override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
     onContentSizeChange()
-    if (selectTextOnFocus && isFocused) {
+    if (selectTextOnFocus && isFocused && !hasSelectedTextOnFocus) {
       // Explicitly call this method to select text when layout is drawn
       selectAll()
       // Prevent text on being selected for next layout pass
-      selectTextOnFocus = false
+      hasSelectedTextOnFocus = false
     }
   }
 
@@ -437,6 +438,7 @@ public open class ReactEditText public constructor(context: Context) : AppCompat
       showSoftKeyboard()
     }
 
+    hasSelectedTextOnFocus = true
     return focused
   }
 
