diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ReactModalHostView.kt b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ReactModalHostView.kt
index c711027..7255545 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ReactModalHostView.kt
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ReactModalHostView.kt
@@ -382,6 +382,20 @@ public class ReactModalHostView(context: ThemedReactContext) :
         }
       }
 
+      // If enforceNavigationBarContrast is explicitly set in the app theme (AppTheme),
+      // we need to override the default behaviour for edge-to-edge mode in WindowUtils.setSystemBarsTranslucency.
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+        val attrs = intArrayOf(android.R.attr.enforceNavigationBarContrast)
+        val ta = context.obtainStyledAttributes(R.style.Theme_FullScreenDialog, attrs)
+
+        if (ta.length() == 1) {
+          val enforceNavigationBarContrastStyleValue = ta.getBoolean(0, false)
+          dialogWindow.isNavigationBarContrastEnforced = enforceNavigationBarContrastStyleValue
+        }
+
+        ta.recycle()
+      }
+
       // Navigation bar cannot be translucent without status bar being translucent too
       if (navigationBarTranslucent) {
         dialogWindow.enableEdgeToEdge()
