diff --git a/node_modules/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp b/node_modules/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp
index bd327ef..a51c45d 100644
--- a/node_modules/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp
+++ b/node_modules/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp
@@ -118,6 +118,9 @@ YogaLayoutableShadowNode::YogaLayoutableShadowNode(
       if (auto layoutableChild =
               std::dynamic_pointer_cast<const YogaLayoutableShadowNode>(
                   child)) {
+        if (static_cast<YGDisplay>(layoutableChild->yogaNode_.style().display()) == YGDisplayContents) {
+          yogaNode_.setDirty(true);
+        }
         yogaLayoutableChildren_.push_back(std::move(layoutableChild));
       }
     }
