diff --git a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
index f732944..8ffdd57 100644
--- a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
@@ -659,6 +659,8 @@ const CGFloat BACKGROUND_COLOR_ZPOSITION = -1024.0f;
   _filterLayer = nil;
   [self clearExistingBackgroundImageLayers];
 
+  self.layer.mask = nil;
+
   self.autoresizingMask = UIViewAutoresizingNone;
 
   _propKeysManagedByAnimated_DO_NOT_USE_THIS_IS_BROKEN = nil;
