diff --git a/node_modules/react-native/React/Views/UIView+React.m b/node_modules/react-native/React/Views/UIView+React.m
index 2ee2c50..3734911 100644
--- a/node_modules/react-native/React/Views/UIView+React.m
+++ b/node_modules/react-native/React/Views/UIView+React.m
@@ -284,6 +284,12 @@ - (UIViewController *)reactViewController
   id responder = [self nextResponder];
   while (responder) {
     if ([responder isKindOfClass:[UIViewController class]]) {
+      // The found view controller is not the topmost one currently displayed.
+      // This code ensures that the next modal is always presented by the topmost currently visible modal.
+      // See: https://github.com/facebook/react-native/issues/52790
+      while ([responder presentedViewController]) {
+          responder = [responder presentedViewController];
+      }
       return responder;
     }
     responder = [responder nextResponder];
