diff --git a/node_modules/@shopify/flash-list/dist/recyclerview/hooks/useRecyclerViewController.js b/node_modules/@shopify/flash-list/dist/recyclerview/hooks/useRecyclerViewController.js
index fe87b9b..b90f898 100644
--- a/node_modules/@shopify/flash-list/dist/recyclerview/hooks/useRecyclerViewController.js
+++ b/node_modules/@shopify/flash-list/dist/recyclerview/hooks/useRecyclerViewController.js
@@ -520,7 +520,7 @@ export function useRecyclerViewController(recyclerViewManager, ref, scrollViewRe
             setTimeout(() => {
                 recyclerViewManager.isInitialScrollComplete = true;
                 pauseOffsetCorrection.current = false;
-            }, 100);
+            }, 500);
             pauseOffsetCorrection.current = true;
             const additionalOffset = (_c = initialScrollIndexParams === null || initialScrollIndexParams === void 0 ? void 0 : initialScrollIndexParams.viewOffset) !== null && _c !== void 0 ? _c : 0;
             const offset = horizontal
