diff --git a/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js b/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js
index ee42f63..4e8d8c0 100644
--- a/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js
+++ b/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js
@@ -380,16 +380,15 @@ const RecyclerViewComponent = (props, ref) => {
         }
         return onScrollHandler;
     }, [onScrollHandler, scrollY, stickyHeaders, stickyHeaderUseNativeDriver]);
-    const shouldMaintainVisibleContentPosition = recyclerViewManager.shouldMaintainVisibleContentPosition();
     const maintainVisibleContentPositionInternal = useMemo(() => {
-        if (shouldMaintainVisibleContentPosition) {
+        if (maintainVisibleContentPosition != null) {
             return {
                 ...maintainVisibleContentPosition,
                 minIndexForVisible: 0,
             };
         }
         return undefined;
-    }, [maintainVisibleContentPosition, shouldMaintainVisibleContentPosition]);
+    }, [maintainVisibleContentPosition]);
     const shouldRenderFromBottom = recyclerViewManager.getDataLength() > 0 &&
         ((_d = maintainVisibleContentPosition === null || maintainVisibleContentPosition === void 0 ? void 0 : maintainVisibleContentPosition.startRenderingFromBottom) !== null && _d !== void 0 ? _d : false);
     // Create view for measuring bounded size
@@ -401,11 +399,11 @@ const RecyclerViewComponent = (props, ref) => {
             }, ref: firstChildViewRef }));
     }, [horizontal, stickyHeaderOffset]);
     const scrollAnchor = useMemo(() => {
-        if (shouldMaintainVisibleContentPosition) {
+        if (maintainVisibleContentPosition != null) {
             return (React.createElement(ScrollAnchor, { horizontal: Boolean(horizontal), scrollAnchorRef: scrollAnchorRef }));
         }
         return null;
-    }, [horizontal, shouldMaintainVisibleContentPosition]);
+    }, [horizontal, maintainVisibleContentPosition]);
     // console.log("render", recyclerViewManager.getRenderStack());
     // Render the main RecyclerView structure
     return (React.createElement(RecyclerViewContextProvider, { value: recyclerViewContext },
diff --git a/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx b/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx
index b2bd67a..d4bf02d 100644
--- a/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx
+++ b/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx
@@ -572,18 +572,15 @@ const RecyclerViewComponent = <T,>(
     return onScrollHandler;
   }, [onScrollHandler, scrollY, stickyHeaders, stickyHeaderUseNativeDriver]);
 
-  const shouldMaintainVisibleContentPosition =
-    recyclerViewManager.shouldMaintainVisibleContentPosition();
-
   const maintainVisibleContentPositionInternal = useMemo(() => {
-    if (shouldMaintainVisibleContentPosition) {
+    if (maintainVisibleContentPosition != null) {
       return {
         ...maintainVisibleContentPosition,
         minIndexForVisible: 0,
       };
     }
     return undefined;
-  }, [maintainVisibleContentPosition, shouldMaintainVisibleContentPosition]);
+  }, [maintainVisibleContentPosition]);
 
   const shouldRenderFromBottom =
     recyclerViewManager.getDataLength() > 0 &&
@@ -604,7 +600,7 @@ const RecyclerViewComponent = <T,>(
   }, [horizontal, stickyHeaderOffset]);
 
   const scrollAnchor = useMemo(() => {
-    if (shouldMaintainVisibleContentPosition) {
+    if (maintainVisibleContentPosition != null) {
       return (
         <ScrollAnchor
           horizontal={Boolean(horizontal)}
@@ -613,7 +609,7 @@ const RecyclerViewComponent = <T,>(
       );
     }
     return null;
-  }, [horizontal, shouldMaintainVisibleContentPosition]);
+  }, [horizontal, maintainVisibleContentPosition]);
 
   // console.log("render", recyclerViewManager.getRenderStack());
 
