diff --git a/node_modules/@shopify/flash-list/dist/recyclerview/utils/measureLayout.web.js b/node_modules/@shopify/flash-list/dist/recyclerview/utils/measureLayout.web.js
index 17d9812..fe112f1 100644
--- a/node_modules/@shopify/flash-list/dist/recyclerview/utils/measureLayout.web.js
+++ b/node_modules/@shopify/flash-list/dist/recyclerview/utils/measureLayout.web.js
@@ -28,7 +28,10 @@ export function areDimensionsEqual(value1, value2) {
     return Math.abs(value1 - value2) <= 1;
 }
 export function roundOffPixel(value) {
-    return value;
+    const dpr = typeof window !== "undefined" && window.devicePixelRatio
+        ? window.devicePixelRatio
+        : 1;
+    return Math.round(value * dpr) / dpr;
 }
 /**
  * Measures the size of the RecyclerView's outer container.
