diff --git a/node_modules/yargs/package.json b/node_modules/yargs/package.json
--- a/node_modules/yargs/package.json
+++ b/node_modules/yargs/package.json
@@ -23,9 +23,9 @@
     "./yargs": [
       {
         "import": "./yargs.mjs",
-        "require": "./yargs"
+        "require": "./yargs.cjs"
       },
-      "./yargs"
+      "./yargs.cjs"
     ]
   },
   "type": "module",
diff --git a/node_modules/yargs/yargs.cjs b/node_modules/yargs/yargs.cjs
new file mode 100644
--- /dev/null
+++ b/node_modules/yargs/yargs.cjs
@@ -0,0 +1,9 @@
+// TODO: consolidate on using a helpers file at some point in the future, which
+// is the approach currently used to export Parser and applyExtends for ESM:
+const {applyExtends, cjsPlatformShim, Parser, Yargs, processArgv} = require('./build/index.cjs')
+Yargs.applyExtends = (config, cwd, mergeExtends) => {
+  return applyExtends(config, cwd, mergeExtends, cjsPlatformShim)
+}
+Yargs.hideBin = processArgv.hideBin
+Yargs.Parser = Parser
+module.exports = Yargs
