diff --git a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
index 4769d28..ef88a07 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
+++ b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
@@ -207,6 +207,14 @@ static void requestPhotoLibraryAccess(RCTPromiseRejectBlock reject, PhotosAuthor
         options.fetchLimit = 1;
         PHFetchResult<PHAsset *> *createdAsset = [PHAsset fetchAssetsWithLocalIdentifiers:@[placeholder.localIdentifier]
                                                                                   options:options];
+
+        if (![createdAsset isKindOfClass:[PHAsset class]]) {
+          resolve(@{
+            @"node": [NSNull null]
+          });
+          return;
+        }
+
         if (createdAsset.count < 1) {
           reject(kErrorUnableToSave, nil, nil);
           return;
