diff --git a/node_modules/@rock-js/provider-s3/dist/src/lib/providerS3.js b/node_modules/@rock-js/provider-s3/dist/src/lib/providerS3.js
index 60f915a..4c47aeb 100644
--- a/node_modules/@rock-js/provider-s3/dist/src/lib/providerS3.js
+++ b/node_modules/@rock-js/provider-s3/dist/src/lib/providerS3.js
@@ -148,10 +148,22 @@ export class S3BuildCache {
         ];
     }
     async upload({ artifactName, uploadArtifactName, }) {
+        console.log('TEST uploadArtifactName STARTS, artifactName: ', artifactName);
         const key = uploadArtifactName
             ? `${this.directory}/${uploadArtifactName}`
             : `${this.directory}/${artifactName}.zip`;
         const presignedUrl = await getSignedUrl(this.s3, new clientS3.GetObjectCommand({ Bucket: this.bucket, Key: key }), { expiresIn: this.linkExpirationTime });
+        
+        console.log('TEST uploadArtifactName from S3, JSON: ', JSON.stringify({
+            name: artifactName,
+            url: presignedUrl,
+        }, null, 2));
+        
+        console.log(JSON.stringify({
+            name: artifactName,
+            url: presignedUrl,
+        }));
+        
         return {
             name: artifactName,
             url: presignedUrl,
