diff --git a/node_modules/react-native-web/dist/exports/Text/index.js b/node_modules/react-native-web/dist/exports/Text/index.js
index 4130386..1076f55 100644
--- a/node_modules/react-native-web/dist/exports/Text/index.js
+++ b/node_modules/react-native-web/dist/exports/Text/index.js
@@ -176,7 +176,7 @@ var styles = StyleSheet.create({
   textMultiLine: {
     display: '-webkit-box',
     maxWidth: '100%',
-    overflow: 'hidden',
+    overflow: 'clip',
     textOverflow: 'ellipsis',
     WebkitBoxOrient: 'vertical'
   },
diff --git a/node_modules/react-native-web/src/exports/Text/index.js b/node_modules/react-native-web/src/exports/Text/index.js
index f79e82c..f27ccec 100644
--- a/node_modules/react-native-web/src/exports/Text/index.js
+++ b/node_modules/react-native-web/src/exports/Text/index.js
@@ -223,7 +223,7 @@ const styles = StyleSheet.create({
   textMultiLine: {
     display: '-webkit-box',
     maxWidth: '100%',
-    overflow: 'hidden',
+    overflow: 'clip',
     textOverflow: 'ellipsis',
     WebkitBoxOrient: 'vertical'
   },
