diff --git a/node_modules/react-native-web/dist/exports/Modal/ModalAnimation.js b/node_modules/react-native-web/dist/exports/Modal/ModalAnimation.js
index 2017707..d07af81 100644
--- a/node_modules/react-native-web/dist/exports/Modal/ModalAnimation.js
+++ b/node_modules/react-native-web/dist/exports/Modal/ModalAnimation.js
@@ -12,21 +12,24 @@ import * as React from 'react';
 import StyleSheet from '../StyleSheet';
 import createElement from '../createElement';
 var ANIMATION_DURATION = 250;
-function getAnimationStyle(animationType, visible) {
+function getAnimationStyle(animationType, visible, zIndex) {
   if (animationType === 'slide') {
     return visible ? animatedSlideInStyles : animatedSlideOutStyles;
   }
   if (animationType === 'fade') {
     return visible ? animatedFadeInStyles : animatedFadeOutStyles;
   }
-  return visible ? styles.container : styles.hidden;
+  return visible ? [styles.container, {
+    zIndex
+  }] : styles.hidden;
 }
 function ModalAnimation(props) {
   var animationType = props.animationType,
     children = props.children,
     onDismiss = props.onDismiss,
     onShow = props.onShow,
-    visible = props.visible;
+    visible = props.visible,
+    zIndex = props.zIndex;
   var _React$useState = React.useState(false),
     isRendering = _React$useState[0],
     setIsRendering = _React$useState[1];
@@ -64,7 +67,7 @@ function ModalAnimation(props) {
     wasVisible.current = visible;
   }, [isAnimated, visible, animationEndCallback]);
   return isRendering || visible ? createElement('div', {
-    style: isRendering ? getAnimationStyle(animationType, visible) : styles.hidden,
+    style: isRendering ? getAnimationStyle(animationType, visible, zIndex) : styles.hidden,
     onAnimationEnd: animationEndCallback,
     children
   }) : null;
@@ -75,8 +78,7 @@ var styles = StyleSheet.create({
     top: 0,
     right: 0,
     bottom: 0,
-    left: 0,
-    zIndex: 9999
+    left: 0
   },
   animatedIn: {
     animationDuration: ANIMATION_DURATION + "ms",
diff --git a/node_modules/react-native-web/dist/exports/Modal/index.js b/node_modules/react-native-web/dist/exports/Modal/index.js
index a9a7c36..27cb5ba 100644
--- a/node_modules/react-native-web/dist/exports/Modal/index.js
+++ b/node_modules/react-native-web/dist/exports/Modal/index.js
@@ -52,6 +52,7 @@ function addActiveModal(modalId, listener) {
   notifyActiveModalListeners();
 }
 var Modal = /*#__PURE__*/React.forwardRef((props, forwardedRef) => {
+  var _props$style$zIndex, _props$style;
   var animationType = props.animationType,
     children = props.children,
     onDismiss = props.onDismiss,
@@ -87,14 +88,13 @@ var Modal = /*#__PURE__*/React.forwardRef((props, forwardedRef) => {
     animationType: animationType,
     onDismiss: onDismissCallback,
     onShow: onShowCallback,
-    visible: visible
-  }, /*#__PURE__*/React.createElement(ModalFocusTrap, {
-    active: isActive
+    visible: visible,
+    zIndex: (_props$style$zIndex = props == null ? void 0 : (_props$style = props.style) == null ? void 0 : _props$style.zIndex) !== null && _props$style$zIndex !== void 0 ? _props$style$zIndex : 9999
   }, /*#__PURE__*/React.createElement(ModalContent, _extends({}, rest, {
     active: isActive,
     onRequestClose: onRequestClose,
     ref: forwardedRef,
     transparent: transparent
-  }), children))));
+  }), children)));
 });
 export default Modal;
\ No newline at end of file
