diff --git a/node_modules/react-native-web/dist/exports/ScrollView/index.js b/node_modules/react-native-web/dist/exports/ScrollView/index.js
index c4f9b5b..cc5076e 100644
--- a/node_modules/react-native-web/dist/exports/ScrollView/index.js
+++ b/node_modules/react-native-web/dist/exports/ScrollView/index.js
@@ -558,8 +558,9 @@ class ScrollView extends React.Component {
     var children = hasStickyHeaderIndices || pagingEnabled ? React.Children.map(this.props.children, (child, i) => {
       var isSticky = hasStickyHeaderIndices && stickyHeaderIndices.indexOf(i) > -1;
       if (child != null && (isSticky || pagingEnabled)) {
+        var stickyItemIndex = (this.props.children.length - 1) - i + 10;
         return /*#__PURE__*/React.createElement(View, {
-          style: [isSticky && styles.stickyHeader, pagingEnabled && styles.pagingEnabledChild]
+          style: [isSticky && styles.stickyHeader, pagingEnabled && styles.pagingEnabledChild, isSticky && {zIndex: stickyItemIndex}]
         }, child);
       } else {
         return child;
@@ -636,7 +637,6 @@ var styles = StyleSheet.create({
   stickyHeader: {
     position: 'sticky',
     top: 0,
-    zIndex: 10
   },
   pagingEnabledHorizontal: {
     scrollSnapType: 'x mandatory'
diff --git a/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js b/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
index 42c4984..f9e6ea3 100644
--- a/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
+++ b/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
@@ -108,6 +108,15 @@ function windowSizeOrDefault(windowSize) {
  *
  */
 class VirtualizedList extends StateSafePureComponent {
+  // reverse push order logic when props.inverted = true
+  pushOrUnshift(input, item) {
+    if (this.props.inverted) {
+      input.unshift(item);
+    } else {
+      input.push(item);
+    }
+  }
+
   // scrollToEnd may be janky without getItemLayout prop
   scrollToEnd(params) {
     var animated = params ? params.animated : true;
@@ -343,6 +352,7 @@ class VirtualizedList extends StateSafePureComponent {
     };
     this._defaultRenderScrollComponent = props => {
       var onRefresh = props.onRefresh;
+      var inversionStyle = this.props.inverted ? this.props.horizontal ? styles.rowReverse : styles.columnReverse : null;
       if (this._isNestedWithSameOrientation()) {
         // $FlowFixMe[prop-missing] - Typing ReactNativeComponent revealed errors
         return /*#__PURE__*/React.createElement(View, props);
@@ -354,6 +364,7 @@ class VirtualizedList extends StateSafePureComponent {
           // $FlowFixMe[prop-missing] Invalid prop usage
           // $FlowFixMe[incompatible-use]
           React.createElement(ScrollView, _extends({}, props, {
+            contentContainerStyle: StyleSheet.compose(inversionStyle, this.props.contentContainerStyle),
             refreshControl: props.refreshControl == null ? /*#__PURE__*/React.createElement(RefreshControl
             // $FlowFixMe[incompatible-type]
             , {
@@ -366,7 +377,9 @@ class VirtualizedList extends StateSafePureComponent {
       } else {
         // $FlowFixMe[prop-missing] Invalid prop usage
         // $FlowFixMe[incompatible-use]
-        return /*#__PURE__*/React.createElement(ScrollView, props);
+        return /*#__PURE__*/React.createElement(ScrollView, _extends({}, props, {
+          contentContainerStyle: StyleSheet.compose(inversionStyle, this.props.contentContainerStyle)
+        }));
       }
     };
     this._onCellLayout = (e, cellKey, index) => {
@@ -679,7 +692,7 @@ class VirtualizedList extends StateSafePureComponent {
         onViewableItemsChanged = _this$props3.onViewableItemsChanged,
         viewabilityConfig = _this$props3.viewabilityConfig;
       if (onViewableItemsChanged) {
-        this._viewabilityTuples.push({
+        this.pushOrUnshift(this._viewabilityTuples, {
           viewabilityHelper: new ViewabilityHelper(viewabilityConfig),
           onViewableItemsChanged: onViewableItemsChanged
         });
@@ -991,15 +1004,15 @@ class VirtualizedList extends StateSafePureComponent {
     var end = getItemCount(data) - 1;
     var prevCellKey;
     last = Math.min(end, last);
-    var _loop = function _loop() {
+    var _loop = () => {
       var item = getItem(data, ii);
       var key = VirtualizedList._keyExtractor(item, ii, _this.props);
       _this._indicesToKeys.set(ii, key);
       if (stickyIndicesFromProps.has(ii + stickyOffset)) {
-        stickyHeaderIndices.push(cells.length);
+        this.pushOrUnshift(stickyHeaderIndices, cells.length);
       }
       var shouldListenForLayout = getItemLayout == null || debug || _this._fillRateHelper.enabled();
-      cells.push(/*#__PURE__*/React.createElement(CellRenderer, _extends({
+      this.pushOrUnshift(cells, /*#__PURE__*/React.createElement(CellRenderer, _extends({
         CellRendererComponent: CellRendererComponent,
         ItemSeparatorComponent: ii < end ? ItemSeparatorComponent : undefined,
         ListItemComponent: ListItemComponent,
@@ -1074,14 +1087,14 @@ class VirtualizedList extends StateSafePureComponent {
     // 1. Add cell for ListHeaderComponent
     if (ListHeaderComponent) {
       if (stickyIndicesFromProps.has(0)) {
-        stickyHeaderIndices.push(0);
+        this.pushOrUnshift(stickyHeaderIndices, 0);
       }
       var _element = /*#__PURE__*/React.isValidElement(ListHeaderComponent) ? ListHeaderComponent :
       /*#__PURE__*/
       // $FlowFixMe[not-a-component]
       // $FlowFixMe[incompatible-type-arg]
       React.createElement(ListHeaderComponent, null);
-      cells.push(/*#__PURE__*/React.createElement(VirtualizedListCellContextProvider, {
+      this.pushOrUnshift(cells, /*#__PURE__*/React.createElement(VirtualizedListCellContextProvider, {
         cellKey: this._getCellKey() + '-header',
         key: "$header"
       }, /*#__PURE__*/React.createElement(View
@@ -1105,7 +1118,7 @@ class VirtualizedList extends StateSafePureComponent {
       // $FlowFixMe[not-a-component]
       // $FlowFixMe[incompatible-type-arg]
       React.createElement(ListEmptyComponent, null);
-      cells.push(/*#__PURE__*/React.createElement(VirtualizedListCellContextProvider, {
+      this.pushOrUnshift(cells, /*#__PURE__*/React.createElement(VirtualizedListCellContextProvider, {
         cellKey: this._getCellKey() + '-empty',
         key: "$empty"
       }, /*#__PURE__*/React.cloneElement(_element2, {
@@ -1145,7 +1158,7 @@ class VirtualizedList extends StateSafePureComponent {
           var firstMetrics = this.__getFrameMetricsApprox(section.first, this.props);
           var lastMetrics = this.__getFrameMetricsApprox(last, this.props);
           var spacerSize = lastMetrics.offset + lastMetrics.length - firstMetrics.offset;
-          cells.push(/*#__PURE__*/React.createElement(View, {
+          this.pushOrUnshift(cells, /*#__PURE__*/React.createElement(View, {
             key: "$spacer-" + section.first,
             style: {
               [spacerKey]: spacerSize
@@ -1168,7 +1181,7 @@ class VirtualizedList extends StateSafePureComponent {
       // $FlowFixMe[not-a-component]
       // $FlowFixMe[incompatible-type-arg]
       React.createElement(ListFooterComponent, null);
-      cells.push(/*#__PURE__*/React.createElement(VirtualizedListCellContextProvider, {
+      this.pushOrUnshift(cells, /*#__PURE__*/React.createElement(VirtualizedListCellContextProvider, {
         cellKey: this._getFooterCellKey(),
         key: "$footer"
       }, /*#__PURE__*/React.createElement(View, {
@@ -1179,6 +1192,14 @@ class VirtualizedList extends StateSafePureComponent {
       _element3)));
     }
 
+    if (this.props.inverted && stickyHeaderIndices.length > 0) {
+      var totalCells = cells.length;
+      stickyHeaderIndices = stickyHeaderIndices.map(function(recordedIndex) {
+        return totalCells - 1 - recordedIndex;
+      });
+    }
+
+
     // 4. Render the ScrollView
     var scrollProps = _objectSpread(_objectSpread({}, this.props), {}, {
       onContentSizeChange: this._onContentSizeChange,
@@ -1353,7 +1374,7 @@ class VirtualizedList extends StateSafePureComponent {
        * suppresses an error found when Flow v0.68 was deployed. To see the
        * error delete this comment and run Flow. */
       if (frame.inLayout) {
-        framesInLayout.push(frame);
+        this.pushOrUnshift(framesInLayout, frame);
       }
     }
     var windowTop = this.__getFrameMetricsApprox(this.state.cellsAroundViewport.first, this.props).offset;
@@ -1526,6 +1547,12 @@ var styles = StyleSheet.create({
   horizontallyInverted: {
     transform: 'scaleX(-1)'
   },
+  rowReverse: {
+    flexDirection: 'row-reverse',
+  },
+  columnReverse: {
+    flexDirection: 'column-reverse',
+  },
   debug: {
     flex: 1
   },
