diff --git a/node_modules/react-native-image-size/android/build.gradle b/node_modules/react-native-image-size/android/build.gradle
index cc2d50a..bb21239 100644
--- a/node_modules/react-native-image-size/android/build.gradle
+++ b/node_modules/react-native-image-size/android/build.gradle
@@ -1,7 +1,6 @@
 buildscript {
     repositories {
         mavenCentral()
-        jcenter()
     }
 
     dependencies {
