diff --git a/node_modules/expensify-common/dist/str.js b/node_modules/expensify-common/dist/str.js
index 124dd32..8d20430 100644
--- a/node_modules/expensify-common/dist/str.js
+++ b/node_modules/expensify-common/dist/str.js
@@ -53,6 +53,126 @@ function resultFn(parameter, ...args) {
     }
     return parameter;
 }
+// Str methods that other Str methods call are plain module-level functions instead of being reached through
+// `this`: react-native-worklets 0.13 no longer keeps an object literal together on a Worklet Runtime, so `this`
+// inside a workletized method does not point to Str there.
+function compare(strA, strB) {
+    if (strA < strB) {
+        return -1;
+    }
+    if (strA > strB) {
+        return 1;
+    }
+    return 0;
+}
+function caseInsensitiveCompare(strA, strB) {
+    const lowerCaseStrA = strA.toLocaleLowerCase();
+    const lowerCaseStrB = strB.toLocaleLowerCase();
+    return compare(lowerCaseStrA, lowerCaseStrB);
+}
+function contains(haystack, needle) {
+    return haystack.indexOf(needle) !== -1;
+}
+function cutAfter(str, substr) {
+    const index = str.indexOf(substr);
+    if (index !== -1) {
+        return str.substring(0, index);
+    }
+    return str;
+}
+function cutBefore(str, substr) {
+    const index = str.indexOf(substr);
+    if (index !== -1) {
+        return str.substring(index + substr.length);
+    }
+    return str;
+}
+function extractEmailDomain(email) {
+    return cutBefore(email, '@');
+}
+function getExtension(url) {
+    var _a, _b;
+    if (typeof url !== 'string') {
+        Log_1.default.warn('Str.getExtension: url is not a string', { url });
+        return undefined;
+    }
+    return (_b = (_a = url.split('.').pop()) === null || _a === void 0 ? void 0 : _a.split('?')[0]) === null || _b === void 0 ? void 0 : _b.toLowerCase();
+}
+function getRawByteSize(inputChar) {
+    const onlyChar = String(inputChar);
+    const c = onlyChar.charCodeAt(0);
+    // If we are grabbing the byte size, we need to temporarily diable no-bitwise for linting
+    /* eslint-disable no-bitwise */
+    if (c < 1 << 7) {
+        return 1;
+    }
+    if (c < 1 << 11) {
+        return 2;
+    }
+    if (c < 1 << 16) {
+        return 3;
+    }
+    if (c < 1 << 21) {
+        return 4;
+    }
+    if (c < 1 << 26) {
+        return 5;
+    }
+    if (c < 1 << 31) {
+        return 6;
+    }
+    /* eslint-enable no-bitwise */
+    return Number.NaN;
+}
+function htmlDecode(s) {
+    return HtmlEntities.decode(s);
+}
+function htmlEncode(s) {
+    return HtmlEntities.encode(s);
+}
+function isTypeOf(obj, type) {
+    return Object.prototype.toString.call(obj) === `[object ${type}]`;
+}
+function isNumber(obj) {
+    return isTypeOf(obj, 'Number');
+}
+function isString(obj) {
+    return isTypeOf(obj, 'String');
+}
+function isUndefined(obj) {
+    // eslint-disable-next-line no-void
+    return obj === void 0;
+}
+function isValidE164Phone(phone) {
+    return Constants.CONST.SMS.E164_REGEX.test(phone);
+}
+function isValidEmail(str) {
+    if (!str || typeof str !== 'string') {
+        return false;
+    }
+    const unicodeVersion = punycode_1.default.toUnicode(str);
+    if (String(unicodeVersion).match(Constants.CONST.REG_EXP.EMOJI_RULE)) {
+        return false;
+    }
+    return !!String(str).match(Constants.CONST.REG_EXP.EMAIL);
+}
+function maskFirstNCharacters(str, num, mask) {
+    // if str is empty, str or mask aren't strings,
+    // or n is not a number, do nothing
+    if (!isString(str) || !isString(mask) || str.length === 0 || !isNumber(num)) {
+        return str;
+    }
+    return str.substring(0, num).replaceAll(/./g, mask) + str.substring(num);
+}
+function removeSMSDomain(text) {
+    return text.replaceAll(REMOVE_SMS_DOMAIN_PATTERN, '');
+}
+function removeTrailingComma(str) {
+    return str.trim().replaceAll(/(,$)/g, '');
+}
+function startsWith(haystack, needle) {
+    return isString(haystack) && isString(needle) && haystack.substring(0, needle.length) === needle;
+}
 const Str = {
     /**
      * Return true if the string is ending with the provided suffix
@@ -110,18 +230,14 @@ const Str = {
      * @param s The string to decode.
      * @returns The decoded string.
      */
-    htmlDecode(s) {
-        return HtmlEntities.decode(s);
-    },
+    htmlDecode,
     /**
      * HTML encodes the given string.
      *
      * @param s The string to encode.
      * @return string @p s HTML encoded.
      */
-    htmlEncode(s) {
-        return HtmlEntities.encode(s);
-    },
+    htmlEncode,
     /**
      * Decodes the given HTML encoded string.
      *
@@ -165,7 +281,7 @@ const Str = {
      * @returns True when first === second, ignoring HTML encoding
      */
     htmlEncodingInsensitiveEquals(first, second) {
-        return first === second || this.htmlDecode(first) === second || this.htmlEncode(first) === second;
+        return first === second || htmlDecode(first) === second || htmlEncode(first) === second;
     },
     /**
      * Creates an ID that can be used as an HTML attribute from @p str.
@@ -251,32 +367,7 @@ const Str = {
      * one.
      * @returns Byte size of the character
      */
-    getRawByteSize(inputChar) {
-        const onlyChar = String(inputChar);
-        const c = onlyChar.charCodeAt(0);
-        // If we are grabbing the byte size, we need to temporarily diable no-bitwise for linting
-        /* eslint-disable no-bitwise */
-        if (c < 1 << 7) {
-            return 1;
-        }
-        if (c < 1 << 11) {
-            return 2;
-        }
-        if (c < 1 << 16) {
-            return 3;
-        }
-        if (c < 1 << 21) {
-            return 4;
-        }
-        if (c < 1 << 26) {
-            return 5;
-        }
-        if (c < 1 << 31) {
-            return 6;
-        }
-        /* eslint-enable no-bitwise */
-        return Number.NaN;
-    },
+    getRawByteSize,
     /**
      * Gets the length of a string in bytes, including non-ASCII characters
      * @returns The number of bytes used by string
@@ -284,7 +375,7 @@ const Str = {
     getByteLength(input) {
         // Force string type
         const stringInput = String(input);
-        const byteLength = Array.from(stringInput).reduce((acc, char) => acc + this.getRawByteSize(char), 0);
+        const byteLength = Array.from(stringInput).reduce((acc, char) => acc + getRawByteSize(char), 0);
         return byteLength;
     },
     /**
@@ -296,7 +387,7 @@ const Str = {
         const stringInput = String(input);
         let totalByteLength = 0;
         for (let i = 0; i < stringInput.length; i++) {
-            const charByteSize = this.getRawByteSize(stringInput[i]);
+            const charByteSize = getRawByteSize(stringInput[i]);
             if (charByteSize + totalByteLength > maxSize) {
                 // If the next character exceeds the limit, stop and return the truncated string.
                 return `${stringInput.substr(0, i - 3)}...`;
@@ -312,9 +403,7 @@ const Str = {
      * @param needle  The case-sensitive string to search for
      * @returns True if the haystack starts with the needle.
      */
-    startsWith(haystack, needle) {
-        return this.isString(haystack) && this.isString(needle) && haystack.substring(0, needle.length) === needle;
-    },
+    startsWith,
     /**
      * Gets the textual value of the given string.
      *
@@ -322,7 +411,7 @@ const Str = {
      * @returns The text from within the HTML string.
      */
     stripHTML(str) {
-        if (!this.isString(str)) {
+        if (!isString(str)) {
             return '';
         }
         return str.replaceAll(/<[^>]*>?/gm, '');
@@ -345,13 +434,7 @@ const Str = {
      * @param substr The substring to search for.
      * @returns The cut/trimmed string.
      */
-    cutAfter(str, substr) {
-        const index = str.indexOf(substr);
-        if (index !== -1) {
-            return str.substring(0, index);
-        }
-        return str;
-    },
+    cutAfter,
     /**
      * Returns a string containing all the characters str from after the first
      * occurrence of substr to the end of the string.
@@ -361,13 +444,7 @@ const Str = {
      * @param substr The substring to search for.
      * @returns The cut/trimmed string.
      */
-    cutBefore(str, substr) {
-        const index = str.indexOf(substr);
-        if (index !== -1) {
-            return str.substring(index + substr.length);
-        }
-        return str;
-    },
+    cutBefore,
     /**
      * Checks that the string is a domain name (e.g. example.com)
      *
@@ -394,16 +471,7 @@ const Str = {
      *
      * @returns True if the string is an email
      */
-    isValidEmail(str) {
-        if (!str || typeof str !== 'string') {
-            return false;
-        }
-        const unicodeVersion = punycode_1.default.toUnicode(str);
-        if (String(unicodeVersion).match(Constants.CONST.REG_EXP.EMOJI_RULE)) {
-            return false;
-        }
-        return !!String(str).match(Constants.CONST.REG_EXP.EMAIL);
-    },
+    isValidEmail,
     /**
      * Checks if the string is an valid email address formed during comment markdown formation.
      *
@@ -421,9 +489,7 @@ const Str = {
      *
      * @returns string with the trailing comma removed
      */
-    removeTrailingComma(str) {
-        return str.trim().replaceAll(/(,$)/g, '');
-    },
+    removeTrailingComma,
     /**
      * Checks that the string is a list of coma separated email addresss.
      *
@@ -432,12 +498,12 @@ const Str = {
      * @returns True if all emails are valid or if input is empty
      */
     areValidEmails(str) {
-        const string = this.removeTrailingComma(str);
+        const string = removeTrailingComma(str);
         if (string === '') {
             return true;
         }
         const emails = string.split(',');
-        const result = emails.every((email) => this.isValidEmail(email.trim()));
+        const result = emails.every((email) => isValidEmail(email.trim()));
         return result;
     },
     /**
@@ -454,9 +520,7 @@ const Str = {
      *
      * @returns The domain name in the email address.
      */
-    extractEmailDomain(email) {
-        return this.cutBefore(email, '@');
-    },
+    extractEmailDomain,
     /**
      * Tries to extract the company name from the given email address
      * (e.g. "yelp" for "joe@yelp.co.uk").
@@ -466,7 +530,7 @@ const Str = {
      * @returns The company name in the email address or null.
      */
     extractCompanyNameFromEmailDomain(email) {
-        const domain = this.extractEmailDomain(email);
+        const domain = extractEmailDomain(email);
         if (!domain) {
             return null;
         }
@@ -485,7 +549,7 @@ const Str = {
      * @returns The local part in the email address.
      */
     extractEmailLocalPart(email) {
-        return this.cutAfter(email, '@');
+        return cutAfter(email, '@');
     },
     /**
      * Sanitize phone number to return only numbers. Return null if non valid phone number.
@@ -543,13 +607,13 @@ const Str = {
      * @returns true if the length is in the range, false otherwise
      */
     isOfLength(str, minimumLength, maximumLength) {
-        if (!this.isString(str)) {
+        if (!isString(str)) {
             return false;
         }
         if (str.length < minimumLength) {
             return false;
         }
-        if (!this.isUndefined(maximumLength) && str.length > maximumLength) {
+        if (!isUndefined(maximumLength) && str.length > maximumLength) {
             return false;
         }
         return true;
@@ -603,9 +667,7 @@ const Str = {
      *
      * @returns Returns true if the haystack contains the needle
      */
-    contains(haystack, needle) {
-        return haystack.indexOf(needle) !== -1;
-    },
+    contains,
     /**
      * Returns true if the haystack contains the needle, ignoring case
      *
@@ -615,7 +677,7 @@ const Str = {
      * @returns Returns true if the haystack contains the needle, ignoring case
      */
     caseInsensitiveContains(haystack, needle) {
-        return this.contains(haystack.toLowerCase(), needle.toLowerCase());
+        return contains(haystack.toLowerCase(), needle.toLowerCase());
     },
     /**
      * Case insensitive compare function
@@ -627,11 +689,7 @@ const Str = {
      *                   1 if first string > second string
      *                   0 if first string = second string
      */
-    caseInsensitiveCompare(strA, strB) {
-        const lowerCaseStrA = strA.toLocaleLowerCase();
-        const lowerCaseStrB = strB.toLocaleLowerCase();
-        return this.compare(lowerCaseStrA, lowerCaseStrB);
-    },
+    caseInsensitiveCompare,
     /**
      * Case insensitive equals
      *
@@ -640,7 +698,7 @@ const Str = {
      * @returns true when first == second except for case
      */
     caseInsensitiveEquals(strA, strB) {
-        return this.caseInsensitiveCompare(strA, strB) === 0;
+        return caseInsensitiveCompare(strA, strB) === 0;
     },
     /**
      * Compare function
@@ -652,15 +710,7 @@ const Str = {
      *                   1 if first string > second string
      *                   0 if first string = second string
      */
-    compare(strA, strB) {
-        if (strA < strB) {
-            return -1;
-        }
-        if (strA > strB) {
-            return 1;
-        }
-        return 0;
-    },
+    compare,
     /**
      * Check if a file extension is supported by SmartReports
      */
@@ -684,48 +734,39 @@ const Str = {
         // Hide these numbers completely
         // We should not be getting account numbers this small or large
         if (len < 6 || len > 20) {
-            return this.maskFirstNCharacters(accountNumber, len, 'X');
+            return maskFirstNCharacters(accountNumber, len, 'X');
         }
         // Can show last 4
         if (len < 14) {
-            return this.maskFirstNCharacters(accountNumber, len - 4, 'X');
+            return maskFirstNCharacters(accountNumber, len - 4, 'X');
         }
         // Can show first 6 and last 4
         const first = accountNumber.substr(0, 6);
         const last = accountNumber.substr(7);
-        const masked = this.maskFirstNCharacters(last, len - 11, 'X');
+        const masked = maskFirstNCharacters(last, len - 11, 'X');
         return `${first}${masked}`;
     },
     /**
      * Checks if something is a string
      * Stolen from underscore
      */
-    isString(obj) {
-        return this.isTypeOf(obj, 'String');
-    },
+    isString,
     /**
      * Checks if something is a number
      * Stolen from underscore
      * @param obj
      */
-    isNumber(obj) {
-        return this.isTypeOf(obj, 'Number');
-    },
+    isNumber,
     /**
      * Checks if something is a certain type
      * Stolen from underscore
      */
-    isTypeOf(obj, type) {
-        return Object.prototype.toString.call(obj) === `[object ${type}]`;
-    },
+    isTypeOf,
     /**
      * Checks to see if something is undefined
      * Stolen from underscore
      */
-    isUndefined(obj) {
-        // eslint-disable-next-line no-void
-        return obj === void 0;
-    },
+    isUndefined,
     /**
      * Replace first N characters of the string with maskChar
      * eg: maskFirstNCharacters( '1234567890', 6, 'X' ) yields XXXXXX7890
@@ -734,14 +775,7 @@ const Str = {
      * @param mask String we want replace the first N chars with
      * @returns Masked string
      */
-    maskFirstNCharacters(str, num, mask) {
-        // if str is empty, str or mask aren't strings,
-        // or n is not a number, do nothing
-        if (!this.isString(str) || !this.isString(mask) || str.length === 0 || !this.isNumber(num)) {
-            return str;
-        }
-        return str.substring(0, num).replaceAll(/./g, mask) + str.substring(num);
-    },
+    maskFirstNCharacters,
     /**
      * Trim a string
      */
@@ -753,7 +787,7 @@ const Str = {
      * @param percentageString The percentage as a string
      */
     percentageStringToNumber(percentageString) {
-        return Number(this.cutAfter(percentageString, '%'));
+        return Number(cutAfter(percentageString, '%'));
     },
     /**
      * Remove all the spaces from a string
@@ -799,7 +833,7 @@ const Str = {
      * Converts a value to boolean, case-insensitive.
      */
     toBool(value) {
-        if (this.isString(value)) {
+        if (isString(value)) {
             return value.toLowerCase() === 'true';
         }
         return !!value;
@@ -849,9 +883,7 @@ const Str = {
     /**
      * Check for whether a phone number is valid according to E.164 standard.
      */
-    isValidE164Phone(phone) {
-        return Constants.CONST.SMS.E164_REGEX.test(phone);
-    },
+    isValidE164Phone,
     /**
      * Check for whether a phone number is valid in different formats/standards. For example:
      * significant: 4404589784
@@ -880,14 +912,12 @@ const Str = {
     /**
      * Returns text without our SMS domain
      */
-    removeSMSDomain(text) {
-        return text.replaceAll(REMOVE_SMS_DOMAIN_PATTERN, '');
-    },
+    removeSMSDomain,
     /**
      * Returns true if the text is a valid E.164 phone number with our SMS domain removed
      */
     isSMSLogin(text) {
-        return this.isValidE164Phone(this.removeSMSDomain(text));
+        return isValidE164Phone(removeSMSDomain(text));
     },
     /**
      * This method will return all matches of a single regex like preg_match_all() in PHP. This is not a common part of
@@ -945,7 +975,7 @@ const Str = {
             return url;
         }
         const website = match[3] ? match[2] : `${defaultScheme}://${match[2]}`;
-        return website.toLowerCase() + this.cutBefore(match[1], match[2]);
+        return website.toLowerCase() + cutBefore(match[1], match[2]);
     },
     /**
      * Checks if parameter is a string or function
@@ -957,14 +987,7 @@ const Str = {
      * Get file extension for a given url with or
      * without query parameters
      */
-    getExtension(url) {
-        var _a, _b;
-        if (typeof url !== 'string') {
-            Log_1.default.warn('Str.getExtension: url is not a string', { url });
-            return undefined;
-        }
-        return (_b = (_a = url.split('.').pop()) === null || _a === void 0 ? void 0 : _a.split('?')[0]) === null || _b === void 0 ? void 0 : _b.toLowerCase();
-    },
+    getExtension,
     /**
      * Takes in a URL and checks if the file extension is PDF
      *
@@ -972,7 +995,7 @@ const Str = {
      * @returns Whether file path is PDF or not
      */
     isPDF(url) {
-        return this.getExtension(url) === 'pdf';
+        return getExtension(url) === 'pdf';
     },
     /**
      * Takes in a URL and checks if the file extension is an image
@@ -983,7 +1006,7 @@ const Str = {
      * https://reactnative.dev/docs/image#source
      */
     isImage(url) {
-        const extension = this.getExtension(url);
+        const extension = getExtension(url);
         if (!extension) {
             return false;
         }
@@ -999,7 +1022,7 @@ const Str = {
      * https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Video_codecs
      */
     isVideo(url) {
-        const extension = this.getExtension(url);
+        const extension = getExtension(url);
         if (!extension) {
             return false;
         }
@@ -1012,7 +1035,7 @@ const Str = {
      * @returns True if is a domain account email, otherwise false.
      */
     isDomainEmail(email) {
-        return this.startsWith(email, '+@');
+        return startsWith(email, '+@');
     },
     /**
      * Find the minimum indentation of any line in the string,
