diff --git a/node_modules/@sentry/core/build/esm/fetch.js b/node_modules/@sentry/core/build/esm/fetch.js
index 3c6eafb..33481e8 100644
--- a/node_modules/@sentry/core/build/esm/fetch.js
+++ b/node_modules/@sentry/core/build/esm/fetch.js
@@ -235,7 +235,10 @@ function _addTracingHeadersToFetchRequest(
 function endSpan(span, handlerData) {
   if (handlerData.response) {
     setHttpStatus(span, handlerData.response.status);
-
+    const requestId = handlerData.response?.headers?.get('x-request-id');
+    if (requestId) {
+        span.setAttribute('request-id', requestId)
+    }
     const contentLength = handlerData.response?.headers?.get('content-length');
 
     if (contentLength) {
