diff --git a/node_modules/react-native-release-profiler/ios/ReleaseProfiler.mm b/node_modules/react-native-release-profiler/ios/ReleaseProfiler.mm
index 975c4e3..af6c16a 100644
--- a/node_modules/react-native-release-profiler/ios/ReleaseProfiler.mm
+++ b/node_modules/react-native-release-profiler/ios/ReleaseProfiler.mm
@@ -1,5 +1,9 @@
 #import <React/RCTBridgeModule.h>
 #import <hermes/hermes.h>
+
+using namespace facebook::hermes;
+using namespace facebook::jsi;
+
 @interface RCT_EXTERN_MODULE(ReleaseProfiler, NSObject)
 
 RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
@@ -12,7 +16,8 @@ RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
 }
 
 RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(startProfiling) {
-    facebook::hermes::HermesRuntime::enableSamplingProfiler();
+    IHermesRootAPI *api = castInterface<IHermesRootAPI>(makeHermesRootAPI());
+    api->enableSamplingProfiler();
     return [NSNumber numberWithInt:1];
 }
 
@@ -27,8 +32,9 @@ RCT_EXPORT_METHOD(stopProfiling:(BOOL)saveInDownload
     [sharedFM createFileAtPath:fileURL.path
                                             contents:[@"" dataUsingEncoding:NSUTF8StringEncoding]
                                     attributes:nil];
-    facebook::hermes::HermesRuntime::dumpSampledTraceToFile(finalPath);
-    facebook::hermes::HermesRuntime::disableSamplingProfiler();
+    IHermesRootAPI *api = castInterface<IHermesRootAPI>(makeHermesRootAPI());
+    api->dumpSampledTraceToFile(finalPath);
+    api->disableSamplingProfiler();
     resolve(fileURL.path);
 }
 
