diff --git a/node_modules/@sentry/browser/build/npm/cjs/dev/tracing/request.js b/node_modules/@sentry/browser/build/npm/cjs/dev/tracing/request.js
index 6bfc053..21d2a37 100644
--- a/node_modules/@sentry/browser/build/npm/cjs/dev/tracing/request.js
+++ b/node_modules/@sentry/browser/build/npm/cjs/dev/tracing/request.js
@@ -227,6 +227,12 @@ function xhrCallback(
     const span = spans[spanId];
     if (span && sentryXhrData.status_code !== undefined) {
       core.setHttpStatus(span, sentryXhrData.status_code);
+      if (handlerData.xhr.getAllResponseHeaders().toLowerCase().includes('x-request-id')) {
+          const requestId = handlerData.xhr.getResponseHeader('x-request-id');
+          if (requestId) {
+              span.setAttribute('request-id', requestId)
+          }
+      }
       span.end();
 
       onRequestSpanEnd?.(span, {
diff --git a/node_modules/@sentry/browser/build/npm/cjs/prod/tracing/request.js b/node_modules/@sentry/browser/build/npm/cjs/prod/tracing/request.js
index 6bfc053..21d2a37 100644
--- a/node_modules/@sentry/browser/build/npm/cjs/prod/tracing/request.js
+++ b/node_modules/@sentry/browser/build/npm/cjs/prod/tracing/request.js
@@ -227,6 +227,12 @@ function xhrCallback(
     const span = spans[spanId];
     if (span && sentryXhrData.status_code !== undefined) {
       core.setHttpStatus(span, sentryXhrData.status_code);
+      if (handlerData.xhr.getAllResponseHeaders().toLowerCase().includes('x-request-id')) {
+          const requestId = handlerData.xhr.getResponseHeader('x-request-id');
+          if (requestId) {
+              span.setAttribute('request-id', requestId)
+          }
+      }
       span.end();
 
       onRequestSpanEnd?.(span, {
diff --git a/node_modules/@sentry/browser/build/npm/esm/dev/tracing/request.js b/node_modules/@sentry/browser/build/npm/esm/dev/tracing/request.js
index 57d2ae8..063b495 100644
--- a/node_modules/@sentry/browser/build/npm/esm/dev/tracing/request.js
+++ b/node_modules/@sentry/browser/build/npm/esm/dev/tracing/request.js
@@ -225,6 +225,12 @@ function xhrCallback(
     const span = spans[spanId];
     if (span && sentryXhrData.status_code !== undefined) {
       setHttpStatus(span, sentryXhrData.status_code);
+      if (handlerData.xhr.getAllResponseHeaders().toLowerCase().includes('x-request-id')) {
+          const requestId = handlerData.xhr.getResponseHeader('x-request-id');
+          if (requestId) {
+              span.setAttribute('request-id', requestId)
+          }
+      }
       span.end();
 
       onRequestSpanEnd?.(span, {
diff --git a/node_modules/@sentry/browser/build/npm/esm/prod/tracing/request.js b/node_modules/@sentry/browser/build/npm/esm/prod/tracing/request.js
index 57d2ae8..063b495 100644
--- a/node_modules/@sentry/browser/build/npm/esm/prod/tracing/request.js
+++ b/node_modules/@sentry/browser/build/npm/esm/prod/tracing/request.js
@@ -225,6 +225,12 @@ function xhrCallback(
     const span = spans[spanId];
     if (span && sentryXhrData.status_code !== undefined) {
       setHttpStatus(span, sentryXhrData.status_code);
+      if (handlerData.xhr.getAllResponseHeaders().toLowerCase().includes('x-request-id')) {
+          const requestId = handlerData.xhr.getResponseHeader('x-request-id');
+          if (requestId) {
+              span.setAttribute('request-id', requestId)
+          }
+      }
       span.end();
 
       onRequestSpanEnd?.(span, {
