diff --git a/node_modules/time-analytics-webpack-plugin/dist/loader.js b/node_modules/time-analytics-webpack-plugin/dist/loader.js
index 7a40d29..2a35893 100644
--- a/node_modules/time-analytics-webpack-plugin/dist/loader.js
+++ b/node_modules/time-analytics-webpack-plugin/dist/loader.js
@@ -57,9 +57,9 @@ function hackWrapLoaderModule(loaderPaths, wrapLoaderModuleCallback) {
     // TODO: check whether thread-loader works for this case or not
     Module.prototype.require = wrapRequire(originModuleRequire);
 }
-const loader = function timeAnalyticHackLoader(source) {
+const loader = function timeAnalyticHackLoader(source, sourceMap, meta) {
     // console.log('Time analytics plugin: normal loader is executed');
-    return source;
+    this.callback(null, source, sourceMap, meta);
 };
 /**
  * Override `require` method, so that we could return a wrapped loader function.
