diff --git a/node_modules/focus-trap/dist/focus-trap.esm.js b/node_modules/focus-trap/dist/focus-trap.esm.js
index 58539d3..f3d5758 100644
--- a/node_modules/focus-trap/dist/focus-trap.esm.js
+++ b/node_modules/focus-trap/dist/focus-trap.esm.js
@@ -183,8 +183,8 @@ var isKeyForward = function isKeyForward(e) {
 var isKeyBackward = function isKeyBackward(e) {
   return isTabEvent(e) && e.shiftKey;
 };
-var delay = function delay(fn) {
-  return setTimeout(fn, 0);
+var delay = function delay(fn, delayTime = 0) {
+  return setTimeout(() => setTimeout(fn, delayTime), 0);
 };
 
 /**
@@ -844,7 +844,7 @@ var createFocusTrap = function createFocusTrap(elements, userOptions) {
     // that caused the focus trap activation.
     state.delayInitialFocusTimer = config.delayInitialFocus ? delay(function () {
       _tryFocus(getInitialFocusNode());
-    }) : _tryFocus(getInitialFocusNode());
+    }, typeof config.delayInitialFocus === 'number' ? config.delayInitialFocus : undefined) : _tryFocus(getInitialFocusNode());
     doc.addEventListener('focusin', checkFocusIn, true);
     doc.addEventListener('mousedown', checkPointerDown, {
       capture: true,
@@ -1088,7 +1088,7 @@ var createFocusTrap = function createFocusTrap(elements, userOptions) {
             _tryFocus(getReturnFocusNode(state.nodeFocusedBeforeActivation));
           }
           onPostDeactivate === null || onPostDeactivate === void 0 || onPostDeactivate();
-        });
+        }, typeof config.delayInitialFocus === 'number' ? config.delayInitialFocus : undefined);
       };
       if (returnFocus && checkCanReturnFocus) {
         checkCanReturnFocus(getReturnFocusNode(state.nodeFocusedBeforeActivation)).then(finishDeactivation, finishDeactivation);
diff --git a/node_modules/focus-trap/index.d.ts b/node_modules/focus-trap/index.d.ts
index fe40111..27afd01 100644
--- a/node_modules/focus-trap/index.d.ts
+++ b/node_modules/focus-trap/index.d.ts
@@ -187,7 +187,7 @@ declare module 'focus-trap' {
      * This prevents elements within the focusable element from capturing
      * the event that triggered the focus trap activation.
      */
-    delayInitialFocus?: boolean;
+    delayInitialFocus?: boolean | number;
     /**
      * Default: `false`. Prevents screen readers from accessing content outside
      * the focus trap. Values of `true` or `'inert'` use the `inert` attribute,
diff --git a/node_modules/focus-trap/index.js b/node_modules/focus-trap/index.js
index 8a463e2..b61052c 100644
--- a/node_modules/focus-trap/index.js
+++ b/node_modules/focus-trap/index.js
@@ -85,8 +85,8 @@ const isKeyBackward = function (e) {
   return isTabEvent(e) && e.shiftKey;
 };
 
-const delay = function (fn) {
-  return setTimeout(fn, 0);
+const delay = function (fn, delayTime = 0) {
+  return setTimeout(() => setTimeout(fn, delayTime), 0);
 };
 
 /**
@@ -860,7 +860,7 @@ const createFocusTrap = function (elements, userOptions) {
     state.delayInitialFocusTimer = config.delayInitialFocus
       ? delay(function () {
           tryFocus(getInitialFocusNode());
-        })
+        }, typeof config.delayInitialFocus === 'number' ? config.delayInitialFocus : undefined)
       : tryFocus(getInitialFocusNode());
 
     doc.addEventListener('focusin', checkFocusIn, true);
@@ -1130,7 +1130,7 @@ const createFocusTrap = function (elements, userOptions) {
             tryFocus(getReturnFocusNode(state.nodeFocusedBeforeActivation));
           }
           onPostDeactivate?.();
-        });
+        }, typeof config.delayInitialFocus === 'number' ? config.delayInitialFocus : undefined);
       };
 
       if (returnFocus && checkCanReturnFocus) {
