diff --git a/node_modules/react-native-reanimated/src/createAnimatedComponent/createAnimatedComponent.tsx b/node_modules/react-native-reanimated/src/createAnimatedComponent/createAnimatedComponent.tsx
index a2f6cf1..93a37f5 100644
--- a/node_modules/react-native-reanimated/src/createAnimatedComponent/createAnimatedComponent.tsx
+++ b/node_modules/react-native-reanimated/src/createAnimatedComponent/createAnimatedComponent.tsx
@@ -503,13 +503,6 @@ export function createAnimatedComponent(
         return;
       }
 
-      if (this._isReducedMotion(currentConfig)) {
-        if (!previousConfig) {
-          return;
-        }
-        currentConfig = undefined;
-      }
-
       updateLayoutAnimations(
         isFabric() && type === LayoutAnimationType.ENTERING
           ? this.reanimatedID
@@ -608,14 +601,6 @@ export function createAnimatedComponent(
       },
     });
 
-    _isReducedMotion(config?: LayoutAnimationOrBuilder): boolean {
-      return config &&
-        'getReduceMotion' in config &&
-        typeof config.getReduceMotion === 'function'
-        ? getReduceMotionFromConfig(config.getReduceMotion())
-        : getReduceMotionFromConfig();
-    }
-
     // This is a component lifecycle method from React, therefore we are not calling it directly.
     // It is called before the component gets rerendered. This way we can access components' position before it changed
     // and later on, in componentDidUpdate, calculate translation for layout transition.
