diff --git a/node_modules/react-native-draggable-flatlist/src/components/NestableDraggableFlatList.tsx b/node_modules/react-native-draggable-flatlist/src/components/NestableDraggableFlatList.tsx
index 1559352..b84ee99 100644
--- a/node_modules/react-native-draggable-flatlist/src/components/NestableDraggableFlatList.tsx
+++ b/node_modules/react-native-draggable-flatlist/src/components/NestableDraggableFlatList.tsx
@@ -56,6 +56,11 @@ function NestableDraggableFlatListInner<T>(
     const onFail = () => {
       console.log("## nested draggable list measure fail");
     };
+
+    if (typeof nodeHandle === "number" ) {
+      return;
+    }
+
     //@ts-ignore
     containerRef.current.measureLayout(nodeHandle, onSuccess, onFail);
   });
