diff --git a/node_modules/@rnmapbox/maps/android/build.gradle b/node_modules/@rnmapbox/maps/android/build.gradle
index dd8e6c6..dfca30e 100644
--- a/node_modules/@rnmapbox/maps/android/build.gradle
+++ b/node_modules/@rnmapbox/maps/android/build.gradle
@@ -84,6 +84,10 @@ android {
         }
 
         packagingOptions {
+            jniLibs {
+                useLegacyPackaging true
+            }
+
             pickFirst 'lib/x86/libc++_shared.so'
             pickFirst 'lib/x86_64/libc++_shared.so'
             pickFirst 'lib/arm64-v8a/libc++_shared.so'
@@ -113,6 +117,12 @@ android {
         versionCode 1
         versionName "1.0"
         buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
+
+        externalNativeBuild {
+            cmake {
+                arguments "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
+            }
+        }
     }
 
     compileOptions {
