diff --git a/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx b/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
index d4a0a0e..0000001 100644
--- a/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
+++ b/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
@@ -134,9 +134,10 @@ export function PagerViewAdapter<T extends Route>({
     };
   });
 
+  const [forceRender, setForceRender] = React.useState(0);
   const memoizedPosition = React.useMemo(
     () => Animated.add(position, offset),
-    [offset, position]
+    [offset, position, forceRender]
   );
 
   return children({
@@ -163,9 +164,11 @@ export function PagerViewAdapter<T extends Route>({
           ],
           { useNativeDriver }
         )}
         onPageSelected={(e) => {
           const index = e.nativeEvent.position;
           indexRef.current = index;
+          position.setValue(index);
+          setForceRender((fr) => fr+1)
           onIndexChange(index);
           onTabSelect?.({ index });
         }}
