diff --git a/node_modules/@react-native/virtualized-lists/Lists/VirtualizedList.js b/node_modules/@react-native/virtualized-lists/Lists/VirtualizedList.js
index 5b8aaa5..45ea377 100644
--- a/node_modules/@react-native/virtualized-lists/Lists/VirtualizedList.js
+++ b/node_modules/@react-native/virtualized-lists/Lists/VirtualizedList.js
@@ -90,6 +90,7 @@ type State = {
   firstVisibleItemKey: ?string,
   // When > 0 the scroll position available in JS is considered stale and should not be used.
   pendingScrollUpdateCount: number,
+  lastItemCount: number,
 };
 
 function getScrollingThreshold(threshold: number, visibleLength: number) {
@@ -403,12 +404,13 @@ class VirtualizedList extends StateSafePureComponent<
 
     const minIndexForVisible =
       this.props.maintainVisibleContentPosition?.minIndexForVisible ?? 0;
+      const itemCount = this.props.getItemCount(this.props.data);
 
     this.state = {
       cellsAroundViewport: initialRenderRegion,
       renderMask: VirtualizedList._createRenderMask(props, initialRenderRegion),
       firstVisibleItemKey:
-        this.props.getItemCount(this.props.data) > minIndexForVisible
+        itemCount > minIndexForVisible
           ? VirtualizedList._getItemKey(this.props, minIndexForVisible)
           : null,
       // When we have a non-zero initialScrollIndex, we will receive a
@@ -419,6 +421,7 @@ class VirtualizedList extends StateSafePureComponent<
         this.props.initialScrollIndex > 0
           ? 1
           : 0,
+        lastItemCount: itemCount,
     };
   }
 
@@ -706,7 +709,7 @@ class VirtualizedList extends StateSafePureComponent<
     // first and last could be stale (e.g. if a new, shorter items props is passed in), so we make
     // sure we're rendering a reasonable range here.
     const itemCount = newProps.getItemCount(newProps.data);
-    if (itemCount === prevState.renderMask.numCells()) {
+    if (itemCount === prevState.renderMask.numCells() && itemCount === prevState.lastItemCount) {
       return prevState;
     }
 
@@ -715,7 +718,7 @@ class VirtualizedList extends StateSafePureComponent<
     const minIndexForVisible =
       newProps.maintainVisibleContentPosition?.minIndexForVisible ?? 0;
     const newFirstVisibleItemKey =
-      newProps.getItemCount(newProps.data) > minIndexForVisible
+      itemCount > minIndexForVisible
         ? VirtualizedList._getItemKey(newProps, minIndexForVisible)
         : null;
     if (
@@ -763,6 +766,7 @@ class VirtualizedList extends StateSafePureComponent<
         maintainVisibleContentPositionAdjustment != null
           ? prevState.pendingScrollUpdateCount + 1
           : prevState.pendingScrollUpdateCount,
+        lastItemCount: itemCount,
     };
   }
 
@@ -1183,7 +1187,7 @@ class VirtualizedList extends StateSafePureComponent<
     }
   }
 
-  componentDidUpdate(prevProps: VirtualizedListProps) {
+  componentDidUpdate(prevProps: VirtualizedListProps, prevState: State) {
     const {data, extraData, getItemLayout} = this.props;
     if (data !== prevProps.data || extraData !== prevProps.extraData) {
       // clear the viewableIndices cache to also trigger
@@ -1205,7 +1209,10 @@ class VirtualizedList extends StateSafePureComponent<
     if (hiPriInProgress) {
       this._hiPriInProgress = false;
     }
-
+    if (this.state.cellsAroundViewport.first !== prevState.cellsAroundViewport.first ||
+      this.state.cellsAroundViewport.last !== prevState.cellsAroundViewport.last) {
+      this._maybeCallOnEdgeReached();
+    }
     // We only call `onEndReached` after we render the last cell, but when
     // getItemLayout is present, we can scroll past the last rendered cell, and
     // never trigger a new layout or bounds change, so we need to check again
@@ -1251,7 +1258,7 @@ class VirtualizedList extends StateSafePureComponent<
     zoomScale: 1,
   };
   _scrollRef: ?React.ElementRef<typeof ScrollView> = null;
-  _sentStartForContentLength = 0;
+  _sentStartForFirstVisibleItemKey: ?string = null;
   _sentEndForContentLength = 0;
   _updateCellsToRenderTimeoutID: ?TimeoutID = null;
   _viewabilityTuples: Array<ViewabilityHelperCallbackTuple> = [];
@@ -1595,16 +1602,16 @@ class VirtualizedList extends StateSafePureComponent<
       onStartReached != null &&
       this.state.cellsAroundViewport.first === 0 &&
       isWithinStartThreshold &&
-      this._listMetrics.getContentLength() !== this._sentStartForContentLength
+      this.state.firstVisibleItemKey !== this._sentStartForFirstVisibleItemKey
     ) {
-      this._sentStartForContentLength = this._listMetrics.getContentLength();
+      this._sentStartForFirstVisibleItemKey = this.state.firstVisibleItemKey;
       onStartReached({distanceFromStart});
     }
 
     // If the user scrolls away from the start or end and back again,
     // cause onStartReached or onEndReached to be triggered again
     if (!isWithinStartThreshold) {
-      this._sentStartForContentLength = 0;
+      this._sentStartForFirstVisibleItemKey = null;
     }
     if (!isWithinEndThreshold) {
       this._sentEndForContentLength = 0;
