diff --git a/node_modules/@sentry/core/build/esm/fetch.js b/node_modules/@sentry/core/build/esm/fetch.js
index da1d1ac..b1e7d38 100644
--- a/node_modules/@sentry/core/build/esm/fetch.js
+++ b/node_modules/@sentry/core/build/esm/fetch.js
@@ -215,6 +215,8 @@ function endSpan(span, handlerData) {
   if (handlerData.response) {
     setHttpStatus(span, handlerData.response.status);
 
+    span.setAttribute('cf-ray', handlerData.response?.headers?.get('cf-ray') ?? 'unknown')
+
     const contentLength = handlerData.response?.headers?.get('content-length');
 
     if (contentLength) {
