diff --git a/node_modules/react-compiler-healthcheck/dist/index.js b/node_modules/react-compiler-healthcheck/dist/index.js
index 460339b..17b0f96 100755
--- a/node_modules/react-compiler-healthcheck/dist/index.js
+++ b/node_modules/react-compiler-healthcheck/dist/index.js
@@ -56902,6 +56902,9 @@ const COMPILER_OPTIONS = {
   noEmit: true,
   compilationMode: 'infer',
   panicThreshold: 'critical_errors',
+  environment: {
+    enableTreatRefLikeIdentifiersAsRefs: true,
+  },
   logger: logger,
 };
 function isActionableDiagnostic(detail) {
diff --git a/node_modules/react-compiler-healthcheck/src/checks/reactCompiler.ts b/node_modules/react-compiler-healthcheck/src/checks/reactCompiler.ts
index 3094548..fd05b76 100644
--- a/node_modules/react-compiler-healthcheck/src/checks/reactCompiler.ts
+++ b/node_modules/react-compiler-healthcheck/src/checks/reactCompiler.ts
@@ -50,6 +50,9 @@ const COMPILER_OPTIONS: Partial<PluginOptions> = {
   noEmit: true,
   compilationMode: 'infer',
   panicThreshold: 'critical_errors',
+  environment: {
+    enableTreatRefLikeIdentifiersAsRefs: true,
+  },
   logger,
 };
 
